From rpjday@crashcourse.ca Fri Apr 25 05:52:59 2008
From: "Robert P. J. Day" <rpjday@crashcourse.ca>
Date: Fri, 25 Apr 2008 08:52:51 -0400 (EDT)
Subject: DEBUGFS: Correct location of debugfs API documentation.
To: Greg KH <greg@kroah.com>
Message-ID: <alpine.LFD.1.10.0804250851530.9369@localhost.localdomain>



Signed-off-by: Robert P. J. Day <rpjday@crashcourse.ca>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/debugfs/file.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/debugfs/file.c
+++ b/fs/debugfs/file.c
@@ -9,7 +9,7 @@
  *	2 as published by the Free Software Foundation.
  *
  *  debugfs is for people to use instead of /proc or /sys.
- *  See Documentation/DocBook/kernel-api for more details.
+ *  See Documentation/DocBook/filesystems for more details.
  *
  */
 
