From jesper.juhl@gmail.com  Mon May 19 16:58:21 2008
From: Robert P. J. Day <rpjday@crashcourse.ca>
Date: Tue, 20 May 2008 00:06:00 +0200
Subject: debugfs: Add a reference to the debugfs API documentation.
To: "Greg KH" <greg@kroah.com>
Cc: "Greg Kroah-Hartman" <gregkh@suse.de>, trivial@kernel.org,  "Robert P. J. Day" <rpjday@crashcourse.ca>
Message-ID: <9a8748490805191506p752e1c9ay7c6d6d37d9dfb25b@mail.gmail.com>
Content-Disposition: inline


Signed-off-by: Robert P. J. Day <rpjday@crashcourse.ca>
Cc: Jesper Juhl <jesper.juhl@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 lib/Kconfig.debug |    3 +++
 1 file changed, 3 insertions(+)

--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -74,6 +74,9 @@ config DEBUG_FS
 	  debugging files into.  Enable this option to be able to read and
 	  write to these files.
 
+	  For detailed documentation on the debugfs API, see
+	  Documentation/DocBook/filesystems.
+
 	  If unsure, say N.
 
 config HEADERS_CHECK
