From kay.sievers@vrfy.org Fri May  2 09:28:46 2008
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Fri, 02 May 2008 06:02:41 +0200
Subject: driver core: lguest: convert to new device API to allow names longer than 20 chars
Message-ID: <1209700961.3100.8.camel@linux.site>


Cc: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/lguest/lguest_device.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/lguest/lguest_device.c
+++ b/drivers/lguest/lguest_device.c
@@ -311,7 +311,7 @@ static struct virtio_config_ops lguest_c
  * /sys/devices/lguest/0,1,2 not /sys/devices/0,1,2. */
 static struct device lguest_root = {
 	.parent = NULL,
-	.bus_id = "lguest",
+	.init_name = "lguest",
 };
 
 /*D:120 This is the core of the lguest bus: actually adding a new device.
