From tshibata@ab.jp.nec.com  Fri Jun 27 14:41:46 2008
From: Tsugikazu Shibata <tshibata@ab.jp.nec.com>
Date: Fri, 20 Jun 2008 10:59:52 +0900 (JST)
Subject: HOWTO: change email addresses of James in HOWTO
To: greg@kroah.com, James.Bottomley@HansenPartnership.com
Cc: tshibata@ab.jp.nec.com
Message-ID: <20080620.105952.95887859.tshibata@ab.jp.nec.com>


Signed-off-by: Tsugikazu Shibata <tshibata@ab.jp.nec.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/HOWTO |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Documentation/HOWTO
+++ b/Documentation/HOWTO
@@ -358,7 +358,7 @@ Here is a list of some of the different 
     - pcmcia, Dominik Brodowski <linux@dominikbrodowski.net>
 	git.kernel.org:/pub/scm/linux/kernel/git/brodo/pcmcia-2.6.git
 
-    - SCSI, James Bottomley <James.Bottomley@SteelEye.com>
+    - SCSI, James Bottomley <James.Bottomley@hansenpartnership.com>
 	git.kernel.org:/pub/scm/linux/kernel/git/jejb/scsi-misc-2.6.git
 
     - x86, Ingo Molnar <mingo@elte.hu>
