From stefanr@s5r6.in-berlin.de  Mon Oct 26 16:32:38 2009
From: Stefan Richter <stefanr@s5r6.in-berlin.de>
Date: Wed, 14 Oct 2009 20:47:32 +0200 (CEST)
Subject: sysfs: mark a locally-only used function static
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: James Morris <jmorris@namei.org>, "Stephen D. Smalley" <sds@tycho.nsa.gov>, linux-kernel@vger.kernel.org, "David P. Quigley" <dpquigl@tycho.nsa.gov>
Message-ID: <tkrat.7aacb9f187b9bcfe@s5r6.in-berlin.de>


Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>
Acked-by: David P. Quigley <dpquigl@tycho.nsa.gov>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/sysfs/inode.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/fs/sysfs/inode.c
+++ b/fs/sysfs/inode.c
@@ -46,7 +46,7 @@ int __init sysfs_inode_init(void)
 	return bdi_init(&sysfs_backing_dev_info);
 }
 
-struct sysfs_inode_attrs *sysfs_init_inode_attrs(struct sysfs_dirent *sd)
+static struct sysfs_inode_attrs *sysfs_init_inode_attrs(struct sysfs_dirent *sd)
 {
 	struct sysfs_inode_attrs *attrs;
 	struct iattr *iattrs;
@@ -64,6 +64,7 @@ struct sysfs_inode_attrs *sysfs_init_ino
 
 	return attrs;
 }
+
 int sysfs_setattr(struct dentry * dentry, struct iattr * iattr)
 {
 	struct inode * inode = dentry->d_inode;
