From 890e799692e95bda436159e2a4176b00652eabaf Mon Sep 17 00:00:00 2001
From: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Date: Sun, 26 Apr 2009 16:04:31 +0200
Subject: [PATCH 011/102] Staging: rt2870: remove dead WSC_AP_SUPPORT code

Signed-off-by: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/rt2870/ap.h |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/staging/rt2870/ap.h
+++ b/drivers/staging/rt2870/ap.h
@@ -490,9 +490,6 @@ BOOLEAN PeerAssocReqCmmSanity(
     OUT UCHAR *RSN,
     OUT UCHAR *pRSNLen,
     OUT BOOLEAN *pbWmmCapable,
-#ifdef WSC_AP_SUPPORT
-    OUT BOOLEAN *pWscCapable,
-#endif // WSC_AP_SUPPORT //
     OUT ULONG  *pRalinkIe,
 #ifdef DOT11N_DRAFT3
     OUT EXT_CAP_INFO_ELEMENT	*pExtCapInfo,
