From alan@linux.intel.com  Mon Oct 26 14:59:21 2009
From: Alan Cox <alan@linux.intel.com>
Date: Tue, 06 Oct 2009 15:50:20 +0100
Subject: Staging: et131x: Kill the RX skb list element - it isn't used
To: greg@kroah.com
Message-ID: <20091006145018.8604.14087.stgit@localhost.localdomain>


Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/et131x/et1310_rx.h |    9 ---------
 1 file changed, 9 deletions(-)

--- a/drivers/staging/et131x/et1310_rx.h
+++ b/drivers/staging/et131x/et1310_rx.h
@@ -272,15 +272,6 @@ typedef enum {
 } eRX_INTERRUPT_STATE_t, *PeRX_INTERRUPT_STATE_t;
 
 /*
- * Structure to hold the skb's in a list
- */
-typedef struct rx_skb_list_elem {
-	struct list_head skb_list_elem;
-	dma_addr_t dma_addr;
-	struct sk_buff *skb;
-} RX_SKB_LIST_ELEM, *PRX_SKB_LIST_ELEM;
-
-/*
  * RX_RING_t is sructure representing the adaptor's local reference(s) to the
  * rings
  */
