From forest@alittletooquiet.net  Thu Jul 16 16:31:07 2009
From: miaofng <miaofng@126.com>
Date: Wed, 15 Jul 2009 10:33:35 -0400
Subject: Staging: vt6656: disable wpa related function due to memory failure of this part of codes
To: miaofng <miaofng@126.com>
Cc: Greg KH <greg@kroah.com>, Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>, Alexander Beregalov <a.beregalov@gmail.com>, Jim Lieb <jim@canonical.com>
Message-ID: <20090715143335.GA28901@alittletooquiet.net>
Content-Disposition: inline


From: miaofng <miaofng@126.com>

Signed-off-by: miaofng <miaofng@126.com>
Cc: Forest Bond <forest@alittletooquiet.net>
Cc: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/vt6656/main_usb.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/staging/vt6656/main_usb.c
+++ b/drivers/staging/vt6656/main_usb.c
@@ -879,13 +879,14 @@ vntwusb_found1(struct usb_interface *int
 
 //2008-07-21-01<Add>by MikeLiu
 //register wpadev
+#if 0
    if(wpa_set_wpadev(pDevice, 1)!=0) {
      printk("Fail to Register WPADEV?\n");
         unregister_netdev(pDevice->dev);
         free_netdev(netdev);
         kfree(pDevice);
    }
-
+#endif
          usb_device_reset(pDevice);
 
 #ifdef SndEvt_ToAPI
