From alan@lxorguk.ukuu.org.uk  Mon Aug 17 12:34:36 2009
From: Alan Cox <alan@lxorguk.ukuu.org.uk>
Date: Fri, 14 Aug 2009 15:40:32 +0100
Subject: Staging: sep: fix a conversion thinko
To: greg@kroah.com, mark.a.allyn@intel.com
Message-ID: <20090814144028.6066.92078.stgit@localhost.localdomain>


From: Alan Cox <alan@linux.intel.com>

Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/sep/sep_driver.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)


--- a/drivers/staging/sep/sep_driver.c
+++ b/drivers/staging/sep/sep_driver.c
@@ -277,7 +277,7 @@ static int sep_map_and_alloc_shared_area
 							unsigned long size)
 {
 	/* shared_addr = ioremap_nocache(0xda00000,shared_area_size); */
-	sep->shared_area = dma_alloc_coherent(&sep->pdev->dev, size,
+	sep->shared_addr = dma_alloc_coherent(&sep->pdev->dev, size,
 					&sep->shared_bus, GFP_KERNEL);
 
 	if (!sep->shared_addr) {
