From wfp5p@virginia.edu Thu Mar 19 13:06:32 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Thu, 19 Mar 2009 17:59:55 -0400
Subject: Staging: comedi: Remove me4000_ao_info_t typedef
To: greg@kroah.com
Cc: abbotti@mev.co.uk, fmhess@users.sourceforge.net, ds@schleef.org
Message-ID: <20090319215954.10220.82054.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/me4000.h |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/comedi/drivers/me4000.h
+++ b/drivers/staging/comedi/drivers/me4000.h
@@ -293,10 +293,10 @@
   Information about the hardware capabilities
   ===========================================================================*/
 
-typedef struct me4000_ao_info {
+struct me4000_ao_info {
 	int count;
 	int fifo_count;
-} me4000_ao_info_t;
+};
 
 typedef struct me4000_ai_info {
 	int count;
@@ -316,7 +316,7 @@ typedef struct me4000_cnt_info {
 typedef struct me4000_board {
 	const char *name;
 	unsigned short device_id;
-	me4000_ao_info_t ao;
+	struct me4000_ao_info ao;
 	me4000_ai_info_t ai;
 	me4000_dio_info_t dio;
 	me4000_cnt_info_t cnt;
