From wfp5p@virginia.edu  Mon Jul 27 14:08:56 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 27 Jul 2009 16:47:32 -0400
Subject: Staging: hv: RndisFilter.c: fix print formatting
To: greg@kroah.com
Cc: hjanssen@microsoft.com
Message-ID: <1248727665-23969-11-git-send-email-wfp5p@virginia.edu>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/RndisFilter.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/hv/RndisFilter.c
+++ b/drivers/staging/hv/RndisFilter.c
@@ -409,7 +409,7 @@ RndisFilterReceiveResponse(
 		}
 		else
 		{
-			DPRINT_ERR(NETVSC, "rndis response buffer overflow detected (size %u max %u)", Response->MessageLength, sizeof(RNDIS_FILTER_PACKET));
+			DPRINT_ERR(NETVSC, "rndis response buffer overflow detected (size %u max %lu)", Response->MessageLength, sizeof(RNDIS_FILTER_PACKET));
 
 			if (Response->NdisMessageType == REMOTE_NDIS_RESET_CMPLT) /* does not have a request id field */
 			{
@@ -539,7 +539,7 @@ RndisFilterOnReceive(
 
 	if ((rndisHeader->NdisMessageType != REMOTE_NDIS_PACKET_MSG) && (rndisHeader->MessageLength > sizeof(RNDIS_MESSAGE)))
 	{
-		DPRINT_ERR(NETVSC, "incoming rndis message buffer overflow detected (got %u, max %u)...marking it an error!",
+		DPRINT_ERR(NETVSC, "incoming rndis message buffer overflow detected (got %u, max %lu)...marking it an error!",
 			rndisHeader->MessageLength, sizeof(RNDIS_MESSAGE));
 	}
 
@@ -744,7 +744,7 @@ RndisFilterInit(
 {
 	DPRINT_ENTER(NETVSC);
 
-	DPRINT_DBG(NETVSC, "sizeof(RNDIS_FILTER_PACKET) == %d", sizeof(RNDIS_FILTER_PACKET));
+	DPRINT_DBG(NETVSC, "sizeof(RNDIS_FILTER_PACKET) == %ld", sizeof(RNDIS_FILTER_PACKET));
 
 	Driver->RequestExtSize = sizeof(RNDIS_FILTER_PACKET);
 	Driver->AdditionalRequestPageBufferCount = 1; /* For rndis header */
