From wfp5p@virginia.edu Thu Mar 19 13:05:46 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Thu, 19 Mar 2009 17:59:07 -0400
Subject: Staging: comedi: Remove ni_660x_private typedef
To: greg@kroah.com
Cc: abbotti@mev.co.uk, fmhess@users.sourceforge.net, ds@schleef.org
Message-ID: <20090319215907.10220.95005.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/ni_660x.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/drivers/staging/comedi/drivers/ni_660x.c
+++ b/drivers/staging/comedi/drivers/ni_660x.c
@@ -421,7 +421,7 @@ static DEFINE_PCI_DEVICE_TABLE(ni_660x_p
 
 MODULE_DEVICE_TABLE(pci, ni_660x_pci_table);
 
-typedef struct {
+struct ni_660x_private {
 	struct mite_struct *mite;
 	struct ni_gpct_device *counter_dev;
 	uint64_t pfi_direction_bits;
@@ -431,9 +431,9 @@ typedef struct {
 	unsigned dma_configuration_soft_copies[NI_660X_MAX_NUM_CHIPS];
 	spinlock_t soft_reg_copy_lock;
 	unsigned short pfi_output_selects[NUM_PFI_CHANNELS];
-} ni_660x_private;
+};
 
-static inline ni_660x_private *private(struct comedi_device * dev)
+static inline struct ni_660x_private *private(struct comedi_device * dev)
 {
 	return dev->private;
 }
@@ -765,7 +765,7 @@ static unsigned ni_gpct_read_register(st
 		ni_660x_register);
 }
 
-static inline struct mite_dma_descriptor_ring *mite_ring(ni_660x_private * priv,
+static inline struct mite_dma_descriptor_ring *mite_ring(struct ni_660x_private * priv,
 	struct ni_gpct *counter)
 {
 	return priv->mite_rings[counter->chip_index][counter->counter_index];
@@ -943,7 +943,7 @@ static int ni_660x_allocate_private(stru
 	int retval;
 	unsigned i;
 
-	if ((retval = alloc_private(dev, sizeof(ni_660x_private))) < 0)
+	if ((retval = alloc_private(dev, sizeof(struct ni_660x_private))) < 0)
 		return retval;
 	spin_lock_init(&private(dev)->mite_channel_lock);
 	spin_lock_init(&private(dev)->soft_reg_copy_lock);
