From wfp5p@virginia.edu  Wed Mar 18 21:47:33 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:18:54 -0400
Subject: Staging: comedi: serial2002: Remove serial2002_private typedef
To: greg@kroah.com
Message-ID: <20090317021853.14467.87393.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/serial2002.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/serial2002.c
+++ b/drivers/staging/comedi/drivers/serial2002.c
@@ -68,7 +68,8 @@ struct serial2002_range_table_t {
 };
 
 
-typedef struct {
+struct serial2002_private {
+
 	int port;		// /dev/ttyS<port>
 	int speed;		// baudrate
 	struct file *tty;
@@ -79,13 +80,14 @@ typedef struct {
 	unsigned char analog_out_mapping[32];
 	unsigned char encoder_in_mapping[32];
 	struct serial2002_range_table_t in_range[32], out_range[32];
-} serial2002_private;
+};
+
 
 /*
  * most drivers define the following macro to make it easy to
  * access the private structure.
  */
-#define devpriv ((serial2002_private *)dev->private)
+#define devpriv ((struct serial2002_private *)dev->private)
 
 static int serial2002_attach(struct comedi_device * dev, struct comedi_devconfig * it);
 static int serial2002_detach(struct comedi_device * dev);
@@ -786,7 +788,7 @@ static int serial2002_attach(struct come
 
 	printk("comedi%d: serial2002: ", dev->minor);
 	dev->board_name = thisboard->name;
-	if (alloc_private(dev, sizeof(serial2002_private)) < 0) {
+	if (alloc_private(dev, sizeof(struct serial2002_private)) < 0) {
 		return -ENOMEM;
 	}
 	dev->open = serial_2002_open;
