From akpm@linux-foundation.org  Mon Apr  6 12:22:01 2009
From: Daniele Napolitano <dnax88@gmail.com>
Date: Wed, 01 Apr 2009 14:53:39 -0700
Subject: Staging: rt2870: add ID for Sitecom WL-315
To: mm-commits@vger.kernel.org
Cc: dnax88@gmail.com, arnd@arndb.de, greg@kroah.com
Message-ID: <200904012153.n31LrdOG032444@imap1.linux-foundation.org>


From: Daniele Napolitano <dnax88@gmail.com>

Signed-off-by: Daniele Napolitano <dnax88@gmail.com>
Cc: Arnd Bergmann <arnd@arndb.de>
Cc: Greg KH <greg@kroah.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rt2870/rt2870.h |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/staging/rt2870/rt2870.h
+++ b/drivers/staging/rt2870/rt2870.h
@@ -96,6 +96,7 @@
 	{USB_DEVICE(0x0DF6,0x002B)}, /* Sitecom */		\
 	{USB_DEVICE(0x0DF6,0x002C)}, /* Sitecom */		\
 	{USB_DEVICE(0x0DF6,0x002D)}, /* Sitecom */		\
+	{USB_DEVICE(0x0DF6,0x0039)}, /* Sitecom */		\
 	{USB_DEVICE(0x14B2,0x3C06)}, /* Conceptronic */		\
 	{USB_DEVICE(0x14B2,0x3C28)}, /* Conceptronic */		\
 	{USB_DEVICE(0x2019,0xED06)}, /* Planex Communications, Inc. */		\
