From alan@linux.intel.com  Thu Aug 27 14:24:18 2009
From: Alan Cox <alan@linux.intel.com>
Date: Thu, 27 Aug 2009 10:59:21 +0100
Subject: Staging: et131x: Take a kref for the PCI pointer we cache
To: greg@kroah.com
Message-ID: <20090827095918.23892.13310.stgit@localhost.localdomain>


Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/et131x/et131x_initpci.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)


--- a/drivers/staging/et131x/et131x_initpci.c
+++ b/drivers/staging/et131x/et131x_initpci.c
@@ -742,6 +742,7 @@ void __devexit et131x_pci_remove(struct 
 	unregister_netdev(netdev);
 	et131x_adapter_memory_free(adapter);
 	iounmap(adapter->CSRAddress);
+	pci_dev_put(adapter->pdev);
 	free_netdev(netdev);
 	pci_release_regions(pdev);
 	pci_disable_device(pdev);
@@ -878,7 +879,7 @@ int __devinit et131x_pci_setup(struct pc
 
 	/* Allocate private adapter struct and copy in relevant information */
 	adapter = netdev_priv(netdev);
-	adapter->pdev = pdev;
+	adapter->pdev = pci_dev_get(pdev);
 	adapter->netdev = netdev;
 
 	/* Do the same for the netdev struct */
@@ -1022,6 +1023,7 @@ err_mem_free:
 err_iounmap:
 	iounmap(adapter->CSRAddress);
 err_free_dev:
+	pci_dev_put(pdev);
 	free_netdev(netdev);
 err_release_res:
 	pci_release_regions(pdev);
