From wfp5p@virginia.edu Mon Mar 16 17:34:01 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:19:15 -0400
Subject: Staging: comedi: Remove rti800_private typedef
To: greg@kroah.com
Message-ID: <20090317021915.14467.52851.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/rti800.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/rti800.c
+++ b/drivers/staging/comedi/drivers/rti800.c
@@ -145,7 +145,7 @@ COMEDI_INITCLEANUP(driver_rti800);
 
 static irqreturn_t rti800_interrupt(int irq, void *dev PT_REGS_ARG);
 
-typedef struct {
+struct rti800_private {
 	enum {
 		adc_diff, adc_pseudodiff, adc_singleended
 	} adc_mux;
@@ -164,9 +164,9 @@ typedef struct {
 	const struct comedi_lrange *ao_range_type_list[2];
 	unsigned int ao_readback[2];
 	int muxgain_bits;
-} rti800_private;
+};
 
-#define devpriv ((rti800_private *)dev->private)
+#define devpriv ((struct rti800_private *)dev->private)
 
 #define RTI800_TIMEOUT 100
 
@@ -351,7 +351,7 @@ static int rti800_attach(struct comedi_d
 
 	if ((ret = alloc_subdevices(dev, 4)) < 0)
 		return ret;
-	if ((ret = alloc_private(dev, sizeof(rti800_private))) < 0)
+	if ((ret = alloc_private(dev, sizeof(struct rti800_private))) < 0)
 		return ret;
 
 	devpriv->adc_mux = it->options[2];
