From wfp5p@virginia.edu Mon Mar 16 17:28:51 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:14:03 -0400
Subject: Staging: comedi: Remove das6402_private typedef
To: greg@kroah.com
Message-ID: <20090317021403.14467.10347.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/das6402.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/das6402.c
+++ b/drivers/staging/comedi/drivers/das6402.c
@@ -109,12 +109,12 @@ static struct comedi_driver driver_das64
 
 COMEDI_INITCLEANUP(driver_das6402);
 
-typedef struct {
+struct das6402_private {
 	int ai_bytes_to_read;
 
 	int das6402_ignoreirq;
-} das6402_private;
-#define devpriv ((das6402_private *)dev->private)
+};
+#define devpriv ((struct das6402_private *)dev->private)
 
 static void das6402_ai_fifo_dregs(struct comedi_device * dev, struct comedi_subdevice * s);
 
@@ -331,7 +331,7 @@ static int das6402_attach(struct comedi_
 	}
 	dev->irq = irq;
 
-	if ((ret = alloc_private(dev, sizeof(das6402_private))) < 0)
+	if ((ret = alloc_private(dev, sizeof(struct das6402_private))) < 0)
 		return ret;
 
 	if ((ret = alloc_subdevices(dev, 1)) < 0)
