From shawn.bohrer@gmail.com  Thu Nov 19 14:35:23 2009
From: Shawn Bohrer <shawn.bohrer@gmail.com>
Date: Sun, 15 Nov 2009 22:17:53 -0600
Subject: staging: line6: Fix checkpatch errors in capture.c
To: Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <1258345082-11811-5-git-send-email-shawn.bohrer@gmail.com>


Signed-off-by: Shawn Bohrer <shawn.bohrer@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/line6/capture.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

--- a/drivers/staging/line6/capture.c
+++ b/drivers/staging/line6/capture.c
@@ -219,8 +219,8 @@ static void audio_in_callback(struct urb
 				       fbuf, fsize * bytes_per_frame);
 			}
 
-			if ((line6pcm->pos_in_done +=
-			     frames) >= runtime->buffer_size)
+			line6pcm->pos_in_done += frames;
+			if (line6pcm->pos_in_done >= runtime->buffer_size)
 				line6pcm->pos_in_done -= runtime->buffer_size;
 		}
 	}
@@ -235,7 +235,8 @@ static void audio_in_callback(struct urb
 	if (!shutdown) {
 		submit_audio_in_urb(substream);
 
-		if ((line6pcm->bytes_in += length) >= line6pcm->period_in) {
+		line6pcm->bytes_in += length;
+		if (line6pcm->bytes_in >= line6pcm->period_in) {
 			line6pcm->bytes_in -= line6pcm->period_in;
 			snd_pcm_period_elapsed(substream);
 		}
