From jmm@debian.org  Wed Feb 18 16:01:20 2009
From: Moritz Muehlenhoff <jmm@debian.org>
Date: Wed, 18 Feb 2009 19:50:09 +0100
Subject: Staging: wlan-ng: Remove yet more dead code from p80211netdev.c
To: greg@kroah.com
Cc: Moritz Muehlenhoff <jmm@debian.org>
Message-ID: <1234983009-5290-3-git-send-email-jmm@debian.org>


Signed-off-by: Moritz Muehlenhoff <jmm@debian.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/wlan-ng/p80211netdev.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/staging/wlan-ng/p80211netdev.c
+++ b/drivers/staging/wlan-ng/p80211netdev.c
@@ -70,10 +70,6 @@
 #include <linux/uaccess.h>
 #include <asm/byteorder.h>
 
-#ifdef SIOCETHTOOL
-#include <linux/ethtool.h>
-#endif
-
 #include <net/iw_handler.h>
 #include <net/net_namespace.h>
 
