From wfp5p@virginia.edu Mon Mar 16 17:35:37 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:20:52 -0400
Subject: Staging: comedi: Remove s526_board typedef
To: greg@kroah.com
Message-ID: <20090317022052.14467.96138.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/s526.c |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

--- a/drivers/staging/comedi/drivers/s526.c
+++ b/drivers/staging/comedi/drivers/s526.c
@@ -156,7 +156,7 @@ struct s526GPCTConfig {
  * boards in this way is optional, and completely driver-dependent.
  * Some drivers use arrays such as this, other do not.
  */
-typedef struct s526_board_struct {
+struct s526_board {
 	const char *name;
 	int gpct_chans;
 	int gpct_bits;
@@ -165,9 +165,9 @@ typedef struct s526_board_struct {
 	int da_chans;
 	int da_bits;
 	int have_dio;
-} s526_board;
+};
 
-static const s526_board s526_boards[] = {
+static const struct s526_board s526_boards[] = {
 	{
 	      name:	"s526",
 	      gpct_chans:4,
@@ -186,7 +186,7 @@ static const s526_board s526_boards[] = 
 /*
  * Useful for shorthand access to the particular board structure
  */
-#define thisboard ((const s526_board *)dev->board_ptr)
+#define thisboard ((const struct s526_board *)dev->board_ptr)
 
 /* this structure is for data unique to this hardware driver.  If
    several hardware drivers keep similar information in this structure,
@@ -243,8 +243,8 @@ static struct comedi_driver driver_s526 
 	 * devices are such boards.
 	 */
       board_name:&s526_boards[0].name,
-      offset:sizeof(s526_board),
-      num_names:sizeof(s526_boards) / sizeof(s526_board),
+      offset:sizeof(struct s526_board),
+      num_names:sizeof(s526_boards) / sizeof(struct s526_board),
 };
 
 static int s526_gpct_rinsn(struct comedi_device * dev, struct comedi_subdevice * s,
