From wfp5p@virginia.edu Mon Mar 16 17:29:27 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:14:41 -0400
Subject: Staging: comedi: Remove dt2801_private typedef
To: greg@kroah.com
Message-ID: <20090317021441.14467.58869.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/dt2801.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/dt2801.c
+++ b/drivers/staging/comedi/drivers/dt2801.c
@@ -220,11 +220,13 @@ static const struct dt2801_board boardty
 #define n_boardtypes ((sizeof(boardtypes))/(sizeof(boardtypes[0])))
 #define boardtype (*(const struct dt2801_board *)dev->board_ptr)
 
-typedef struct {
+struct dt2801_private {
+
 	const struct comedi_lrange *dac_range_types[2];
 	unsigned int ao_readback[2];
-} dt2801_private;
-#define devpriv ((dt2801_private *)dev->private)
+};
+
+#define devpriv ((struct dt2801_private *)dev->private)
 
 static int dt2801_ai_insn_read(struct comedi_device * dev, struct comedi_subdevice * s,
 	struct comedi_insn * insn, unsigned int * data);
@@ -521,7 +523,7 @@ static int dt2801_attach(struct comedi_d
 	if ((ret = alloc_subdevices(dev, 4)) < 0)
 		goto out;
 
-	if ((ret = alloc_private(dev, sizeof(dt2801_private))) < 0)
+	if ((ret = alloc_private(dev, sizeof(struct dt2801_private))) < 0)
 		goto out;
 
 	dev->board_name = boardtype.name;
