From wfp5p@virginia.edu Mon Mar 16 17:30:46 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:15:56 -0400
Subject: Staging: comedi: Remove NI_660xRegisterData typedef
To: greg@kroah.com
Message-ID: <20090317021556.14467.6924.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/ni_660x.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/drivers/staging/comedi/drivers/ni_660x.c
+++ b/drivers/staging/comedi/drivers/ni_660x.c
@@ -192,14 +192,16 @@ static inline unsigned NI_660X_GPCT_SUBD
 	return NI_660X_GPCT_SUBDEV_0 + index;
 }
 
-typedef struct {
+struct NI_660xRegisterData {
+
 	const char *name;	// Register Name
 	int offset;		// Offset from base address from GPCT chip
 	enum ni_660x_register_direction direction;
 	enum ni_660x_register_width size;	// 1 byte, 2 bytes, or 4 bytes
-} NI_660xRegisterData;
+};
+
 
-static const NI_660xRegisterData registerData[NumRegisters] = {
+static const struct NI_660xRegisterData registerData[NumRegisters] = {
 	{"G0 Interrupt Acknowledge", 0x004, NI_660x_WRITE, DATA_2B},
 	{"G0 Status Register", 0x004, NI_660x_READ, DATA_2B},
 	{"G1 Interrupt Acknowledge", 0x006, NI_660x_WRITE, DATA_2B},
