From wfp5p@virginia.edu Mon Mar 16 17:22:55 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:08:08 -0400
Subject: Staging: comedi: Remove adq12b_private typedef
To: greg@kroah.com
Message-ID: <20090317020808.14467.4031.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/adq12b.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/adq12b.c
+++ b/drivers/staging/comedi/drivers/adq12b.c
@@ -146,15 +146,15 @@ static const struct adq12b_board adq12b_
 
 #define thisboard ((const struct adq12b_board *)dev->board_ptr)
 
-typedef struct{
+struct adq12b_private {
         int unipolar;          /* option 2 of comedi_config (1 is iobase) */
         int differential;      /* option 3 of comedi_config */
         int last_channel;
         int last_range;
         unsigned int digital_state;
- }adq12b_private;
+};
 
-#define devpriv ((adq12b_private *)dev->private)
+#define devpriv ((struct adq12b_private *)dev->private)
 
 /*
  * The struct comedi_driver structure tells the Comedi core module
@@ -219,7 +219,7 @@ static int adq12b_attach(struct comedi_d
  * Allocate the private structure area.  alloc_private() is a
  * convenient macro defined in comedidev.h.
  */
-        if(alloc_private(dev, sizeof(adq12b_private)) < 0)
+        if(alloc_private(dev, sizeof(struct adq12b_private)) < 0)
           return -ENOMEM;
 
 /* fill in devpriv structure */
