From wfp5p@virginia.edu Mon Mar 16 17:35:26 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:20:41 -0400
Subject: Staging: comedi: Remove s526_gpct_config_t typedef
To: greg@kroah.com
Message-ID: <20090317022041.14467.36426.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/s526.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/comedi/drivers/s526.c
+++ b/drivers/staging/comedi/drivers/s526.c
@@ -146,10 +146,10 @@ enum S526_GPCT_APP_CLASS {
 /* Config struct for different GPCT subdevice Application Classes and
    their options
 */
-typedef struct s526GPCTConfig {
+struct s526GPCTConfig {
 	enum S526_GPCT_APP_CLASS app;
 	int data[MAX_GPCT_CONFIG_DATA];
-} s526_gpct_config_t;
+};
 
 /*
  * Board descriptions for two imaginary boards.  Describing the
@@ -201,7 +201,7 @@ struct s526_private {
 	/* Used for AO readback */
 	unsigned int ao_readback[2];
 
-	s526_gpct_config_t s526_gpct_config[4];
+	struct s526GPCTConfig s526_gpct_config[4];
 	unsigned short s526_ai_config;
 };
 
