From wfp5p@virginia.edu Thu Mar 19 13:05:52 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Thu, 19 Mar 2009 17:59:12 -0400
Subject: Staging: comedi: Remove ni_670x_board typedef
To: greg@kroah.com
Cc: abbotti@mev.co.uk, fmhess@users.sourceforge.net, ds@schleef.org
Message-ID: <20090319215912.10220.84055.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/ni_670x.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/ni_670x.c
+++ b/drivers/staging/comedi/drivers/ni_670x.c
@@ -60,13 +60,14 @@ Commands are not supported.
 
 /* Board description*/
 
-typedef struct ni_670x_board_struct {
+struct ni_670x_board {
 	unsigned short dev_id;
 	const char *name;
 	unsigned short ao_chans;
 	unsigned short ao_bits;
-} ni_670x_board;
-static const ni_670x_board ni_670x_boards[] = {
+};
+
+static const struct ni_670x_board ni_670x_boards[] = {
 	{
 	      dev_id:	0x2c90,
 	      name:	"PCI-6703",
@@ -96,7 +97,7 @@ static DEFINE_PCI_DEVICE_TABLE(ni_670x_p
 
 MODULE_DEVICE_TABLE(pci, ni_670x_pci_table);
 
-#define thisboard ((ni_670x_board *)dev->board_ptr)
+#define thisboard ((struct ni_670x_board *)dev->board_ptr)
 
 struct ni_670x_private {
 
