From wfp5p@virginia.edu Mon Mar 16 17:33:18 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:18:32 -0400
Subject: Staging: comedi: Remove pcl726_private typedef
To: greg@kroah.com
Message-ID: <20090317021832.14467.37097.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/pcl726.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/pcl726.c
+++ b/drivers/staging/comedi/drivers/pcl726.c
@@ -161,12 +161,14 @@ static struct comedi_driver driver_pcl72
 
 COMEDI_INITCLEANUP(driver_pcl726);
 
-typedef struct {
+struct pcl726_private {
+
 	int bipolar[12];
 	const struct comedi_lrange *rangelist[12];
 	unsigned int ao_readback[12];
-} pcl726_private;
-#define devpriv ((pcl726_private *)dev->private)
+};
+
+#define devpriv ((struct pcl726_private *)dev->private)
 
 static int pcl726_ao_insn(struct comedi_device * dev, struct comedi_subdevice * s,
 	struct comedi_insn * insn, unsigned int * data)
@@ -260,7 +262,7 @@ static int pcl726_attach(struct comedi_d
 
 	dev->board_name = this_board->name;
 
-	if ((ret = alloc_private(dev, sizeof(pcl726_private))) < 0)
+	if ((ret = alloc_private(dev, sizeof(struct pcl726_private))) < 0)
 		return -ENOMEM;
 
 	for (i = 0; i < 12; i++) {
