From arve@android.com  Tue Apr  7 11:14:55 2009
From: Arve Hjønnevåg <arve@android.com>
Date: Mon,  6 Apr 2009 15:12:54 -0700
Subject: Staging: binder: Don't create two proc entries with the same name if the driver is opened twice in one process.
To: gregkh@suse.de
Cc: swetland@google.com, viro@zeniv.linux.org.uk, torvalds@linux-foundation.org, hugh@veritas.com, Arve Hjønnevåg <arve@android.com>
Message-ID: <1239055980-24607-2-git-send-email-arve@android.com>


Signed-off-by: Arve Hjønnevåg <arve@android.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/android/binder.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/staging/android/binder.c
+++ b/drivers/staging/android/binder.c
@@ -2776,6 +2776,7 @@ static int binder_open(struct inode *nod
 	if (binder_proc_dir_entry_proc) {
 		char strbuf[11];
 		snprintf(strbuf, sizeof(strbuf), "%u", proc->pid);
+		remove_proc_entry(strbuf, binder_proc_dir_entry_proc);
 		create_proc_read_entry(strbuf, S_IRUGO, binder_proc_dir_entry_proc, binder_read_proc_proc, proc);
 	}
 
