From 9374f744d6d36acd1ffded58a1fe68aeddcbe5fd Mon Sep 17 00:00:00 2001
From: Frank Mori Hess <fmhess@users.sourceforge.net>
Date: Mon, 6 Apr 2009 14:01:19 -0400
Subject: Staging: comedi: Fixed Kconfig option for COMEDI_PCMCIA_DRIVERS

Signed-off-by: Frank Mori Hess <fmhess@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/comedi/Kconfig
+++ b/drivers/staging/comedi/Kconfig
@@ -29,7 +29,7 @@ config COMEDI_PCI_DRIVERS
 
 config COMEDI_PCMCIA_DRIVERS
 	tristate "Comedi PCMCIA drivers"
-	depends on COMEDI && PCMCIAI
+	depends on COMEDI && PCMCIA
 	default N
 	---help---
 	  Enable lots of comedi PCMCIA drivers to be built
