From jmm@debian.org  Wed Feb  4 15:18:53 2009
From: Moritz Muehlenhoff <jmm@debian.org>
Date: Sun,  1 Feb 2009 13:29:06 +0100
Subject: Staging: wlan-ng: Remove dead/unused code from p80211ioctl.h
To: greg@kroah.com
Cc: Moritz Muehlenhoff <jmm@debian.org>
Message-ID: <1233491348-14411-9-git-send-email-jmm@debian.org>


Signed-off-by: Moritz Muehlenhoff <jmm@debian.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/wlan-ng/p80211ioctl.h |   21 ---------------------
 1 file changed, 21 deletions(-)

--- a/drivers/staging/wlan-ng/p80211ioctl.h
+++ b/drivers/staging/wlan-ng/p80211ioctl.h
@@ -78,18 +78,6 @@
 
 #define P80211_IOCTL_MAGIC	(0x4a2d464dUL)
 
-/*----------------------------------------------------------------*/
-/* Netlink protocol numbers for the indication interface */
-
-#define P80211_NL_SOCK_IND	NETLINK_USERSOCK
-
-/*----------------------------------------------------------------*/
-/* Netlink multicast bits for different types of messages */
-
-#define P80211_NL_MCAST_GRP_MLME	BIT(0)	/* Local station messages */
-#define P80211_NL_MCAST_GRP_SNIFF	BIT(1)	/* Sniffer messages */
-#define P80211_NL_MCAST_GRP_DIST	BIT(2)	/* Distribution system messages */
-
 /*================================================================*/
 /* Types */
 
@@ -107,13 +95,4 @@ typedef struct p80211ioctl_req
 	u32	result;
 } __attribute__((packed)) p80211ioctl_req_t;
 
-
-/*================================================================*/
-/* Extern Declarations */
-
-
-/*================================================================*/
-/* Function Declarations */
-
-
 #endif /* _P80211IOCTL_H */
