From jmm@debian.org  Wed Feb  4 15:15:32 2009
From: Moritz Muehlenhoff <jmm@debian.org>
Date: Sun,  1 Feb 2009 13:28:58 +0100
Subject: Staging: wlan-ng: Remove dead/unused code from p80211req.c
To: greg@kroah.com
Cc: Moritz Muehlenhoff <jmm@debian.org>
Message-ID: <1233491348-14411-1-git-send-email-jmm@debian.org>


Signed-off-by: Moritz Muehlenhoff <jmm@debian.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/wlan-ng/p80211req.c |   18 ------------------
 1 file changed, 18 deletions(-)

--- a/drivers/staging/wlan-ng/p80211req.c
+++ b/drivers/staging/wlan-ng/p80211req.c
@@ -53,8 +53,6 @@
 /*================================================================*/
 /* System Includes */
 
-
-
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/sched.h>
@@ -84,21 +82,6 @@
 #include "p80211req.h"
 
 /*================================================================*/
-/* Local Constants */
-
-/* Maximum amount of time we'll wait for a request to complete */
-#define P80211REQ_MAXTIME	3*HZ	/* 3 seconds */
-
-/*================================================================*/
-/* Local Macros */
-
-/*================================================================*/
-/* Local Types */
-
-/*================================================================*/
-/* Local Static Definitions */
-
-/*================================================================*/
 /* Local Function Declarations */
 
 static void p80211req_handlemsg( wlandevice_t *wlandev, p80211msg_t *msg);
@@ -287,4 +270,3 @@ static int p80211req_mibset_mibget(wland
 
 	return 0;
 }
-
