From foo@baz Mon Aug 17 17:21:51 PDT 2009
Date: Mon, 17 Aug 2009 17:21:51 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: cleanup coding style issues in VersionInfo.h


Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/VersionInfo.h |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/hv/VersionInfo.h
+++ b/drivers/staging/hv/VersionInfo.h
@@ -21,9 +21,11 @@
  *
  */
 
+#ifndef __HV_VERSION_INFO
+#define __HV_VERSION_INFO
 
-#pragma once
+static const char VersionDate[] = __DATE__;
+static const char VersionTime[] = __TIME__;
+static const char VersionDesc[] = "Version 2.0";
 
-static const char VersionDate[]=__DATE__;
-static const char VersionTime[]=__TIME__;
-static const char VersionDesc[]= "Version 2.0";
+#endif
