From dbaryshkov@gmail.com  Thu Dec  4 11:36:26 2008
From: Dmitry Baryshkov <dbaryshkov@gmail.com>
Date: Fri, 28 Nov 2008 00:01:28 +0300
Subject: STAGING: Move staging drivers back to staging-specific menu
Cc: Greg Kroah-Hartman <gregkh@suse.de>, Dmitry Baryshkov <dbaryshkov@gmail.com>
Message-ID: <1227819688-24035-1-git-send-email-dbaryshkov@gmail.com>


Signed-off-by: Dmitry Baryshkov <dbaryshkov@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/Kconfig |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/staging/Kconfig
+++ b/drivers/staging/Kconfig
@@ -22,6 +22,8 @@ menuconfig STAGING
 	  If in doubt, say N here.
 
 
+if STAGING
+
 config STAGING_EXCLUDE_BUILD
 	bool "Exclude Staging drivers from being built" if STAGING
 	default y
@@ -62,3 +64,4 @@ source "drivers/staging/at76_usb/Kconfig
 source "drivers/staging/poch/Kconfig"
 
 endif # !STAGING_EXCLUDE_BUILD
+endif # STAGING
