From wfp5p@virginia.edu  Wed Mar 18 21:43:29 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:17:11 -0400
Subject: Staging: comedi: pcl816: Remove boardtype typedef
To: greg@kroah.com
Message-ID: <20090317021711.14467.89427.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/pcl816.c |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

--- a/drivers/staging/comedi/drivers/pcl816.c
+++ b/drivers/staging/comedi/drivers/pcl816.c
@@ -101,7 +101,8 @@ static const struct comedi_lrange range_
 			UNI_RANGE(1.25),
 	}
 };
-typedef struct {
+struct pcl816_board {
+
 	const char *name;	// board name
 	int n_ranges;		// len of range list
 	int n_aichan;		// num of A/D chans in diferencial mode
@@ -119,9 +120,10 @@ typedef struct {
 	int ai_chanlist;	// allowed len of channel list A/D
 	int ao_chanlist;	// allowed len of channel list D/A
 	int i8254_osc_base;	// 1/frequency of on board oscilator in ns
-} boardtype;
+};
+
 
-static const boardtype boardtypes[] = {
+static const struct pcl816_board boardtypes[] = {
 	{"pcl816", 8, 16, 10000, 1, 16, 16, &range_pcl816,
 			&range_pcl816, PCLx1x_RANGE,
 			0x00fc,	// IRQ mask
@@ -142,9 +144,9 @@ static const boardtype boardtypes[] = {
 		100},
 };
 
-#define n_boardtypes (sizeof(boardtypes)/sizeof(boardtype))
+#define n_boardtypes (sizeof(boardtypes)/sizeof(struct pcl816_board))
 #define devpriv ((struct pcl816_private *)dev->private)
-#define this_board ((const boardtype *)dev->board_ptr)
+#define this_board ((const struct pcl816_board *)dev->board_ptr)
 
 static int pcl816_attach(struct comedi_device * dev, struct comedi_devconfig * it);
 static int pcl816_detach(struct comedi_device * dev);
@@ -161,7 +163,7 @@ static struct comedi_driver driver_pcl81
       detach:pcl816_detach,
       board_name:&boardtypes[0].name,
       num_names:n_boardtypes,
-      offset:sizeof(boardtype),
+      offset:sizeof(struct pcl816_board),
 };
 
 COMEDI_INITCLEANUP(driver_pcl816);
