From roel.kluin@gmail.com  Wed Feb  4 15:03:42 2009
From: Roel Kluin <roel.kluin@gmail.com>
Date: Sat, 31 Jan 2009 11:44:02 +0100
Subject: Staging: agnx: i reaches -1, tested 0
To: Greg KH <gregkh@suse.de>
Message-ID: <49842B72.8070405@gmail.com>

From: Roel Kluin <roel.kluin@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/agnx/rf.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/agnx/rf.c
+++ b/drivers/staging/agnx/rf.c
@@ -669,7 +669,7 @@ static inline void calibra_delay(struct 
 	unsigned int i = 100;
 
 	wmb();
-	while (i--) {
+	while (--i) {
 		reg = (ioread32(ctl + AGNX_ACI_STATUS));
 		if (reg == 0x4000)
 			break;
