From wfp5p@virginia.edu  Wed Mar 18 21:46:53 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:17:33 -0400
Subject: Staging: comedi: rti800: Remove boardtype typedef
To: greg@kroah.com
Message-ID: <20090317021733.14467.22636.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/rti800.c |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

--- a/drivers/staging/comedi/drivers/rti800.c
+++ b/drivers/staging/comedi/drivers/rti800.c
@@ -118,16 +118,18 @@ static const struct comedi_lrange range_
 	}
 };
 
-typedef struct {
+struct rti800_board {
+
 	const char *name;
 	int has_ao;
-} boardtype;
-static const boardtype boardtypes[] = {
+};
+
+static const struct rti800_board boardtypes[] = {
 	{"rti800", 0},
 	{"rti815", 1},
 };
 
-#define this_board ((const boardtype *)dev->board_ptr)
+#define this_board ((const struct rti800_board *)dev->board_ptr)
 
 static int rti800_attach(struct comedi_device * dev, struct comedi_devconfig * it);
 static int rti800_detach(struct comedi_device * dev);
@@ -136,9 +138,9 @@ static struct comedi_driver driver_rti80
       module:THIS_MODULE,
       attach:rti800_attach,
       detach:rti800_detach,
-      num_names:sizeof(boardtypes) / sizeof(boardtype),
+      num_names:sizeof(boardtypes) / sizeof(struct rti800_board),
       board_name:&boardtypes[0].name,
-      offset:sizeof(boardtype),
+      offset:sizeof(struct rti800_board),
 };
 
 COMEDI_INITCLEANUP(driver_rti800);
