From wfp5p@virginia.edu Thu Mar 19 13:06:31 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Thu, 19 Mar 2009 17:59:49 -0400
Subject: Staging: comedi: Remove das08_board typedef
To: greg@kroah.com
Cc: abbotti@mev.co.uk, fmhess@users.sourceforge.net, ds@schleef.org
Message-ID: <20090319215949.10220.13898.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/das08.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/comedi/drivers/das08.h
+++ b/drivers/staging/comedi/drivers/das08.h
@@ -30,7 +30,7 @@ enum das08_ai_encoding { das08_encode12,
 enum das08_lrange { das08_pg_none, das08_bipolar5, das08_pgh, das08_pgl,
 		das08_pgm };
 
-typedef struct das08_board_struct {
+struct das08_board_struct {
 	const char *name;
 	unsigned int id;	// id for pci/pcmcia boards
 	enum das08_bustype bustype;
@@ -46,7 +46,7 @@ typedef struct das08_board_struct {
 	unsigned int i8255_offset;
 	unsigned int i8254_offset;
 	unsigned int iosize;	// number of ioports used
-} das08_board;
+};
 
 struct i8254_struct {
 	int channels;		// available channels. Some could be used internally.
