From wfp5p@virginia.edu Mon Mar 16 17:29:53 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:15:02 -0400
Subject: Staging: comedi: Remove dt3k_boardtype typedef
To: greg@kroah.com
Message-ID: <20090317021502.14467.67802.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/dt3000.c |   12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

--- a/drivers/staging/comedi/drivers/dt3000.c
+++ b/drivers/staging/comedi/drivers/dt3000.c
@@ -81,7 +81,8 @@ static const struct comedi_lrange range_
 	}
 };
 
-typedef struct {
+struct dt3k_boardtype {
+
 	const char *name;
 	unsigned int device_id;
 	int adchan;
@@ -90,9 +91,10 @@ typedef struct {
 	const struct comedi_lrange *adrange;
 	int dachan;
 	int dabits;
-} dt3k_boardtype;
+};
+
 
-static const dt3k_boardtype dt3k_boardtypes[] = {
+static const struct dt3k_boardtype dt3k_boardtypes[] = {
       {name:"dt3001",
 	      device_id:0x22,
 	      adchan:	16,
@@ -158,8 +160,8 @@ static const dt3k_boardtype dt3k_boardty
 		},
 };
 
-#define n_dt3k_boards sizeof(dt3k_boardtypes)/sizeof(dt3k_boardtype)
-#define this_board ((const dt3k_boardtype *)dev->board_ptr)
+#define n_dt3k_boards sizeof(dt3k_boardtypes)/sizeof(struct dt3k_boardtype)
+#define this_board ((const struct dt3k_boardtype *)dev->board_ptr)
 
 static DEFINE_PCI_DEVICE_TABLE(dt3k_pci_table) = {
 	{PCI_VENDOR_ID_DT, 0x0022, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
