From wfp5p@virginia.edu Mon Mar 16 17:34:07 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:19:20 -0400
Subject: Staging: comedi: Remove rti802_private typedef
To: greg@kroah.com
Message-ID: <20090317021920.14467.29679.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/rti802.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/rti802.c
+++ b/drivers/staging/comedi/drivers/rti802.c
@@ -58,15 +58,15 @@ static struct comedi_driver driver_rti80
 
 COMEDI_INITCLEANUP(driver_rti802);
 
-typedef struct {
+struct rti802_private {
 	enum {
 		dac_2comp, dac_straight
 	} dac_coding[8];
 	const struct comedi_lrange *range_type_list[8];
 	unsigned int ao_readback[8];
-} rti802_private;
+};
 
-#define devpriv ((rti802_private *)dev->private)
+#define devpriv ((struct rti802_private *)dev->private)
 
 static int rti802_ao_insn_read(struct comedi_device * dev, struct comedi_subdevice * s,
 	struct comedi_insn * insn, unsigned int * data)
@@ -113,7 +113,7 @@ static int rti802_attach(struct comedi_d
 	dev->board_name = "rti802";
 
 	if (alloc_subdevices(dev, 1) < 0
-		|| alloc_private(dev, sizeof(rti802_private))) {
+		|| alloc_private(dev, sizeof(struct rti802_private))) {
 		return -ENOMEM;
 	}
 
