From shane.warden@onyxneon.com  Fri Oct  2 14:56:36 2009
From: Shane Warden <shane.warden@onyxneon.com>
Date: Tue, 22 Sep 2009 16:13:03 -0700
Subject: Staging: comedi: comedi_compat32.h: Fixed checkpatch.pl issues
To: greg@kroah.com
Cc: Shane Warden <shane.warden@onyxneon.com>
Message-ID: <1253661184-8318-2-git-send-email-shane.warden@onyxneon.com>



Signed-off-by: Shane Warden <shane.warden@onyxneon.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/comedi/comedi_compat32.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/staging/comedi/comedi_compat32.h
+++ b/drivers/staging/comedi/comedi_compat32.h
@@ -28,7 +28,9 @@
 #define _COMEDI_COMPAT32_H
 
 #include <linux/compat.h>
-#include <linux/fs.h>		/* For HAVE_COMPAT_IOCTL and HAVE_UNLOCKED_IOCTL */
+
+/* For HAVE_COMPAT_IOCTL and HAVE_UNLOCKED_IOCTL */
+#include <linux/fs.h>
 
 #ifdef CONFIG_COMPAT
 
