From wfp5p@virginia.edu  Mon Jul 27 14:11:58 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 27 Jul 2009 16:47:41 -0400
Subject: Staging: hv: remove VMBUS_CONNECT_STATE typedef
To: greg@kroah.com
Cc: hjanssen@microsoft.com
Message-ID: <1248727665-23969-20-git-send-email-wfp5p@virginia.edu>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/VmbusPrivate.h |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/hv/VmbusPrivate.h
+++ b/drivers/staging/hv/VmbusPrivate.h
@@ -54,18 +54,18 @@
 /* Data types */
 
 
-typedef enum {
+enum VMBUS_CONNECT_STATE {
 	Disconnected,
 	Connecting,
 	Connected,
 	Disconnecting
-} VMBUS_CONNECT_STATE;
+};
 
 #define MAX_SIZE_CHANNEL_MESSAGE			HV_MESSAGE_PAYLOAD_BYTE_COUNT
 
 typedef struct _VMBUS_CONNECTION {
 
-	VMBUS_CONNECT_STATE					ConnectState;
+	enum VMBUS_CONNECT_STATE					ConnectState;
 
 	u32								NextGpadlHandle;
 
