From wfp5p@virginia.edu Mon Mar 16 17:29:53 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:14:57 -0400
Subject: Staging: comedi: Remove dt2815_private typedef
To: greg@kroah.com
Message-ID: <20090317021457.14467.48634.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/dt2815.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/dt2815.c
+++ b/drivers/staging/comedi/drivers/dt2815.c
@@ -88,12 +88,14 @@ COMEDI_INITCLEANUP(driver_dt2815);
 
 static void dt2815_free_resources(struct comedi_device * dev);
 
-typedef struct {
+struct dt2815_private {
+
 	const struct comedi_lrange *range_type_list[8];
 	unsigned int ao_readback[8];
-} dt2815_private;
+};
+
 
-#define devpriv ((dt2815_private *)dev->private)
+#define devpriv ((struct dt2815_private *)dev->private)
 
 static int dt2815_wait_for_status(struct comedi_device * dev, int status)
 {
@@ -196,7 +198,7 @@ static int dt2815_attach(struct comedi_d
 
 	if (alloc_subdevices(dev, 1) < 0)
 		return -ENOMEM;
-	if (alloc_private(dev, sizeof(dt2815_private)) < 0)
+	if (alloc_private(dev, sizeof(struct dt2815_private)) < 0)
 		return -ENOMEM;
 
 	s = dev->subdevices;
