From a.beregalov@gmail.com  Tue Feb 17 17:39:05 2009
From: Alexander Beregalov <a.beregalov@gmail.com>
Date: Fri, 13 Feb 2009 18:06:54 +0300
Subject: Staging: pohmelfs should depend on CRYPTO
To: zbr@ioremap.net, gregkh@suse.de
Message-ID: <20090213150654.GA7390@orion>



Signed-off-by: Alexander Beregalov <a.beregalov@gmail.com>
Cc: Evgeniy Polyakov <zbr@ioremap.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/pohmelfs/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/staging/pohmelfs/Kconfig
+++ b/drivers/staging/pohmelfs/Kconfig
@@ -1,6 +1,7 @@
 config POHMELFS
 	tristate "POHMELFS filesystem support"
 	select CONNECTOR
+	select CRYPTO
 	select CRYPTO_BLKCIPHER
 	select CRYPTO_HMAC
 	help
