From wfp5p@virginia.edu Mon Mar 16 17:30:26 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:15:34 -0400
Subject: Staging: comedi: Remove skel_private typedef
To: greg@kroah.com
Message-ID: <20090317021534.14467.81216.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/mpc624.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/mpc624.c
+++ b/drivers/staging/comedi/drivers/mpc624.c
@@ -120,11 +120,13 @@ Configuration Options:
 #define MPC624_SPEED_13_75_Hz   (MPC624_OSR4 | MPC624_OSR3                             | MPC624_OSR0)
 #define MPC624_SPEED_6_875_Hz   (MPC624_OSR4 | MPC624_OSR3 | MPC624_OSR2 | MPC624_OSR1 | MPC624_OSR0)
 //----------------------------------------------------------------------------
-typedef struct {
+struct skel_private {
+
 	unsigned long int ulConvertionRate;	// set by mpc624_attach() from driver's parameters
-} skel_private;
+};
+
 
-#define devpriv ((skel_private *)dev->private)
+#define devpriv ((struct skel_private *)dev->private)
 //----------------------------------------------------------------------------
 static const struct comedi_lrange range_mpc624_bipolar1 = {
 	1,
@@ -174,7 +176,7 @@ static int mpc624_attach(struct comedi_d
 	dev->board_name = "mpc624";
 
 	// Private structure initialization
-	if (alloc_private(dev, sizeof(skel_private)) < 0)
+	if (alloc_private(dev, sizeof(struct skel_private)) < 0)
 		return -ENOMEM;
 
 	switch (it->options[1]) {
