From wfp5p@virginia.edu  Wed Mar 18 21:42:23 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:16:55 -0400
Subject: Staging: comedi: pcl726: Remove boardtype typedef
To: greg@kroah.com
Message-ID: <20090317021655.14467.65542.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/pcl726.c |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

--- a/drivers/staging/comedi/drivers/pcl726.c
+++ b/drivers/staging/comedi/drivers/pcl726.c
@@ -114,7 +114,8 @@ static const struct comedi_lrange *const
 static int pcl726_attach(struct comedi_device * dev, struct comedi_devconfig * it);
 static int pcl726_detach(struct comedi_device * dev);
 
-typedef struct {
+struct pcl726_board {
+
 	const char *name;	// driver name
 	int n_aochan;		// num of D/A chans
 	int num_of_ranges;	// num of ranges
@@ -126,9 +127,10 @@ typedef struct {
 	int do_hi;
 	int do_lo;
 	const struct comedi_lrange *const *range_type_list;	// list of supported ranges
-} boardtype;
+};
+
 
-static const boardtype boardtypes[] = {
+static const struct pcl726_board boardtypes[] = {
 	{"pcl726", 6, 6, 0x0000, PCL726_SIZE, 1,
 			PCL726_DI_HI, PCL726_DI_LO, PCL726_DO_HI, PCL726_DO_LO,
 		&rangelist_726[0],},
@@ -146,8 +148,8 @@ static const boardtype boardtypes[] = {
 		&rangelist_728[0],},
 };
 
-#define n_boardtypes (sizeof(boardtypes)/sizeof(boardtype))
-#define this_board ((const boardtype *)dev->board_ptr)
+#define n_boardtypes (sizeof(boardtypes)/sizeof(struct pcl726_board))
+#define this_board ((const struct pcl726_board *)dev->board_ptr)
 
 static struct comedi_driver driver_pcl726 = {
       driver_name:"pcl726",
@@ -156,7 +158,7 @@ static struct comedi_driver driver_pcl72
       detach:pcl726_detach,
       board_name:&boardtypes[0].name,
       num_names:n_boardtypes,
-      offset:sizeof(boardtype),
+      offset:sizeof(struct pcl726_board),
 };
 
 COMEDI_INITCLEANUP(driver_pcl726);
