From m.kozlowski@tuxland.pl  Wed Mar 11 14:34:46 2009
From: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Date: Tue,  3 Mar 2009 19:38:33 +0100
Subject: Staging: comedi: usbduxfast: balance semaphores up/down in attach error path
To: gregkh@suse.de
Cc: m.kozlowski@tuxland.pl
Message-ID: <1236105518-19565-4-git-send-email-m.kozlowski@tuxland.pl>



Signed-off-by: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Cc: Bernd Porr <BerndPorr@f2s.com>
Cc: Ian Abbott <abbotti@mev.co.uk>
Cc: Frank Mori Hess <fmhess@users.sourceforge.net>
Cc: David Schleef <ds@schleef.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/staging/comedi/drivers/usbduxfast.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/staging/comedi/drivers/usbduxfast.c
+++ b/drivers/staging/comedi/drivers/usbduxfast.c
@@ -1766,6 +1766,7 @@ static int usbduxfast_attach(comedi_devi
 	if (ret < 0) {
 		printk(KERN_ERR "comedi%d: usbduxfast: error alloc space for "
 		       "subdev\n", dev->minor);
+		up(&(usbduxfastsub[index].sem));
 		up(&start_stop_sem);
 		return ret;
 	}
