From m.kozlowski@tuxland.pl  Wed Mar 11 14:36:03 2009
From: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Date: Tue,  3 Mar 2009 19:38:36 +0100
Subject: Staging: comedi: usbdux: annotate __init and __exit functions
To: gregkh@suse.de
Cc: m.kozlowski@tuxland.pl
Message-ID: <1236105518-19565-7-git-send-email-m.kozlowski@tuxland.pl>



Signed-off-by: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Cc: Bernd Porr <BerndPorr@f2s.com>
Cc: Ian Abbott <abbotti@mev.co.uk>
Cc: Frank Mori Hess <fmhess@users.sourceforge.net>
Cc: David Schleef <ds@schleef.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/usbdux.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/comedi/drivers/usbdux.c
+++ b/drivers/staging/comedi/drivers/usbdux.c
@@ -2930,7 +2930,7 @@ static struct usb_driver usbduxsub_drive
 /* Can't use the nice macro as I have also to initialise the USB */
 /* subsystem: */
 /* registering the usb-system _and_ the comedi-driver */
-static int init_usbdux(void)
+static int __init init_usbdux(void)
 {
 	printk(KERN_INFO KBUILD_MODNAME ": "
 	       DRIVER_VERSION ":" DRIVER_DESC "\n");
@@ -2940,7 +2940,7 @@ static int init_usbdux(void)
 }
 
 /* deregistering the comedi driver and the usb-subsystem */
-static void exit_usbdux(void)
+static void __exit exit_usbdux(void)
 {
 	comedi_driver_unregister(&driver_usbdux);
 	usb_deregister(&usbduxsub_driver);
