From penberg@cs.helsinki.fi  Mon Aug 17 12:59:08 2009
From: Pekka Enberg <penberg@cs.helsinki.fi>
Date: Wed, 12 Aug 2009 11:03:37 +0300
Subject: Staging: w35und: Remove unused struct _STATISTICS_INFO
To: gregkh@suse.de
Cc: linux-kernel@vger.kernel.org, Pekka Enberg <penberg@cs.helsinki.fi>, Pavel Machek <pavel@ucw.cz>
Message-ID: <1250064219-7372-5-git-send-email-penberg@cs.helsinki.fi>


Acked-by: Pavel Machek <pavel@ucw.cz>
Signed-off-by: Pekka Enberg <penberg@cs.helsinki.fi>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/winbond/mto.h |   49 ------------------------------------------
 1 file changed, 49 deletions(-)

--- a/drivers/staging/winbond/mto.h
+++ b/drivers/staging/winbond/mto.h
@@ -209,55 +209,6 @@ typedef struct {
 	u8 tx_retry_rate;
 } TXRETRY_REC;
 
-typedef struct _STATISTICS_INFO {
-	u32   Rate54M;
-	u32   Rate48M;
-	u32   Rate36M;
-	u32   Rate24M;
-	u32   Rate18M;
-	u32   Rate12M;
-	u32   Rate9M;
-	u32   Rate6M;
-	u32   Rate11MS;
-	u32   Rate11ML;
-	u32   Rate55MS;
-	u32   Rate55ML;
-	u32   Rate2MS;
-	u32   Rate2ML;
-	u32   Rate1M;
-	u32   Rate54MOK;
-	u32   Rate48MOK;
-	u32   Rate36MOK;
-	u32   Rate24MOK;
-	u32   Rate18MOK;
-	u32   Rate12MOK;
-	u32   Rate9MOK;
-	u32   Rate6MOK;
-	u32   Rate11MSOK;
-	u32   Rate11MLOK;
-	u32   Rate55MSOK;
-	u32   Rate55MLOK;
-	u32   Rate2MSOK;
-	u32   Rate2MLOK;
-	u32   Rate1MOK;
-	u32   SQ3;
-	s32   RSSIAVG;
-	s32   RSSIMAX;
-	s32   TXRATE;
-	s32   TxRetryRate;
-	s32   BSS_PK_CNT;
-	s32   NIDLESLOT;
-	s32   SLOT_CNT;
-	s32   INTERF_CNT;
-	s32   GAP_CNT;
-	s32   DS_EVM;
-	s32   RcvBeaconNum;
-	s32   RXRATE;
-	s32   RxBytes;
-	s32   TxBytes;
-	s32   Antenna;
-} STATISTICS_INFO, *PSTATISTICS_INFO;
-
 extern void MTO_Init(struct wbsoft_priv *);
 extern void MTO_PeriodicTimerExpired(struct wbsoft_priv *);
 extern void MTO_SetDTORateRange(struct wbsoft_priv *, u8 *, u8);
