From wfp5p@virginia.edu Mon Mar 16 17:28:24 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:13:36 -0400
Subject: Staging: comedi: Remove das16m1_board typedef
To: greg@kroah.com
Message-ID: <20090317021336.14467.53255.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/das16m1.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/das16m1.c
+++ b/drivers/staging/comedi/drivers/das16m1.c
@@ -152,12 +152,12 @@ static unsigned int das16m1_set_pacer(st
 
 static int das16m1_irq_bits(unsigned int irq);
 
-typedef struct das16m1_board_struct {
+struct das16m1_board {
 	const char *name;
 	unsigned int ai_speed;
-} das16m1_board;
+};
 
-static const das16m1_board das16m1_boards[] = {
+static const struct das16m1_board das16m1_boards[] = {
 	{
 	      name:	"cio-das16/m1",	// CIO-DAS16_M1.pdf
 	      ai_speed:1000,	// 1MHz max speed
@@ -191,7 +191,7 @@ struct das16m1_private_struct {
 	unsigned int divisor2;	// divides master clock to obtain conversion speed
 };
 #define devpriv ((struct das16m1_private_struct *)(dev->private))
-#define thisboard ((const das16m1_board *)(dev->board_ptr))
+#define thisboard ((const struct das16m1_board *)(dev->board_ptr))
 
 COMEDI_INITCLEANUP(driver_das16m1);
 
