From wfp5p@virginia.edu  Mon Jul 27 14:09:57 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 27 Jul 2009 16:47:35 -0400
Subject: Staging: hv: check return value of bus_register()
To: greg@kroah.com
Cc: hjanssen@microsoft.com
Message-ID: <1248727665-23969-14-git-send-email-wfp5p@virginia.edu>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/vmbus_drv.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/drivers/staging/hv/vmbus_drv.c
+++ b/drivers/staging/hv/vmbus_drv.c
@@ -380,7 +380,12 @@ int vmbus_bus_init(PFN_DRIVERINITIALIZE 
 	tasklet_init(&vmbus_drv_ctx->event_dpc, vmbus_event_dpc, (unsigned long)vmbus_drv_obj);
 
 	/* Now, register the bus driver with LDM */
-	bus_register(&vmbus_drv_ctx->bus);
+	ret = bus_register(&vmbus_drv_ctx->bus);
+	if (ret)
+	{
+		ret = -1;
+		goto cleanup;
+	}
 
 	/* Get the interrupt resource */
 	ret = request_irq(vmbus_irq,
