From wfp5p@virginia.edu Mon Mar 16 17:30:37 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:15:45 -0400
Subject: Staging: comedi: Remove ni6527_board typedef
To: greg@kroah.com
Message-ID: <20090317021545.14467.88634.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/ni_6527.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/ni_6527.c
+++ b/drivers/staging/comedi/drivers/ni_6527.c
@@ -84,11 +84,13 @@ static struct comedi_driver driver_ni652
       detach:ni6527_detach,
 };
 
-typedef struct {
+struct ni6527_board {
+
 	int dev_id;
 	const char *name;
-} ni6527_board;
-static const ni6527_board ni6527_boards[] = {
+};
+
+static const struct ni6527_board ni6527_boards[] = {
 	{
 	      dev_id:	0x2b20,
 	      name:	"pci-6527",
@@ -100,7 +102,7 @@ static const ni6527_board ni6527_boards[
 };
 
 #define n_ni6527_boards (sizeof(ni6527_boards)/sizeof(ni6527_boards[0]))
-#define this_board ((const ni6527_board *)dev->board_ptr)
+#define this_board ((const struct ni6527_board *)dev->board_ptr)
 
 static DEFINE_PCI_DEVICE_TABLE(ni6527_pci_table) = {
 	{PCI_VENDOR_ID_NATINST, 0x2b10, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
