From himanshu@symmetricore.com  Wed Jan 27 18:25:46 2010
From: Himanshu Chauhan <himanshu@symmetricore.com>
Date: Sat, 23 Jan 2010 01:51:57 +0530
Subject: Staging: usbip: minor code cleanup
To: greg@kroah.com
Cc: Himanshu Chauhan <Himanshu@symmetricore.com>, Himanshu Chauhan <himanshu@symmetricore.com>
Message-ID: <1264191718-12481-2-git-send-email-himanshu@symmetricore.com>


From: Himanshu Chauhan <Himanshu@symmetricore.com>

Signed-off-by: Himanshu Chauhan <himanshu@symmetricore.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/usbip/usbip_common.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/drivers/staging/usbip/usbip_common.c
+++ b/drivers/staging/usbip/usbip_common.c
@@ -55,10 +55,7 @@ static ssize_t show_flag(struct device *
 static ssize_t store_flag(struct device *dev, struct device_attribute *attr,
 		const char *buf, size_t count)
 {
-	unsigned long flag;
-
-	sscanf(buf, "%lx", &flag);
-	usbip_debug_flag = flag;
+	sscanf(buf, "%lx", &usbip_debug_flag);
 
 	return count;
 }
