From bzolnier@gmail.com  Thu Jul  2 12:32:52 2009
From: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Date: Sun, 28 Jun 2009 17:40:50 +0200
Subject: Staging: rtl8192su: add TODO
To: "Greg Kroah-Hartman" <gregkh@suse.de>
Cc: linux-kernel@vger.kernel.org
Message-ID: <200906281740.50426.bzolnier@gmail.com>
Content-Disposition: inline



Signed-off-by: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/rtl8192su/TODO |   23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

--- /dev/null
+++ b/drivers/staging/rtl8192su/TODO
@@ -0,0 +1,23 @@
+TODO:
+- make ieee80211 stack private (may conflict with libipw currently)
+- prepare private ieee80211 stack for merge with rtl8187se's version:
+  - remove JOHN_DUMP[_DESC] ifdefs
+  - remove [IN]_OPENSUSE_SLED definitions
+  - remove superflous container_of definition from ieee80211.h
+  - remove superfluous ieee80211_* definitions from ieee80211.h
+  - remove rtl8192su's specific dead code
+  - cleanup ieee80211.h
+  - move rtl8192su's specific code out from ieee80211.h
+  - abstract rtl819su's specific code
+  - use list_for_each_safe() in ieee80211_crypto_deinit
+- switch to use shared "librtl" instead of private ieee80211 stack
+- switch to use LIB80211
+- switch to use MAC80211
+- switch to use EEPROM_93CX6
+- use kernel coding style
+- checkpatch.pl fixes
+- sparse fixes
+- integrate with drivers/net/wireless/rtl818x
+
+Please send any patches to Greg Kroah-Hartman <greg@kroah.com> and
+Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>.
