From wfp5p@virginia.edu Mon Mar 16 17:33:30 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:18:43 -0400
Subject: Staging: comedi: Remove pcl816_private typedef
To: greg@kroah.com
Message-ID: <20090317021842.14467.30253.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/pcl816.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/pcl816.c
+++ b/drivers/staging/comedi/drivers/pcl816.c
@@ -143,7 +143,7 @@ static const boardtype boardtypes[] = {
 };
 
 #define n_boardtypes (sizeof(boardtypes)/sizeof(boardtype))
-#define devpriv ((pcl816_private *)dev->private)
+#define devpriv ((struct pcl816_private *)dev->private)
 #define this_board ((const boardtype *)dev->board_ptr)
 
 static int pcl816_attach(struct comedi_device * dev, struct comedi_devconfig * it);
@@ -166,7 +166,8 @@ static struct comedi_driver driver_pcl81
 
 COMEDI_INITCLEANUP(driver_pcl816);
 
-typedef struct {
+struct pcl816_private {
+
 	unsigned int dma;	// used DMA, 0=don't use DMA
 	int dma_rtc;		// 1=RTC used with DMA, 0=no RTC alloc
 #ifdef unused
@@ -204,7 +205,8 @@ typedef struct {
 	struct timer_list rtc_irq_timer;	// timer for RTC sanity check
 	unsigned long rtc_freq;	// RTC int freq
 #endif
-} pcl816_private;
+};
+
 
 /*
 ==============================================================================
@@ -1037,7 +1039,7 @@ static int pcl816_attach(struct comedi_d
 		return -EIO;
 	}
 
-	if ((ret = alloc_private(dev, sizeof(pcl816_private))) < 0)
+	if ((ret = alloc_private(dev, sizeof(struct pcl816_private))) < 0)
 		return ret;	/* Can't alloc mem */
 
 	/* set up some name stuff */
