From wfp5p@virginia.edu Mon Mar 16 17:29:53 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:14:52 -0400
Subject: Staging: comedi: Remove dt2814_private typedef
To: greg@kroah.com
Message-ID: <20090317021451.14467.86151.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/dt2814.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/dt2814.c
+++ b/drivers/staging/comedi/drivers/dt2814.c
@@ -72,11 +72,13 @@ COMEDI_INITCLEANUP(driver_dt2814);
 
 static irqreturn_t dt2814_interrupt(int irq, void *dev PT_REGS_ARG);
 
-typedef struct {
+struct dt2814_private {
+
 	int ntrig;
 	int curadchan;
-} dt2814_private;
-#define devpriv ((dt2814_private *)dev->private)
+};
+
+#define devpriv ((struct dt2814_private *)dev->private)
 
 #define DT2814_TIMEOUT 10
 #define DT2814_MAX_SPEED 100000	/* Arbitrary 10 khz limit */
@@ -309,7 +311,7 @@ static int dt2814_attach(struct comedi_d
 
 	if ((ret = alloc_subdevices(dev, 1)) < 0)
 		return ret;
-	if ((ret = alloc_private(dev, sizeof(dt2814_private))) < 0)
+	if ((ret = alloc_private(dev, sizeof(struct dt2814_private))) < 0)
 		return ret;
 
 	s = dev->subdevices + 0;
