From shawn.bohrer@gmail.com  Mon Oct 19 14:41:09 2009
From: Shawn Bohrer <shawn.bohrer@gmail.com>
Date: Sun, 18 Oct 2009 15:32:19 -0500
Subject: Staging: comedi: remove EXTERN macro since it is not used
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: devel@driverdev.osuosl.org
Message-ID: <1255897939-9247-5-git-send-email-shawn.bohrer@gmail.com>


Signed-off-by: Shawn Bohrer <shawn.bohrer@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/s626.h |    8 --------
 1 file changed, 8 deletions(-)

--- a/drivers/staging/comedi/drivers/s626.h
+++ b/drivers/staging/comedi/drivers/s626.h
@@ -76,14 +76,6 @@
 #define FALSE   (0)
 #endif
 
-#if !defined(EXTERN)
-#if defined(__cplusplus)
-#define EXTERN extern "C"
-#else
-#define EXTERN extern
-#endif
-#endif
-
 #if !defined(INLINE)
 #define INLINE static __inline
 #endif
