From foo@baz Thu Dec  3 16:27:19 PST 2009
Date: Thu, 03 Dec 2009 16:27:19 -0800
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: samsung-laptop: add TODO file

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/samsung-laptop/TODO |    5 +++++
 1 file changed, 5 insertions(+)

--- /dev/null
+++ b/drivers/staging/samsung-laptop/TODO
@@ -0,0 +1,5 @@
+TODO:
+	- review from other developers
+	- figure out ACPI video issues
+
+Please send patches to Greg Kroah-Hartman <gregkh@suse.de>
