From jmm@debian.org  Mon Jan 26 09:50:18 2009
From: Moritz Muehlenhoff <jmm@debian.org>
Date: Sun, 25 Jan 2009 21:55:09 +0100
Subject: Staging: wlan-ng: Remove dead/unused code from p80211conv.c
To: greg@kroah.com
Cc: Moritz Muehlenhoff <jmm@debian.org>
Message-ID: <1232916910-5075-15-git-send-email-jmm@debian.org>


Signed-off-by: Moritz Muehlenhoff <jmm@debian.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/wlan-ng/p80211conv.c |   19 -------------------
 1 file changed, 19 deletions(-)

--- a/drivers/staging/wlan-ng/p80211conv.c
+++ b/drivers/staging/wlan-ng/p80211conv.c
@@ -83,27 +83,12 @@
 
 
 /*================================================================*/
-/* Local Constants */
-
-/*================================================================*/
-/* Local Macros */
-
-
-/*================================================================*/
-/* Local Types */
-
-
-/*================================================================*/
 /* Local Static Definitions */
 
 static u8	oui_rfc1042[] = {0x00, 0x00, 0x00};
 static u8	oui_8021h[] = {0x00, 0x00, 0xf8};
 
 /*================================================================*/
-/* Local Function Declarations */
-
-
-/*================================================================*/
 /* Function Definitions */
 
 /*----------------------------------------------------------------
@@ -225,11 +210,7 @@ int skb_ether_to_p80211( wlandevice_t *w
 	if ((wlandev->hostwep & HOSTWEP_PRIVACYINVOKED) && (wlandev->hostwep & HOSTWEP_ENCRYPT)) {
 		// XXXX need to pick keynum other than default?
 
-#if 1
 		p80211_wep->data = kmalloc(skb->len, GFP_ATOMIC);
-#else
-		p80211_wep->data = skb->data;
-#endif
 
 		if ((foo = wep_encrypt(wlandev, skb->data, p80211_wep->data,
 				       skb->len,
