From wfp5p@virginia.edu  Wed Mar 18 21:42:44 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:17:01 -0400
Subject: Staging: comedi: pcl730: Remove boardtype typedef
To: greg@kroah.com
Message-ID: <20090317021700.14467.47657.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/pcl730.c |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

--- a/drivers/staging/comedi/drivers/pcl730.c
+++ b/drivers/staging/comedi/drivers/pcl730.c
@@ -29,19 +29,21 @@ The ACL-7130 card have an 8254 timer/cou
 static int pcl730_attach(struct comedi_device * dev, struct comedi_devconfig * it);
 static int pcl730_detach(struct comedi_device * dev);
 
-typedef struct {
+struct pcl730_board {
+
 	const char *name;	// board name
 	unsigned int io_range;	// len of I/O space
-} boardtype;
+};
+
 
-static const boardtype boardtypes[] = {
+static const struct pcl730_board boardtypes[] = {
 	{"pcl730", PCL730_SIZE,},
 	{"iso730", PCL730_SIZE,},
 	{"acl7130", ACL7130_SIZE,},
 };
 
-#define n_boardtypes (sizeof(boardtypes)/sizeof(boardtype))
-#define this_board ((const boardtype *)dev->board_ptr)
+#define n_boardtypes (sizeof(boardtypes)/sizeof(struct pcl730_board))
+#define this_board ((const struct pcl730_board *)dev->board_ptr)
 
 static struct comedi_driver driver_pcl730 = {
       driver_name:"pcl730",
@@ -50,7 +52,7 @@ static struct comedi_driver driver_pcl73
       detach:pcl730_detach,
       board_name:&boardtypes[0].name,
       num_names:n_boardtypes,
-      offset:sizeof(boardtype),
+      offset:sizeof(struct pcl730_board),
 };
 
 COMEDI_INITCLEANUP(driver_pcl730);
