From rpjday@crashcourse.ca  Thu Nov 19 11:51:13 2009
From: "Robert P. J. Day" <rpjday@crashcourse.ca>
Date: Fri, 13 Nov 2009 15:13:43 -0500 (EST)
Subject: Staging: usbip: Fix typo "Contoroller".
Cc: Greg KH <greg@kroah.com>
Message-ID: <alpine.LFD.2.00.0911131512480.12802@localhost>


Signed-off-by: Robert P. J. Day <rpjday@crashcourse.ca>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/usbip/vhci_hcd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/usbip/vhci_hcd.c
+++ b/drivers/staging/usbip/vhci_hcd.c
@@ -56,7 +56,7 @@ static void vhci_stop(struct usb_hcd *hc
 static int vhci_get_frame_number(struct usb_hcd *hcd);
 
 static const char driver_name[] = "vhci_hcd";
-static const char driver_desc[] = "USB/IP Virtual Host Contoroller";
+static const char driver_desc[] = "USB/IP Virtual Host Controller";
 
 struct vhci_hcd *the_controller;
 
