From wfp5p@virginia.edu Mon Mar 16 17:23:53 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:09:07 -0400
Subject: Staging: comedi: Remove counter_mode_register_t typedef
To: greg@kroah.com
Message-ID: <20090317020907.14467.5812.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/s526.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/comedi/drivers/s526.c
+++ b/drivers/staging/comedi/drivers/s526.c
@@ -112,7 +112,7 @@ static const int s526_ports[] = {
 	REG_EEC
 };
 
-typedef struct {
+struct counter_mode_register_t {
 	unsigned short coutSource:1;
 	unsigned short coutPolarity:1;
 	unsigned short autoLoadResetRcap:3;
@@ -124,10 +124,10 @@ typedef struct {
 	unsigned short outputRegLatchCtrl:1;
 	unsigned short preloadRegSel:1;
 	unsigned short reserved:1;
-} counter_mode_register_t;
+};
 
 union {
-	counter_mode_register_t reg;
+	struct counter_mode_register_t reg;
 	unsigned short value;
 } cmReg;
 
