From wfp5p@virginia.edu Thu Mar 19 13:05:19 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Thu, 19 Mar 2009 17:58:41 -0400
Subject: Staging: comedi: Remove ni6527_private typedef
To: greg@kroah.com
Cc: abbotti@mev.co.uk, fmhess@users.sourceforge.net, ds@schleef.org
Message-ID: <20090319215841.10220.91017.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/ni_6527.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/ni_6527.c
+++ b/drivers/staging/comedi/drivers/ni_6527.c
@@ -112,12 +112,13 @@ static DEFINE_PCI_DEVICE_TABLE(ni6527_pc
 
 MODULE_DEVICE_TABLE(pci, ni6527_pci_table);
 
-typedef struct {
+struct ni6527_private {
 	struct mite_struct *mite;
 	unsigned int filter_interval;
 	unsigned int filter_enable;
-} ni6527_private;
-#define devpriv ((ni6527_private *)dev->private)
+};
+
+#define devpriv ((struct ni6527_private *)dev->private)
 
 static int ni6527_find_device(struct comedi_device * dev, int bus, int slot);
 
@@ -370,7 +371,7 @@ static int ni6527_attach(struct comedi_d
 
 	printk("comedi%d: ni6527:", dev->minor);
 
-	if ((ret = alloc_private(dev, sizeof(ni6527_private))) < 0)
+	if ((ret = alloc_private(dev, sizeof(struct ni6527_private))) < 0)
 		return ret;
 
 	ret = ni6527_find_device(dev, it->options[0], it->options[1]);
