From alan@lxorguk.ukuu.org.uk  Wed Aug 12 13:22:33 2009
From: Alan Cox <alan@lxorguk.ukuu.org.uk>
Date: Fri, 07 Aug 2009 19:25:06 +0100
Subject: Staging: sep: remove module int macro
To: greg@kroah.com, mark.a.allyn@intel.com
Message-ID: <20090807182504.19360.77643.stgit@localhost.localdomain>


From: Alan Cox <alan@linux.intel.com>

Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/sep/sep_driver.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)


--- a/drivers/staging/sep/sep_driver.c
+++ b/drivers/staging/sep/sep_driver.c
@@ -153,7 +153,8 @@ static void sep_load_rom_code(struct sep
 --------------------------------------------*/
 
 /* debug messages level */
-INT_MODULE_PARM(sepDebug, 0x0);
+static int sepDebug;
+module_param(sepDebug, int , 0);
 MODULE_PARM_DESC(sepDebug, "Flag to enable SEP debug messages");
 
 /* Keep this a single static object for now to keep the conversion easy */
