From wfp5p@virginia.edu  Mon Jul 27 14:09:31 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 27 Jul 2009 16:47:34 -0400
Subject: Staging: hv: vmbus_drv.c: fix print formatting
To: greg@kroah.com
Cc: hjanssen@microsoft.com
Message-ID: <1248727665-23969-13-git-send-email-wfp5p@virginia.edu>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/vmbus_drv.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/hv/vmbus_drv.c
+++ b/drivers/staging/hv/vmbus_drv.c
@@ -647,7 +647,7 @@ static int vmbus_child_device_register(D
 	ret = child_device_ctx->probe_error;
 
 	if (ret)
-		DPRINT_ERR(VMBUS_DRV, "unable to register child device (%p) (%d)", &child_device_ctx->device);
+		DPRINT_ERR(VMBUS_DRV, "unable to register child device (%p)", &child_device_ctx->device);
 	else
 		DPRINT_INFO(VMBUS_DRV, "child device (%p) registered", &child_device_ctx->device);
 
