From alan@linux.intel.com  Thu Aug 27 14:22:31 2009
From: Alan Cox <alan@linux.intel.com>
Date: Thu, 27 Aug 2009 10:58:37 +0100
Subject: Staging: et131x: kill unused RCV_REF macros
To: greg@kroah.com
Message-ID: <20090827095833.23892.77351.stgit@localhost.localdomain>


Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/et131x/et131x_adapter.h |    2 --
 1 file changed, 2 deletions(-)


--- a/drivers/staging/et131x/et131x_adapter.h
+++ b/drivers/staging/et131x/et131x_adapter.h
@@ -108,8 +108,6 @@
 #define MP_TEST_FLAGS(_M, _F)       (((_M)->Flags & (_F)) == (_F))
 #define MP_IS_FLAG_CLEAR(_M, _F)    (((_M)->Flags & (_F)) == 0)
 
-#define MP_INC_RCV_REF(_A)          atomic_inc(&(_A)->RcvRefCount)
-#define MP_DEC_RCV_REF(_A)          atomic_dec(&(_A)->RcvRefCount)
 #define MP_GET_RCV_REF(_A)          atomic_read(&(_A)->RcvRefCount)
 
 /* Macros specific to the private adapter structure */
