From bbeare1@gmail.com  Fri Dec 18 10:07:29 2009
From: Bruce Beare <bbeare1@gmail.com>
Date: Fri, 20 Nov 2009 15:46:10 -0800
Subject: Staging: comedi: pcmad: Cleanup: Remove unneeded braces
To: gregkh@suse.de
Cc: devel@driverdev.osuosl.org, Bruce Beare <bbeare1@gmail.com>, linux-kernel@vger.kernel.org
Message-ID: <1258760771-24311-3-git-send-email-bbeare1@gmail.com>



Signed-off-by: Bruce Beare <bbeare1@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/pcmad.c |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/pcmad.c
+++ b/drivers/staging/comedi/drivers/pcmad.c
@@ -113,9 +113,8 @@ static int pcmad_ai_insn_read(struct com
 		data[n] = inb(dev->iobase + PCMAD_LSB);
 		data[n] |= (inb(dev->iobase + PCMAD_MSB) << 8);
 
-		if (devpriv->twos_comp) {
+		if (devpriv->twos_comp)
 			data[n] ^= (1 << (this_board->n_ai_bits - 1));
-		}
 	}
 
 	return n;
@@ -168,9 +167,8 @@ static int pcmad_detach(struct comedi_de
 {
 	printk("comedi%d: pcmad: remove\n", dev->minor);
 
-	if (dev->irq) {
+	if (dev->irq)
 		free_irq(dev->irq, dev);
-	}
 	if (dev->iobase)
 		release_region(dev->iobase, PCMAD_SIZE);
 
