From wfp5p@virginia.edu Mon Mar 16 17:26:55 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:12:10 -0400
Subject: Staging: comedi: Remove daqboard2000_private typedef
To: greg@kroah.com
Message-ID: <20090317021210.14467.16089.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/daqboard2000.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/daqboard2000.c
+++ b/drivers/staging/comedi/drivers/daqboard2000.c
@@ -325,7 +325,7 @@ static DEFINE_PCI_DEVICE_TABLE(daqboard2
 
 MODULE_DEVICE_TABLE(pci, daqboard2000_pci_table);
 
-typedef struct {
+struct daqboard2000_private {
 	enum {
 		card_daqboard_2000
 	} card;
@@ -334,9 +334,9 @@ typedef struct {
 	void *plx;
 	int got_regions;
 	unsigned int ao_readback[2];
-} daqboard2000_private;
+};
 
-#define devpriv ((daqboard2000_private*)dev->private)
+#define devpriv ((struct daqboard2000_private *)dev->private)
 
 static void writeAcqScanListEntry(struct comedi_device * dev, u16 entry)
 {
@@ -731,7 +731,7 @@ static int daqboard2000_attach(struct co
 	bus = it->options[0];
 	slot = it->options[1];
 
-	result = alloc_private(dev, sizeof(daqboard2000_private));
+	result = alloc_private(dev, sizeof(struct daqboard2000_private));
 	if (result < 0) {
 		return -ENOMEM;
 	}
