From zbr@ioremap.net  Thu Feb 12 11:38:35 2009
From: Evgeniy Polyakov <zbr@ioremap.net>
Date: Tue, 10 Feb 2009 00:30:15 +0300
Subject: Staging: pohmelfs: select crypto modules from the config.
To: GregKH <greg@kroah.com>
Message-ID: <20090209213015.GA14562@ioremap.net>
Content-Disposition: inline


Signed-off-by: Evgeniy Polyakov <zbr@ioremap.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/pohmelfs/Kconfig |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/staging/pohmelfs/Kconfig
+++ b/drivers/staging/pohmelfs/Kconfig
@@ -1,6 +1,8 @@
 config POHMELFS
 	tristate "POHMELFS filesystem support"
 	select CONNECTOR
+	select CRYPTO_BLKCIPHER
+	select CRYPTO_HMAC
 	help
 	  POHMELFS stands for Parallel Optimized Host Message Exchange Layered File System.
 	  This is a network filesystem which supports coherent caching of data and metadata
