From zbr@ioremap.net  Thu Apr  9 14:38:25 2009
From: Evegniy Polyakov <zbr@ioremap.net>
Date: Fri, 27 Mar 2009 15:04:20 +0300
Subject: Staging: Pohmelfs: Disable read lock in pohmelfs_getattr().
To: GregKH <greg@kroah.com>
Cc: Evgeniy Polyakov <zbr@ioremap.net>
Message-ID: <1238155470-21068-4-git-send-email-zbr@ioremap.net>


From: Evgeniy Polyakov <zbr@ioremap.net>


Signed-off-by: Evgeniy Polyakov <zbr@ioremap.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/pohmelfs/inode.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/pohmelfs/inode.c
+++ b/drivers/staging/pohmelfs/inode.c
@@ -1171,11 +1171,11 @@ static int pohmelfs_getattr(struct vfsmo
 	struct inode *inode = dentry->d_inode;
 	struct pohmelfs_inode *pi = POHMELFS_I(inode);
 	int err;
-
+#if 0
 	err = pohmelfs_data_lock(pi, 0, ~0, POHMELFS_READ_LOCK);
 	if (err)
 		return err;
-
+#endif
 	dprintk("%s: ino: %llu, mode: %o, uid: %u, gid: %u, size: %llu.\n",
 			__func__, pi->ino, inode->i_mode, inode->i_uid,
 			inode->i_gid, inode->i_size);
