From wfp5p@virginia.edu Thu Mar 19 13:06:45 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Thu, 19 Mar 2009 18:00:05 -0400
Subject: Staging: comedi: Remove me4000_dio_info_t typedef
To: greg@kroah.com
Cc: abbotti@mev.co.uk, fmhess@users.sourceforge.net, ds@schleef.org
Message-ID: <20090319220005.10220.78203.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/me4000.h |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/comedi/drivers/me4000.h
+++ b/drivers/staging/comedi/drivers/me4000.h
@@ -305,9 +305,9 @@ struct me4000_ai_info {
 	int ex_trig_analog;
 };
 
-typedef struct me4000_dio_info {
+struct me4000_dio_info {
 	int count;
-} me4000_dio_info_t;
+};
 
 typedef struct me4000_cnt_info {
 	int count;
@@ -318,7 +318,7 @@ typedef struct me4000_board {
 	unsigned short device_id;
 	struct me4000_ao_info ao;
 	struct me4000_ai_info ai;
-	me4000_dio_info_t dio;
+	struct me4000_dio_info dio;
 	me4000_cnt_info_t cnt;
 } me4000_board_t;
 
