From foo@baz Fri Aug 28 16:26:11 PDT 2009
Date: Fri, 28 Aug 2009 16:26:11 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: remove typedefs from ChannelMgmt.c

Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/ChannelMgmt.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/hv/ChannelMgmt.c
+++ b/drivers/staging/hv/ChannelMgmt.c
@@ -32,10 +32,10 @@
 
 typedef void (*PFN_CHANNEL_MESSAGE_HANDLER)(struct vmbus_channel_message_header *msg);
 
-typedef struct _VMBUS_CHANNEL_MESSAGE_TABLE_ENTRY {
+struct vmbus_channel_message_table_entry {
 	enum vmbus_channel_message_type	messageType;
 	PFN_CHANNEL_MESSAGE_HANDLER messageHandler;
-} VMBUS_CHANNEL_MESSAGE_TABLE_ENTRY;
+};
 
 /* Internal routines */
 static void VmbusChannelOnOffer(struct vmbus_channel_message_header *hdr);
@@ -92,7 +92,7 @@ static const struct hv_guid gSupportedDe
 };
 
 /* Channel message dispatch table */
-static VMBUS_CHANNEL_MESSAGE_TABLE_ENTRY gChannelMessageTable[ChannelMessageCount]= {
+static struct vmbus_channel_message_table_entry gChannelMessageTable[ChannelMessageCount]= {
     {ChannelMessageInvalid,					NULL},
     {ChannelMessageOfferChannel,            VmbusChannelOnOffer},
     {ChannelMessageRescindChannelOffer,     VmbusChannelOnOfferRescind},
