From otavio@ossystems.com.br  Mon Oct 26 16:28:01 2009
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Mon, 12 Oct 2009 23:47:36 -0300
Subject: Staging: vt6656: use lowercase for VIA USB vendor id
To: Greg KH <greg@kroah.com>
Message-ID: <be7a9f3d0910121947o48a39380wf081303944a8ea0d@mail.gmail.com>

From: Otavio Salvador <otavio@ossystems.com.br>

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/vt6656/device.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/vt6656/device.h
+++ b/drivers/staging/vt6656/device.h
@@ -103,7 +103,7 @@
 #include "card.h"
 
 /*---------------------  Export Definitions -------------------------*/
-#define VNT_USB_VENDOR_ID                     0x160A
+#define VNT_USB_VENDOR_ID                     0x160a
 #define VNT_USB_PRODUCT_ID                    0x3184
 
 #define MAC_MAX_CONTEXT_REG     (256+128)
