From wfp5p@virginia.edu Thu Mar 19 13:05:40 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Thu, 19 Mar 2009 17:59:02 -0400
Subject: Staging: comedi: Remove ni_660x_board typedef
To: greg@kroah.com
Cc: abbotti@mev.co.uk, fmhess@users.sourceforge.net, ds@schleef.org
Message-ID: <20090319215902.10220.50479.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/ni_660x.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/ni_660x.c
+++ b/drivers/staging/comedi/drivers/ni_660x.c
@@ -379,13 +379,13 @@ static const unsigned GPCT_OFFSET[2] = {
 							   0x00, etc. */
 
 /* Board description*/
-typedef struct {
+struct ni_660x_board {
 	unsigned short dev_id;	/* `lspci` will show you this */
 	const char *name;
 	unsigned n_chips;	/* total number of TIO chips */
-} ni_660x_board;
+};
 
-static const ni_660x_board ni_660x_boards[] = {
+static const struct ni_660x_board ni_660x_boards[] = {
 	{
 	      dev_id:	0x2c60,
 	      name:	"PCI-6601",
@@ -439,7 +439,7 @@ static inline ni_660x_private *private(s
 }
 
 /* initialized in ni_660x_find_device() */
-static inline const ni_660x_board *board(struct comedi_device * dev)
+static inline const struct ni_660x_board *board(struct comedi_device * dev)
 {
 	return dev->board_ptr;
 }
