From wfp5p@virginia.edu Mon Mar 16 17:32:15 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:17:28 -0400
Subject: Staging: comedi: Remove pcmda12_private typedef
To: greg@kroah.com
Message-ID: <20090317021727.14467.46204.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/pcmda12.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/pcmda12.c
+++ b/drivers/staging/comedi/drivers/pcmda12.c
@@ -92,12 +92,14 @@ static const pcmda12_board pcmda12_board
  */
 #define thisboard ((const pcmda12_board *)dev->board_ptr)
 
-typedef struct {
+struct pcmda12_private {
+
 	unsigned int ao_readback[CHANS];
 	int simultaneous_xfer_mode;
-} pcmda12_private;
+};
+
 
-#define devpriv ((pcmda12_private *)(dev->private))
+#define devpriv ((struct pcmda12_private *)(dev->private))
 
 /*
  * The struct comedi_driver structure tells the Comedi core module
@@ -174,7 +176,7 @@ static int pcmda12_attach(struct comedi_
  * Allocate the private structure area.  alloc_private() is a
  * convenient macro defined in comedidev.h.
  */
-	if (alloc_private(dev, sizeof(pcmda12_private)) < 0) {
+	if (alloc_private(dev, sizeof(struct pcmda12_private)) < 0) {
 		printk("cannot allocate private data structure\n");
 		return -ENOMEM;
 	}
