From me@felipebalbi.com  Mon Jul 27 08:49:20 2009
From: Felipe Balbi <me@felipebalbi.com>
Date: Tue, 21 Jul 2009 15:52:25 +0300
Subject: Staging: cpc-usb: convert comment style in cpcusb.h
Cc: Greg Kroah-Hartman <gregkh@suse.de>, Felipe Balbi <me@felipebalbi.com>
Message-ID: <1248180746-18140-4-git-send-email-me@felipebalbi.com>


Signed-off-by: Felipe Balbi <me@felipebalbi.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/cpc-usb/cpcusb.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/cpc-usb/cpcusb.h
+++ b/drivers/staging/cpc-usb/cpcusb.h
@@ -75,7 +75,7 @@ typedef struct CPC_USB {
 #define CPC_DRIVER_VERSION "0.724"
 #define CPC_DRIVER_SERIAL  "not applicable"
 
-#define OBUF_SIZE 255		// 4096
+#define OBUF_SIZE 255		/* 4096 */
 
 /* read timeouts -- RD_NAK_TIMEOUT * RD_EXPIRE = Number of seconds */
 #define RD_NAK_TIMEOUT (10*HZ)	/* Default number of X seconds to wait */
