From wfp5p@virginia.edu Mon Mar 16 17:33:12 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:18:27 -0400
Subject: Staging: comedi: Remove pcl711_private typedef
To: greg@kroah.com
Message-ID: <20090317021826.14467.5247.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/pcl711.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/pcl711.c
+++ b/drivers/staging/comedi/drivers/pcl711.c
@@ -168,7 +168,8 @@ static struct comedi_driver driver_pcl71
 
 COMEDI_INITCLEANUP(driver_pcl711);
 
-typedef struct {
+struct pcl711_private {
+
 	int board;
 	int adchan;
 	int ntrig;
@@ -177,9 +178,10 @@ typedef struct {
 	unsigned int ao_readback[2];
 	unsigned int divisor1;
 	unsigned int divisor2;
-} pcl711_private;
+};
+
 
-#define devpriv ((pcl711_private *)dev->private)
+#define devpriv ((struct pcl711_private *)dev->private)
 
 static irqreturn_t pcl711_interrupt(int irq, void *d PT_REGS_ARG)
 {
@@ -548,7 +550,7 @@ static int pcl711_attach(struct comedi_d
 
 	if ((ret = alloc_subdevices(dev, 4)) < 0)
 		return ret;
-	if ((ret = alloc_private(dev, sizeof(pcl711_private))) < 0)
+	if ((ret = alloc_private(dev, sizeof(struct pcl711_private))) < 0)
 		return ret;
 
 	s = dev->subdevices + 0;
