From wfp5p@virginia.edu Mon Mar 16 17:34:43 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:19:58 -0400
Subject: Staging: comedi: Remove S526_GPCT_APP_CLASS typedef
To: greg@kroah.com
Message-ID: <20090317021958.14467.72165.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/s526.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/staging/comedi/drivers/s526.c
+++ b/drivers/staging/comedi/drivers/s526.c
@@ -135,19 +135,19 @@ union {
 
 /* Different Application Classes for GPCT Subdevices */
 /* The list is not exhaustive and needs discussion! */
-typedef enum {
+enum S526_GPCT_APP_CLASS {
 	CountingAndTimeMeasurement,
 	SinglePulseGeneration,
 	PulseTrainGeneration,
 	PositionMeasurement,
 	Miscellaneous
-} S526_GPCT_APP_CLASS;
+};
 
 /* Config struct for different GPCT subdevice Application Classes and
    their options
 */
 typedef struct s526GPCTConfig {
-	S526_GPCT_APP_CLASS app;
+	enum S526_GPCT_APP_CLASS app;
 	int data[MAX_GPCT_CONFIG_DATA];
 } s526_gpct_config_t;
 
