From shawn.bohrer@gmail.com  Wed Oct 28 09:38:50 2009
From: Shawn Bohrer <shawn.bohrer@gmail.com>
Date: Sun, 25 Oct 2009 22:46:49 -0500
Subject: Staging: vt6655 remove unneeded version.h
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: devel@driverdev.osuosl.org, Forest Bond <forest@alittletooquiet.net>, Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Message-ID: <1256528811-31981-5-git-send-email-shawn.bohrer@gmail.com>


Signed-off-by: Shawn Bohrer <shawn.bohrer@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/vt6655/device.h |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/staging/vt6655/device.h
+++ b/drivers/staging/vt6655/device.h
@@ -44,7 +44,6 @@
 #include <linux/timer.h>
 #include <linux/slab.h>
 #include <linux/interrupt.h>
-#include <linux/version.h>
 #include <linux/string.h>
 #include <linux/wait.h>
 #include <linux/if_arp.h>
