From wfp5p@virginia.edu Mon Mar 16 17:27:08 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:12:21 -0400
Subject: Staging: comedi: Remove das16_board typedef
To: greg@kroah.com
Message-ID: <20090317021221.14467.53809.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/das16.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/drivers/staging/comedi/drivers/das16.c
+++ b/drivers/staging/comedi/drivers/das16.c
@@ -355,7 +355,7 @@ static unsigned int das16_suggest_transf
 
 static void reg_dump(struct comedi_device * dev);
 
-typedef struct das16_board_struct {
+struct das16_board {
 	const char *name;
 	void *ai;
 	unsigned int ai_nbits;
@@ -371,9 +371,9 @@ typedef struct das16_board_struct {
 
 	unsigned int size;
 	unsigned int id;
-} das16_board;
+};
 
-static const struct das16_board_struct das16_boards[] = {
+static const struct das16_board das16_boards[] = {
 	{
 	      name:	"das-16",
 	      ai:	das16_ai_rinsn,
@@ -696,7 +696,7 @@ static const struct das16_board_struct d
 #endif
 };
 
-#define n_das16_boards ((sizeof(das16_boards))/(sizeof(das16_board)))
+#define n_das16_boards ((sizeof(das16_boards))/(sizeof(struct das16_board)))
 
 static int das16_attach(struct comedi_device * dev, struct comedi_devconfig * it);
 static int das16_detach(struct comedi_device * dev);
@@ -740,7 +740,7 @@ struct das16_private_struct {
 	volatile short timer_mode;	// true if using timer mode
 };
 #define devpriv ((struct das16_private_struct *)(dev->private))
-#define thisboard ((struct das16_board_struct *)(dev->board_ptr))
+#define thisboard ((struct das16_board *)(dev->board_ptr))
 
 static int das16_cmd_test(struct comedi_device * dev, struct comedi_subdevice * s,
 	struct comedi_cmd * cmd)
