From 61b835e32821fedc129daf3fe702dc7d2fd768db Mon Sep 17 00:00:00 2001
From: Frank Mori Hess <fmhess@users.sourceforge.ne
Date: Tue, 9 Dec 2008 14:47:22 +0000
Subject: Staging: comedi: Fixed minor numbers for subdevice files.


From: Frank Mori Hess <fmhess@users.sourceforge.ne
Cc: Ian Abbott <abbotti@mev.co.uk>
Cc: David Schleef <ds@schleef.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/comedi_fops.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/comedi/comedi_fops.c
+++ b/drivers/staging/comedi/comedi_fops.c
@@ -2188,7 +2188,7 @@ int comedi_alloc_subdevice_minor(comedi_
 	info->read_subdevice = s;
 	info->write_subdevice = s;
 	comedi_spin_lock_irqsave(&comedi_file_info_table_lock, flags);
-	for (i = COMEDI_FIRST_SUBDEVICE_MINOR; i < COMEDI_NUM_BOARD_MINORS; ++i) {
+	for (i = COMEDI_FIRST_SUBDEVICE_MINOR; i < COMEDI_NUM_MINORS; ++i) {
 		if (comedi_file_info_table[i] == NULL) {
 			comedi_file_info_table[i] = info;
 			break;
