From wfp5p@virginia.edu Mon Mar 16 17:29:18 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:14:30 -0400
Subject: Staging: comedi: Remove boardtype_t typedef in dt2801.c
To: greg@kroah.com
Message-ID: <20090317021430.14467.17899.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/dt2801.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/dt2801.c
+++ b/drivers/staging/comedi/drivers/dt2801.c
@@ -135,7 +135,8 @@ static const struct comedi_lrange range_
 	}
 };
 
-typedef struct {
+struct dt2801_board {
+
 	const char *name;
 	int boardcode;
 	int ad_diff;
@@ -143,12 +144,13 @@ typedef struct {
 	int adbits;
 	int adrangetype;
 	int dabits;
-} boardtype_t;
+};
+
 
 /* Typeid's for the different boards of the DT2801-series
    (taken from the test-software, that comes with the board)
    */
-static const boardtype_t boardtypes[] = {
+static const struct dt2801_board boardtypes[] = {
 	{
 	      name:	"dt2801",
 	      boardcode:0x09,
@@ -216,7 +218,7 @@ static const boardtype_t boardtypes[] = 
 };
 
 #define n_boardtypes ((sizeof(boardtypes))/(sizeof(boardtypes[0])))
-#define boardtype (*(const boardtype_t *)dev->board_ptr)
+#define boardtype (*(const struct dt2801_board *)dev->board_ptr)
 
 typedef struct {
 	const struct comedi_lrange *dac_range_types[2];
