From 2ameya@gmail.com  Fri Dec 18 10:21:11 2009
From: Ameya Palande <2ameya@gmail.com>
Date: Fri, 11 Dec 2009 19:10:11 +0200
Subject: Staging: mimio: Remove dead code
To: devel@driverdev.osuosl.org
Cc: mwilder@cs.nmsu.edu
Message-ID: <a08ab0d7491845527e866473eaafed4da56afb2d.1260550872.git.2ameya@gmail.com>


Signed-off-by: Ameya Palande <2ameya@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/mimio/mimio.c |   24 ------------------------
 1 file changed, 24 deletions(-)

--- a/drivers/staging/mimio/mimio.c
+++ b/drivers/staging/mimio/mimio.c
@@ -505,23 +505,6 @@ static int mimio_probe(struct usb_interf
 	input_set_abs_params(input_dev, ABS_Y, 0, MIMIO_YRANGE_MAX, 0, 0);
 	input_dev->absbit[BIT_WORD(ABS_MISC)] |= BIT_MASK(ABS_MISC);
 
-#if 0
-	input_dev->absmin[ABS_X] = 0;
-	input_dev->absmin[ABS_Y] = 0;
-	input_dev->absmax[ABS_X] = 9600;
-	input_dev->absmax[ABS_Y] = 4800;
-	input_dev->absfuzz[ABS_X] = 0;
-	input_dev->absfuzz[ABS_Y] = 0;
-	input_dev->absflat[ABS_X] = 0;
-	input_dev->absflat[ABS_Y] = 0;
-#endif
-
-#if 0
-	/* this will just reduce the precision */
-	input_dev->absfuzz[ABS_X] = 8; /* experimental; may need to change */
-	input_dev->absfuzz[ABS_Y] = 8; /* experimental; may need to change */
-#endif
-
 	/*
 	 * Register the input device.
 	 */
@@ -720,13 +703,6 @@ static void mimio_rx_handler(struct mimi
 			if (instr_ofst[mimio->pktbuf.instr] >= 0) {
 				int code = BTN_TOOL_PEN +
 					   instr_ofst[mimio->last_pen_down];
-#if 0
-				/* Utter hack to ensure we get forwarded _AND_
-				 * so we can identify when a complete signal is
-				 * received */
-				mimio->idev->abs[ABS_Y] = -1;
-				mimio->idev->abs[ABS_X] = -1;
-#endif
 				/* input_regs(mimio->idev, regs); */
 				input_report_abs(mimio->idev, ABS_X, x);
 				input_report_abs(mimio->idev, ABS_Y, y);
