From wfp5p@virginia.edu Mon Mar 16 17:29:53 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:15:08 -0400
Subject: Staging: comedi: Remove fl512_private typedef
To: greg@kroah.com
Message-ID: <20090317021507.14467.13136.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/fl512.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/fl512.c
+++ b/drivers/staging/comedi/drivers/fl512.c
@@ -24,10 +24,12 @@ Configuration options:
 #include <linux/ioport.h>
 
 #define FL512_SIZE 16		/* the size of the used memory */
-typedef struct {
+struct fl512_private {
+
 	short ao_readback[2];
-} fl512_private;
-#define devpriv ((fl512_private *) dev->private)
+};
+
+#define devpriv ((struct fl512_private *) dev->private)
 
 static const struct comedi_lrange range_fl512 = { 4, {
 			BIP_RANGE(0.5),
@@ -138,7 +140,7 @@ static int fl512_attach(struct comedi_de
 	}
 	dev->iobase = iobase;
 	dev->board_name = "fl512";
-	if (alloc_private(dev, sizeof(fl512_private)) < 0)
+	if (alloc_private(dev, sizeof(struct fl512_private)) < 0)
 		return -ENOMEM;
 
 #if DEBUG
