From wfp5p@virginia.edu Mon Mar 16 17:33:48 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:18:59 -0400
Subject: Staging: comedi: Remove dt2811_private typedef
To: greg@kroah.com
Message-ID: <20090317021859.14467.22067.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/dt2811.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/dt2811.c
+++ b/drivers/staging/comedi/drivers/dt2811.c
@@ -238,7 +238,8 @@ static int dt2811_do_insn_bits(struct co
 	struct comedi_insn * insn, unsigned int * data);
 
 enum { card_2811_pgh, card_2811_pgl };
-typedef struct {
+
+struct dt2811_private {
 	int ntrig;
 	int curadchan;
 	enum {
@@ -249,9 +250,9 @@ typedef struct {
 	} dac_range[2];
 	const struct comedi_lrange *range_type_list[2];
 	unsigned int ao_readback[2];
-} dt2811_private;
+};
 
-#define devpriv ((dt2811_private *)dev->private)
+#define devpriv ((struct dt2811_private *)dev->private)
 
 static const struct comedi_lrange *dac_range_types[] = {
 	&range_bipolar5,
@@ -377,7 +378,7 @@ static int dt2811_attach(struct comedi_d
 
 	if ((ret = alloc_subdevices(dev, 4)) < 0)
 		return ret;
-	if ((ret = alloc_private(dev, sizeof(dt2811_private))) < 0)
+	if ((ret = alloc_private(dev, sizeof(struct dt2811_private))) < 0)
 		return ret;
 	switch (it->options[2]) {
 	case 0:
