From 2ameya@gmail.com  Fri Dec 18 10:20:54 2009
From: Ameya Palande <2ameya@gmail.com>
Date: Fri, 11 Dec 2009 19:10:10 +0200
Subject: Staging: mimio: Fix checkpatch.pl spacing errors
To: devel@driverdev.osuosl.org
Cc: mwilder@cs.nmsu.edu
Message-ID: <ab20d9e8fc69add31b4f1b84c75ed4c5be832420.1260550872.git.2ameya@gmail.com>


Signed-off-by: Ameya Palande <2ameya@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/mimio/mimio.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/mimio/mimio.c
+++ b/drivers/staging/mimio/mimio.c
@@ -145,7 +145,7 @@ static void mimio_rx_handler(struct mimi
 static int mimio_tx(struct mimio *, const char *, int);
 
 static char mimio_name[] = "VirtualInk mimio-Xi";
-static struct usb_device_id mimio_table [] = {
+static struct usb_device_id mimio_table[] = {
 	{ USB_DEVICE(MIMIO_VENDOR_ID, MIMIO_PRODUCT_ID) },
 	{ USB_DEVICE(0x0525, 0xa4a0) }, /* gadget zero firmware */
 	{ }
@@ -588,7 +588,7 @@ static int handle_mimio_rx_penupdown(str
 			*(mimio->pktbuf.p + 2);
 		if (x != *(mimio->pktbuf.p + 3)) {
 			dev_dbg(&mimio->idev->dev, "EV_PEN%s: bad xsum.\n",
-				down ? "DOWN":"UP");
+				down ? "DOWN" : "UP");
 			/* skip this event data */
 			mimio->pktbuf.p += 4;
 			/* decode any remaining events */
