From wfp5p@virginia.edu  Wed Mar 18 21:47:11 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:17:49 -0400
Subject: Staging: comedi: skel.c: Remove skel_private typedef
To: greg@kroah.com
Message-ID: <20090317021749.14467.47092.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/skel.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/skel.c
+++ b/drivers/staging/comedi/drivers/skel.c
@@ -131,7 +131,8 @@ MODULE_DEVICE_TABLE(pci, skel_pci_table)
 /* this structure is for data unique to this hardware driver.  If
    several hardware drivers keep similar information in this structure,
    feel free to suggest moving the variable to the struct comedi_device struct.  */
-typedef struct {
+struct skel_private {
+
 	int data;
 
 	/* would be useful for a PCI device */
@@ -139,12 +140,13 @@ typedef struct {
 
 	/* Used for AO readback */
 	unsigned int ao_readback[2];
-} skel_private;
+};
+
 /*
  * most drivers define the following macro to make it easy to
  * access the private structure.
  */
-#define devpriv ((skel_private *)dev->private)
+#define devpriv ((struct skel_private *)dev->private)
 
 /*
  * The struct comedi_driver structure tells the Comedi core module
@@ -225,7 +227,7 @@ static int skel_attach(struct comedi_dev
  * Allocate the private structure area.  alloc_private() is a
  * convenient macro defined in comedidev.h.
  */
-	if (alloc_private(dev, sizeof(skel_private)) < 0)
+	if (alloc_private(dev, sizeof(struct skel_private)) < 0)
 		return -ENOMEM;
 
 /*
