From bzolnier@gmail.com  Tue Aug 25 22:22:43 2009
From: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Date: Sun, 23 Aug 2009 15:31:09 +0200
Subject: Staging: rt28x0: MlmeAssocReqAction() fixes for wpa_supplicant support
To: "Greg Kroah-Hartman" <gregkh@suse.de>
Cc: devel@driverdev.osuosl.org, Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>, Marcin Slusarz <marcin.slusarz@gmail.com>, Mike Galbraith <efault@gmx.de>, linux-kernel@vger.kernel.org
Message-ID: <20090823133109.17784.11110.sendpatchset@localhost.localdomain>


From: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>

Signed-off-by: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/rt2860/sta/assoc.c |    6 ------
 1 file changed, 6 deletions(-)

--- a/drivers/staging/rt2860/sta/assoc.c
+++ b/drivers/staging/rt2860/sta/assoc.c
@@ -454,9 +454,7 @@ VOID MlmeAssocReqAction(
 				RSNIe = IE_WPA2;
 			}
 
-#ifdef RT30xx
 			if (pAd->StaCfg.WpaSupplicantUP != 1)
-#endif
             RTMPMakeRSNIE(pAd, pAd->StaCfg.AuthMode, pAd->StaCfg.WepStatus, BSS0);
 
             // Check for WPA PMK cache list
@@ -483,7 +481,6 @@ VOID MlmeAssocReqAction(
 				}
 			}
 
-#ifdef RT30xx
 			if (pAd->StaCfg.WpaSupplicantUP == 1)
 			{
 				MakeOutgoingFrame(pOutBuffer + FrameLen,    		&tmp,
@@ -491,7 +488,6 @@ VOID MlmeAssocReqAction(
 		                        	END_OF_ARGS);
 			}
 			else
-#endif
 			{
 				MakeOutgoingFrame(pOutBuffer + FrameLen,    		&tmp,
 				              		1,                              &RSNIe,
@@ -502,9 +498,7 @@ VOID MlmeAssocReqAction(
 
 			FrameLen += tmp;
 
-#ifdef RT30xx
 			if (pAd->StaCfg.WpaSupplicantUP != 1)
-#endif
 			{
 	            // Append Variable IE
 	            NdisMoveMemory(pAd->StaCfg.ReqVarIEs + VarIesOffset, &RSNIe, 1);
