From wfp5p@virginia.edu Mon Mar 16 17:32:24 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 16 Mar 2009 22:17:38 -0400
Subject: Staging: comedi: Remove s526_private typedef
To: greg@kroah.com
Message-ID: <20090317021738.14467.63505.stgit@viridian.itc.Virginia.EDU>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/staging/comedi/drivers/s526.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/staging/comedi/drivers/s526.c
+++ b/drivers/staging/comedi/drivers/s526.c
@@ -191,7 +191,8 @@ static const s526_board s526_boards[] = 
 /* this structure is for data unique to this hardware driver.  If
    several hardware drivers keep similar information in this structure,
    feel free to suggest moving the variable to the struct comedi_device struct.  */
-typedef struct {
+struct s526_private {
+
 	int data;
 
 	/* would be useful for a PCI device */
@@ -202,12 +203,13 @@ typedef struct {
 
 	s526_gpct_config_t s526_gpct_config[4];
 	unsigned short s526_ai_config;
-} s526_private;
+};
+
 /*
  * most drivers define the following macro to make it easy to
  * access the private structure.
  */
-#define devpriv ((s526_private *)dev->private)
+#define devpriv ((struct s526_private *)dev->private)
 
 /*
  * The struct comedi_driver structure tells the Comedi core module
@@ -307,7 +309,7 @@ static int s526_attach(struct comedi_dev
  * Allocate the private structure area.  alloc_private() is a
  * convenient macro defined in comedidev.h.
  */
-	if (alloc_private(dev, sizeof(s526_private)) < 0)
+	if (alloc_private(dev, sizeof(struct s526_private)) < 0)
 		return -ENOMEM;
 
 /*
