Index: ubuntu-saucy/drivers/platform/x86/chromeos_laptop.c
===================================================================
--- ubuntu-saucy.orig/drivers/platform/x86/chromeos_laptop.c	2013-10-12 15:19:39.000000000 +0200
+++ ubuntu-saucy/drivers/platform/x86/chromeos_laptop.c	2013-10-12 15:46:20.633292838 +0200
@@ -27,6 +27,7 @@
 #include <linux/input.h>
 #include <linux/interrupt.h>
 #include <linux/module.h>
+#include <linux/platform_device.h>
 
 #define ATMEL_TP_I2C_ADDR	0x4b
 #define ATMEL_TP_I2C_BL_ADDR	0x25
@@ -301,6 +302,20 @@
 	return 0;
 }
 
+static struct platform_device *kb_backlight_device;
+
+static int __init setup_keyboard_backlight(const struct dmi_system_id *id)
+{
+  	kb_backlight_device =
+    		platform_device_register_simple("chromeos-keyboard-leds",
+            	-1, NULL, 0);
+  	if (IS_ERR(kb_backlight_device)) {
+    		pr_warn("Error registering Chrome OS keyboard LEDs.\n");
+    		kb_backlight_device = NULL;
+  	}
+  	return 0;
+}
+
 static struct dmi_system_id __initdata chromeos_laptop_dmi_table[] = {
 	{
 		.ident = "Samsung Series 5 550 - Touchpad",
@@ -377,6 +392,14 @@
 		},
 		.callback = setup_tsl2563_als,
 	},
+  	{
+    		.ident = "Chromebook Pixel - Keyboard backlight",
+    		.matches = {
+      			DMI_MATCH(DMI_SYS_VENDOR, "GOOGLE"),
+      			DMI_MATCH(DMI_PRODUCT_NAME, "Link"),
+    		},
+    		.callback = setup_keyboard_backlight,
+  	},
 	{ }
 };
 MODULE_DEVICE_TABLE(dmi, chromeos_laptop_dmi_table);
@@ -398,6 +421,8 @@
 		i2c_unregister_device(tp);
 	if (ts)
 		i2c_unregister_device(ts);
+	if (kb_backlight_device)
+		platform_device_unregister(kb_backlight_device);
 }
 
 module_init(chromeos_laptop_init);
Index: ubuntu-saucy/drivers/video/backlight/Makefile
===================================================================
--- ubuntu-saucy.orig/drivers/video/backlight/Makefile	2013-09-29 16:57:53.492097421 +0200
+++ ubuntu-saucy/drivers/video/backlight/Makefile	2013-10-12 15:48:10.521295057 +0200
@@ -50,3 +50,4 @@
 obj-$(CONFIG_BACKLIGHT_TOSA)		+= tosa_bl.o
 obj-$(CONFIG_BACKLIGHT_TPS65217)	+= tps65217_bl.o
 obj-$(CONFIG_BACKLIGHT_WM831X)		+= wm831x_bl.o
+obj-$(CONFIG_BACKLIGHT_CHROMEOS_KEYBOARD) += chromeos_keyboard_bl.o
