---
 lib/dynamic_printk.c |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/lib/dynamic_printk.c
+++ b/lib/dynamic_printk.c
@@ -288,6 +288,9 @@ static ssize_t pr_debug_write(struct fil
 					num_enabled++;
 					dynamic_enabled = DYNAMIC_ENABLED_SOME;
 					err = 0;
+					printk(KERN_DEBUG
+					       "debugging enabled for module %s",
+					       elem->name);
 				} else if (!value && (elem->enable == 1)) {
 					elem->enable = 0;
 					num_enabled--;
@@ -304,6 +307,9 @@ static ssize_t pr_debug_write(struct fil
 						dynamic_enabled =
 							DYNAMIC_ENABLED_NONE;
 					err = 0;
+					printk(KERN_DEBUG
+					       "debugging disabled for module "
+					       "%s", elem->name);
 				}
 			}
 		}
