---
 drivers/usb/host/ehci-hub.c |    3 +++
 1 file changed, 3 insertions(+)

--- gregkh-2.6.orig/drivers/usb/host/ehci-hub.c
+++ gregkh-2.6/drivers/usb/host/ehci-hub.c
@@ -225,6 +225,8 @@ ehci_hub_status_data (struct usb_hcd *hc
 	int		ports, i, retval = 1;
 	unsigned long	flags;
 
+	printk("GREGKH: %s: hcd->state=0x%x\n", __FUNCTION__, hcd->state);
+
 	/* if !USB_SUSPEND, root hub timers won't get shut down ... */
 	if (!HC_IS_RUNNING(hcd->state))
 		return 0;
@@ -254,6 +256,7 @@ ehci_hub_status_data (struct usb_hcd *hc
 		}
 		if (!(temp & PORT_CONNECT))
 			ehci->reset_done [i] = 0;
+		printk("GREGKH: %s: temp & PORT_RESUME=%d\n", __FUNCTION__, temp&PORT_RESUME);
 		if ((temp & (PORT_CSC | PORT_PEC | PORT_OCC)) != 0
 				// PORT_STAT_C_SUSPEND?
 				|| ((temp & PORT_RESUME) != 0
