 include/linux/i2o.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/include/linux/i2o.h	2005-09-22 07:40:42.000000000 -0700
+++ gregkh-2.6/include/linux/i2o.h	2005-09-22 08:33:38.000000000 -0700
@@ -67,7 +67,7 @@
 
 	struct semaphore lock;	/* device lock */
 
-	struct class_device classdev;	/* i2o device class */
+	struct class_device *classdev;	/* i2o device class device */
 };
 
 /*
