---
 drivers/usb/serial/usb-serial.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/usb/serial/usb-serial.c
+++ b/drivers/usb/serial/usb-serial.c
@@ -1212,12 +1212,15 @@ void usb_serial_deregister(struct usb_se
 
 /* If the usb-serial core is built into the core, the usb-serial drivers
    need these symbols to load properly as modules. */
-EXPORT_SYMBOL_GPL(usb_serial_register);
+//EXPORT_SYMBOL_GPL(usb_serial_register);
+EXPORT_SYMBOL_NS(usb, usb_serial_register);
 EXPORT_SYMBOL_GPL(usb_serial_deregister);
 EXPORT_SYMBOL_GPL(usb_serial_probe);
 EXPORT_SYMBOL_GPL(usb_serial_disconnect);
 EXPORT_SYMBOL_GPL(usb_serial_port_softint);
 
+MODULE_NAMESPACE_ALLOW(usb, visor);
+MODULE_NAMESPACE_ALLOW(usb, sierra);
 
 /* Module information */
 MODULE_AUTHOR( DRIVER_AUTHOR );
