Subject: 

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- gregkh-2.6.orig/drivers/usb/core/hub.c	2005-09-29 00:23:55.000000000 -0700
+++ gregkh-2.6/drivers/usb/core/hub.c	2005-09-30 14:49:35.000000000 -0700
@@ -570,7 +570,8 @@
 	dev_info (hub_dev, "%d port%s detected\n", hdev->maxchild,
 		(hdev->maxchild == 1) ? "" : "s");
 
-	le16_to_cpus(&hub->descriptor->wHubCharacteristics);
+	hub->descriptor->wHubCharacteristics =
+		le16_to_cpu((__le16)hub->descriptor->wHubCharacteristics);
 
 	if (hub->descriptor->wHubCharacteristics & HUB_CHAR_COMPOUND) {
 		int	i;
