From akpm@osdl.org Sun Nov  6 23:40:06 2005
Message-Id: <200511070739.jA77dcqh023271@shell0.pdx.osdl.net>
Subject: PCI: pci_find_device remove (alpha/kernel/sys_alcor.c)
To: greg@kroah.com
Cc: akpm@osdl.org, jirislaby@gmail.com, xslaby@fi.muni.cz
From: akpm@osdl.org
Date: Sun, 06 Nov 2005 23:39:30 -0800


From: Jiri Slaby <jirislaby@gmail.com>

Signed-off-by: Jiri Slaby <xslaby@fi.muni.cz>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 arch/alpha/kernel/sys_alcor.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/arch/alpha/kernel/sys_alcor.c
+++ gregkh-2.6/arch/alpha/kernel/sys_alcor.c
@@ -254,7 +254,7 @@ alcor_init_pci(void)
 	 * motherboard, by looking for a 21040 TULIP in slot 6, which is
 	 * built into XLT and BRET/MAVERICK, but not available on ALCOR.
 	 */
-	dev = pci_find_device(PCI_VENDOR_ID_DEC,
+	dev = pci_get_device(PCI_VENDOR_ID_DEC,
 			      PCI_DEVICE_ID_DEC_TULIP,
 			      NULL);
 	if (dev && dev->devfn == PCI_DEVFN(6,0)) {
@@ -262,6 +262,7 @@ alcor_init_pci(void)
 		printk(KERN_INFO "%s: Detected AS500 or XLT motherboard.\n",
 		       __FUNCTION__);
 	}
+	pci_dev_put(dev);
 }
 
 
