From akpm@osdl.org Mon Aug 14 23:07:45 2006
Message-Id: <200608150607.k7F67cR1013148@shell0.pdx.osdl.net>
Subject: PCI: drivers/pci/hotplug/acpiphp_glue.c: make a function static
To: greg@kroah.com
Cc: akpm@osdl.org, bunk@stusta.de, muneda.takahiro@jp.fujitsu.com
From: akpm@osdl.org
Date: Mon, 14 Aug 2006 23:07:38 -0700

From: Adrian Bunk <bunk@stusta.de>

Signed-off-by: Adrian Bunk <bunk@stusta.de>
Acked-by: MUNEDA Takahiro <muneda.takahiro@jp.fujitsu.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/hotplug/acpiphp_glue.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/pci/hotplug/acpiphp_glue.c
+++ gregkh-2.6/drivers/pci/hotplug/acpiphp_glue.c
@@ -997,7 +997,7 @@ acpiphp_bus_add_out:
  * @handle: handle to acpi namespace
  *
  */
-int acpiphp_bus_trim(acpi_handle handle)
+static int acpiphp_bus_trim(acpi_handle handle)
 {
 	struct acpi_device *device;
 	int retval;
