From akpm@linux-foundation.org Mon Jul  9 11:56:09 2007
From: Milind Arun Choudhary <milindchoudhary@gmail.com>
Date: Mon, 09 Jul 2007 11:55:54 -0700
Subject: PCI: pcie: remove SPIN_LOCK_UNLOCKED
To: greg@kroah.com
Cc: akpm@linux-foundation.org, milindchoudhary@gmail.com
Message-ID: <200707091855.l69ItsDD002119@imap1.linux-foundation.org>


From: Milind Arun Choudhary <milindchoudhary@gmail.com>


Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/pcie/aer/aerdrv.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/pci/pcie/aer/aerdrv.c
+++ b/drivers/pci/pcie/aer/aerdrv.c
@@ -157,7 +157,7 @@ static struct aer_rpc* aer_alloc_rpc(str
 	 * Initialize Root lock access, e_lock, to Root Error Status Reg,
 	 * Root Error ID Reg, and Root error producer/consumer index.
 	 */
-	rpc->e_lock = SPIN_LOCK_UNLOCKED;
+	spin_lock_init(&rpc->e_lock);
 
 	rpc->rpd = dev;
 	INIT_WORK(&rpc->dpc_handler, aer_isr);
