From owner-linux-pci@atrey.karlin.mff.cuni.cz Mon Aug 27 16:17:57 2007
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Mon, 27 Aug 2007 16:17:47 -0700
Subject: PCI: quirk amd_8131_mmrbc: Omit reading pci revision ID
To: gregkh@suse.de
Cc: peter.oruba@amd.com, linux-pci@atrey.karlin.mff.cuni.cz
Message-ID: <20070827231743.27448.72060.stgit@localhost.localdomain>


Signed-off-by: Auke Kok <auke-jan.h.kok@intel.com>
Cc: Peter Oruba <peter.oruba@amd.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/quirks.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -629,12 +629,9 @@ DECLARE_PCI_FIXUP_RESUME(PCI_VENDOR_ID_A
  */
 static void __init quirk_amd_8131_mmrbc(struct pci_dev *dev)
 {
-	unsigned char revid;
-
-	pci_read_config_byte(dev, PCI_REVISION_ID, &revid);
-	if (dev->subordinate && revid <= 0x12) {
+	if (dev->subordinate && dev->revision <= 0x12) {
 		printk(KERN_INFO "AMD8131 rev %x detected, disabling PCI-X "
-				"MMRBC\n", revid);
+				"MMRBC\n", dev->revision);
 		dev->subordinate->bus_flags |= PCI_BUS_FLAGS_NO_MMRBC;
 	}
 }
