From akpm@osdl.org Sun Nov  6 23:40:13 2005
Message-Id: <200511070739.jA77dhPf023293@shell0.pdx.osdl.net>
Subject: PCI: pci_find_device remove (sparc64/kernel/ebus.c)
To: greg@kroah.com
Cc: akpm@osdl.org, jirislaby@gmail.com, xslaby@fi.muni.cz
From: akpm@osdl.org
Date: Sun, 06 Nov 2005 23:39:35 -0800


From: Jiri Slaby <jirislaby@gmail.com>

Signed-off-by: Jiri Slaby <xslaby@fi.muni.cz>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 arch/sparc64/kernel/ebus.c |   15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

--- gregkh-2.6.orig/arch/sparc64/kernel/ebus.c
+++ gregkh-2.6/arch/sparc64/kernel/ebus.c
@@ -527,18 +527,12 @@ static struct pci_dev *find_next_ebus(st
 {
 	struct pci_dev *pdev = start;
 
-	do {
-		pdev = pci_find_device(PCI_VENDOR_ID_SUN, PCI_ANY_ID, pdev);
-		if (pdev &&
-		    (pdev->device == PCI_DEVICE_ID_SUN_EBUS ||
-		     pdev->device == PCI_DEVICE_ID_SUN_RIO_EBUS))
+	while ((pdev = pci_get_device(PCI_VENDOR_ID_SUN, PCI_ANY_ID, pdev)))
+		if (pdev->device == PCI_DEVICE_ID_SUN_EBUS ||
+			pdev->device == PCI_DEVICE_ID_SUN_RIO_EBUS)
 			break;
-	} while (pdev != NULL);
 
-	if (pdev && (pdev->device == PCI_DEVICE_ID_SUN_RIO_EBUS))
-		*is_rio_p = 1;
-	else
-		*is_rio_p = 0;
+	*is_rio_p = !!(pdev && (pdev->device == PCI_DEVICE_ID_SUN_RIO_EBUS));
 
 	return pdev;
 }
@@ -637,6 +631,7 @@ void __init ebus_init(void)
 		ebus->is_rio = is_rio;
 		++num_ebus;
 	}
+	pci_dev_put(pdev); /* XXX for the case, when ebusnd is 0, is it OK? */
 
 #ifdef CONFIG_SUN_AUXIO
 	auxio_probe();
