From akpm@osdl.org Tue Oct 10 14:39:23 2006
Message-Id: <200610102139.k9ALd1F2025273@shell0.pdx.osdl.net>
From: Alan Cox <alan@lxorguk.ukuu.org.uk>
Subject: PCI: quirks: switch quirks code offender to use pci_get API
To: greg@kroah.com
Cc: akpm@osdl.org, alan@lxorguk.ukuu.org.uk, alan@redhat.com
Date: Tue, 10 Oct 2006 14:39:00 -0700

From: Alan Cox <alan@lxorguk.ukuu.org.uk>

Signed-off-by: Alan Cox <alan@redhat.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/quirks.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/pci/quirks.c
+++ gregkh-2.6/drivers/pci/quirks.c
@@ -1840,7 +1840,7 @@ static void __devinit quirk_nvidia_ck804
 	/* check HT MSI cap on this chipset and the root one.
 	 * a single one having MSI is enough to be sure that MSI are supported.
 	 */
-	pdev = pci_find_slot(dev->bus->number, 0);
+	pdev = pci_get_slot(dev->bus, 0);
 	if (dev->subordinate && !msi_ht_cap_enabled(dev)
 	    && !msi_ht_cap_enabled(pdev)) {
 		printk(KERN_WARNING "PCI: MSI quirk detected. "
@@ -1848,6 +1848,7 @@ static void __devinit quirk_nvidia_ck804
 		       pci_name(dev));
 		dev->subordinate->bus_flags |= PCI_BUS_FLAGS_NO_MSI;
 	}
+	pci_dev_put(pdev);
 }
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_NVIDIA, PCI_DEVICE_ID_NVIDIA_CK804_PCIE,
 			quirk_nvidia_ck804_msi_ht_cap);
