From eike-kernel@sf-tec.de Tue Jul 10 04:35:14 2007
From: Rolf Eike Beer <eike-kernel@sf-tec.de>
Date: Tue, 10 Jul 2007 13:35:05 +0200
Subject: PCI: Fix typo in include/linux/pci.h
To: linux-kernel@vger.kernel.org
Cc: Andrew Morton <akpm@linux-foundation.org>, trivial@kernel.org, Greg KH <greg@kroah.com>
Message-ID: <200707101335.05637.eike-kernel@sf-tec.de>
Content-Disposition: inline


Signed-off-by: Rolf Eike Beer <eike-kernel@sf-tec.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/pci.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -315,7 +315,7 @@ struct pci_dynids {
 
 /* ---------------------------------------------------------------- */
 /** PCI Error Recovery System (PCI-ERS).  If a PCI device driver provides
- *  a set fof callbacks in struct pci_error_handlers, then that device driver
+ *  a set of callbacks in struct pci_error_handlers, then that device driver
  *  will be notified of PCI bus errors, and will be driven to recovery
  *  when an error occurs.
  */
