From johnpol@2ka.mipt.ru Thu Mar 23 08:00:09 2006
Cc: Evgeniy Polyakov <johnpol@2ka.mipt.ru>
Subject: [PATCH 4/5] w1: Move w1-connector definitions into linux/include/connector.h
Date: Thu, 23 Mar 2006 19:11:58 +0300
Message-Id: <11431303183514@2ka.mipt.ru>
To: GregKH <greg@kroah.com>
From: Evgeniy Polyakov <johnpol@2ka.mipt.ru>


Signed-off-by: Evgeniy Polyakov <johnpol@2ka.mipt.ru>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/w1/w1_netlink.h   |    3 ---
 include/linux/connector.h |    5 ++++-
 2 files changed, 4 insertions(+), 4 deletions(-)

--- gregkh-2.6.orig/drivers/w1/w1_netlink.h
+++ gregkh-2.6/drivers/w1/w1_netlink.h
@@ -27,9 +27,6 @@
 
 #include "w1.h"
 
-#define CN_W1_IDX	3
-#define CN_W1_VAL	1
-
 enum w1_netlink_message_types {
 	W1_SLAVE_ADD = 0,
 	W1_SLAVE_REMOVE,
--- gregkh-2.6.orig/include/linux/connector.h
+++ gregkh-2.6/include/linux/connector.h
@@ -34,8 +34,11 @@
 #define CN_VAL_PROC			0x1
 #define CN_IDX_CIFS			0x2
 #define CN_VAL_CIFS                     0x1
+#define CN_W1_IDX			0x3	/* w1 communication */
+#define CN_W1_VAL			0x1
 
-#define CN_NETLINK_USERS		1
+
+#define CN_NETLINK_USERS		4
 
 /*
  * Maximum connector's message size.
