From kay.sievers@novell.com Sat Oct  7 13:11:36 2006
From: Kay Sievers <kay.sievers@novell.com>
Date: Sat, 07 Oct 2006 21:54:55 +0200
Subject: Driver core: swich input_device to device_type
Message-Id: <1160250895.4235.23.camel@localhost>


Signed-off-by: Kay Sievers <kay.sievers@novell.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/input/input.c |   16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

--- gregkh-2.6.orig/drivers/input/input.c
+++ gregkh-2.6/drivers/input/input.c
@@ -863,16 +863,10 @@ static int input_add_uevent_modalias_var
 static int input_dev_uevent(struct device *d, char **envp,
 			    int num_envp, char *buffer, int buffer_size)
 {
-	struct input_dev *dev;
+	struct input_dev *dev = to_input_dev(d);
 	int i = 0;
 	int len = 0;
 
-	/* input is a single class, this is only valid for input_dev's */
-	if (strncmp("input", kobject_name(&d->kobj), 5) != 0)
-		return 0;
-
-	dev = to_input_dev(d);
-
 	INPUT_ADD_HOTPLUG_VAR("PRODUCT=%x/%x/%x/%x",
 				dev->id.bustype, dev->id.vendor,
 				dev->id.product, dev->id.version);
@@ -908,10 +902,13 @@ static int input_dev_uevent(struct devic
 	return 0;
 }
 
+struct device_type input_dev_type = {
+	.uevent 		= input_dev_uevent,
+	.release 		= input_dev_release,
+};
+
 struct class input_class = {
 	.name			= "input",
-	.dev_release		= input_dev_release,
-	.dev_uevent		= input_dev_uevent,
 };
 EXPORT_SYMBOL_GPL(input_class);
 
@@ -932,6 +929,7 @@ struct input_dev *input_allocate_device(
 	if (dev) {
 		device_initialize(&dev->d);
 		dev->d.class = &input_class;
+		dev->d.type = &input_dev_type;
 		mutex_init(&dev->mutex);
 		INIT_LIST_HEAD(&dev->h_list);
 		INIT_LIST_HEAD(&dev->node);
