---
 Documentation/feature-removal-schedule/README |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

--- /dev/null
+++ b/Documentation/feature-removal-schedule/README
@@ -0,0 +1,16 @@
+The files in this directory describe a number of files and features
+that are that are going to be removed in the kernel source tree.  Every
+entry should contain what exactly is going away, why it is happening,
+and who is going to be doing the work.  When the feature is removed from
+the kernel, the file describing the removal should also be removed.
+
+The files should use the following template to help explain this
+information:
+
+---------------------------
+What:	the foo->baz field is going away
+When:	October 31, 2038
+Why:	Using this field is broken by design, please switch to using the
+	newer foo->bar field instead.  For more details on this, please
+	see the directions in Documentation/foo/using_bar.txt
+Who:	Joe Weezle <joe@weezle>
