From lcapitulino@mandriva.com.br Thu Oct 26 09:03:38 2006
Message-Id: <20061026160316.398145311@mandriva.com.br>
Date: Thu, 26 Oct 2006 13:02:45 -0300
From: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net
Subject: USB: aircable: Use usb_endpoint_* functions
Content-Disposition: inline; filename=aircable-use-usb-endpoint-functions.patch

Signed-off-by: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/aircable.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- gregkh-2.6.orig/drivers/usb/serial/aircable.c
+++ gregkh-2.6/drivers/usb/serial/aircable.c
@@ -305,9 +305,7 @@ static int aircable_probe(struct usb_ser
 
 	for (i = 0; i < iface_desc->desc.bNumEndpoints; i++) {
 		endpoint = &iface_desc->endpoint[i].desc;
-		if (((endpoint->bEndpointAddress & 0x80) == 0x00) &&
-			((endpoint->bmAttributes & 3) == 0x02)) {
-			/* we found our bulk out endpoint */
+		if (usb_endpoint_is_bulk_out(endpoint)) {
 			dbg("found bulk out on endpoint %d", i);
 			++num_bulk_out;
 		}
