From alan@lxorguk.ukuu.org.uk Wed Aug 22 14:58:46 2007
From: Alan Cox <alan@lxorguk.ukuu.org.uk>
Date: Wed, 22 Aug 2007 23:06:48 +0100
Subject: USB: cp2101: Coding style police
To: greg@kroah.com
Message-ID: <20070822230648.7a9d65da@the-village.bc.nu>


Signed-off-by: Alan Cox <alan@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/cp2101.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/cp2101.c
+++ b/drivers/usb/serial/cp2101.c
@@ -521,7 +521,7 @@ static void cp2101_set_termios (struct u
 
 	dbg("%s - port %d", __FUNCTION__, port->number);
 
-	if ((!port->tty) || (!port->tty->termios)) {
+	if (!port->tty || !port->tty->termios) {
 		dbg("%s - no tty structures", __FUNCTION__);
 		return;
 	}
