From linux-usb-owner@vger.kernel.org Sun Mar  9 17:45:25 2008
From: "Robert P. J. Day" <rpjday@crashcourse.ca>
Date: Sun, 9 Mar 2008 20:44:52 -0400 (EDT)
Subject: USB: Fix cut-and-paste error in rtl8150.c
To: Linux USB mailing list <linux-usb@vger.kernel.org>
Cc: petkan@users.sourceforge.net
Message-ID: <alpine.LFD.1.00.0803092043330.4597@localhost.localdomain>

Signed-off-by: Robert P. J. Day <rpjday@crashcourse.ca>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/net/usb/rtl8150.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/usb/rtl8150.c
+++ b/drivers/net/usb/rtl8150.c
@@ -376,7 +376,7 @@ static int alloc_all_urbs(rtl8150_t * de
 		return 0;
 	}
 	dev->ctrl_urb = usb_alloc_urb(0, GFP_KERNEL);
-	if (!dev->intr_urb) {
+	if (!dev->ctrl_urb) {
 		usb_free_urb(dev->rx_urb);
 		usb_free_urb(dev->tx_urb);
 		usb_free_urb(dev->intr_urb);
