From wfp5p@viridian.itc.virginia.edu  Tue May  4 13:00:13 2010
From: wfp5p@viridian.itc.virginia.edu (Bill Pemberton)
Date: Thu, 29 Apr 2010 10:04:56 -0400 (EDT)
Subject: usb: u132-hcd.c: fix shadows sparse warning
To: sshtylyov@mvista.com (Sergei Shtylyov)
Cc: gregkh@suse.de, linux-usb@vger.kernel.org
Message-ID: <20100429140456.9794140778@viridian.itc.virginia.edu>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/u132-hcd.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/host/u132-hcd.c
+++ b/drivers/usb/host/u132-hcd.c
@@ -1446,7 +1446,7 @@ static void u132_hcd_endp_work_scheduler
 			return;
 		} else {
 			int retval;
-			u8 address = u132->addr[endp->usb_addr].address;
+			address = u132->addr[endp->usb_addr].address;
 			struct urb *urb = endp->urb_list[ENDP_QUEUE_MASK &
 				endp->queue_next];
 			endp->active = 1;
@@ -3120,8 +3120,8 @@ static int __devinit u132_probe(struct p
 		ftdi_elan_gone_away(pdev);
 		return -ENOMEM;
 	} else {
-		int retval = 0;
 		struct u132 *u132 = hcd_to_u132(hcd);
+		retval = 0;
 		hcd->rsrc_start = 0;
 		mutex_lock(&u132_module_lock);
 		list_add_tail(&u132->u132_list, &u132_static_list);
