From akpm@osdl.org Tue Oct 10 14:50:44 2006
Message-Id: <200610102142.k9ALgtsa025533@shell0.pdx.osdl.net>
From: Alexey Dobriyan <adobriyan@gmail.com>
Subject: USB: net: use BUILD_BUG_ON
To: greg@kroah.com
Cc: akpm@osdl.org, adobriyan@gmail.com
Date: Tue, 10 Oct 2006 14:42:54 -0700

From: Alexey Dobriyan <adobriyan@gmail.com>

Signed-off-by: Alexey Dobriyan <adobriyan@gmail.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

diff -puN drivers/usb/net/cdc_ether.c~drivers-usb-net-use-build_bug_on drivers/usb/net/cdc_ether.c
--- a/drivers/usb/net/cdc_ether.c~drivers-usb-net-use-build_bug_on
+++ a/drivers/usb/net/cdc_ether.c
@@ -498,7 +498,7 @@ static struct usb_driver cdc_driver = {
 
 static int __init cdc_init(void)
 {
-	BUG_ON((sizeof(((struct usbnet *)0)->data)
+	BUILD_BUG_ON((sizeof(((struct usbnet *)0)->data)
 			< sizeof(struct cdc_state)));
 
  	return usb_register(&cdc_driver);
diff -puN drivers/usb/net/usbnet.c~drivers-usb-net-use-build_bug_on drivers/usb/net/usbnet.c
--- a/drivers/usb/net/usbnet.c~drivers-usb-net-use-build_bug_on
+++ a/drivers/usb/net/usbnet.c
@@ -1225,7 +1225,7 @@ EXPORT_SYMBOL_GPL(usbnet_resume);
 static int __init usbnet_init(void)
 {
 	/* compiler should optimize this out */
-	BUG_ON (sizeof (((struct sk_buff *)0)->cb)
+	BUILD_BUG_ON (sizeof (((struct sk_buff *)0)->cb)
 			< sizeof (struct skb_data));
 
 	random_ether_addr(node_id);
_

