From lcapitulino@mandriva.com.br Tue Jul 25 12:58:49 2006
Date: Tue, 25 Jul 2006 16:58:30 -0300
From: "Luiz Fernando N. Capitulino" <lcapitulino@mandriva.com.br>
To: Greg KH <gregkh@suse.de>
Subject: USB: pl2303: Removes unneeded goto.
Message-ID: <20060725165830.15b06627@doriath.conectiva>


Signed-off-by: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/pl2303.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- gregkh-2.6.orig/drivers/usb/serial/pl2303.c
+++ gregkh-2.6/drivers/usb/serial/pl2303.c
@@ -824,16 +824,13 @@ static void pl2303_update_line_status(st
 	}
 
 	if (actual_length < length)
-		goto exit;
+		return;
 
         /* Save off the uart status for others to look at */
 	spin_lock_irqsave(&priv->lock, flags);
 	priv->line_status = data[status_idx];
 	spin_unlock_irqrestore(&priv->lock, flags);
 	wake_up_interruptible (&priv->delta_msr_wait);
-
-exit:
-	return;
 }
 
 static void pl2303_read_int_callback (struct urb *urb, struct pt_regs *regs)
