From joe@perches.com  Tue Jun 15 12:03:38 2010
From: Joe Perches <joe@perches.com>
Date: Mon, 31 May 2010 20:23:19 -0700
Subject: USB: isd200.c: Remove unnecessary kmalloc cast
To: linux-kernel@vger.kernel.org
Cc: Matthew Dharm <mdharm-usb@one-eyed-alien.net>, linux-usb@vger.kernel.org, usb-storage@lists.one-eyed-alien.net
Message-ID: <75c229f6bedef9b1924ef1d7daaf4df383e06fa7.1275360951.git.joe@perches.com>


Signed-off-by: Joe Perches <joe@perches.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/isd200.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/usb/storage/isd200.c
+++ b/drivers/usb/storage/isd200.c
@@ -1456,8 +1456,7 @@ static int isd200_init_info(struct us_da
 	int retStatus = ISD200_GOOD;
 	struct isd200_info *info;
 
-	info = (struct isd200_info *)
-			kzalloc(sizeof(struct isd200_info), GFP_KERNEL);
+	info = kzalloc(sizeof(struct isd200_info), GFP_KERNEL);
 	if (!info)
 		retStatus = ISD200_ERROR;
 	else {
