From cascardo@holoscopio.com  Wed Jan 27 14:33:31 2010
From: Thadeu Lima de Souza Cascardo <cascardo@holoscopio.com>
Date: Sun, 17 Jan 2010 19:24:03 -0200
Subject: USB: trivial: missing newline in usb core warning message
To: trivial@kernel.org
Cc: Thadeu Lima de Souza Cascardo <cascardo@holoscopio.com>, Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <1263763443-9613-1-git-send-email-cascardo@holoscopio.com>


Signed-off-by: Thadeu Lima de Souza Cascardo <cascardo@holoscopio.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/message.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/core/message.c
+++ b/drivers/usb/core/message.c
@@ -1316,7 +1316,7 @@ int usb_set_interface(struct usb_device 
 
 	alt = usb_altnum_to_altsetting(iface, alternate);
 	if (!alt) {
-		dev_warn(&dev->dev, "selecting invalid altsetting %d",
+		dev_warn(&dev->dev, "selecting invalid altsetting %d\n",
 			 alternate);
 		return -EINVAL;
 	}
