From mhund@ld-didactic.de Thu Feb  2 00:37:00 2006
Message-ID: <43E1C4AE.6090801@ld-didactic.de>
Date: Thu, 02 Feb 2006 09:37:02 +0100
From: Michael Hund <mhund@ld-didactic.de>
To: gregkh@suse.de, "Hund, Dr. Michael" <MHund@ld-didactic.de>
Subject: USB: change ldusb's experimental state

Signed-off-by: Michael Hund <mhund@ld-didactic.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/misc/Kconfig
+++ gregkh-2.6/drivers/usb/misc/Kconfig
@@ -141,7 +141,7 @@ source "drivers/usb/misc/sisusbvga/Kconf
 
 config USB_LD
 	tristate "USB LD driver"
-	depends on USB && EXPERIMENTAL
+	depends on USB
 	help
 	  This driver is for generic USB devices that use interrupt transfers,
 	  like LD Didactic's USB devices.
