From shurik@gwu.edu  Mon Mar 23 16:35:39 2009
From: Alexander Shumakovitch <shurik@gwu.edu>
Date: Sat, 21 Mar 2009 00:50:16 -0400
Subject: USB: qcserial: add device id for HP devices
To: Greg KH <gregkh@suse.de>
Message-ID: <20090321045016.GA11112@gwu.edu>


From: Alexander Shumakovitch <shurik@gwu.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/qcserial.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/serial/qcserial.c
+++ b/drivers/usb/serial/qcserial.c
@@ -24,6 +24,8 @@ static int debug;
 static struct usb_device_id id_table[] = {
 	{USB_DEVICE(0x05c6, 0x9211)},	/* Acer Gobi QDL device */
 	{USB_DEVICE(0x05c6, 0x9212)},	/* Acer Gobi Modem Device */
+	{USB_DEVICE(0x03f0, 0x1f1d)},	/* HP un2400 Gobi Modem Device */
+	{USB_DEVICE(0x03f0, 0x201d)},	/* HP un2400 Gobi QDL Device */
 	{ }				/* Terminating entry */
 };
 MODULE_DEVICE_TABLE(usb, id_table);
