From linux-usb-owner@vger.kernel.org Mon Feb  4 23:58:14 2008
From: Andrew Morton <akpm@linux-foundation.org>
Date: Mon, 04 Feb 2008 23:57:48 -0800
Subject: USB: usb-ohci-sm501-driver: use the conventional convention for suspend and resume
To: greg@kroah.com
Cc: linux-usb@vger.kernel.org, akpm@linux-foundation.org, damm@igel.co.jp, stern@rowland.harvard.edu
Message-ID: <200802050757.m157vTnM010929@imap1.linux-foundation.org>


From: Andrew Morton <akpm@linux-foundation.org>

Cc: Alan Stern <stern@rowland.harvard.edu>
Cc: Magnus Damm <damm@igel.co.jp>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ohci-sm501.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/usb/host/ohci-sm501.c
+++ b/drivers/usb/host/ohci-sm501.c
@@ -242,6 +242,9 @@ static int ohci_sm501_resume(struct plat
 	usb_hcd_resume_root_hub(platform_get_drvdata(pdev));
 	return 0;
 }
+#else
+#define ohci_sm501_suspend NULL
+#define ohci_sm501_resume NULL
 #endif
 
 /*-------------------------------------------------------------------------*/
@@ -253,10 +256,8 @@ static struct platform_driver ohci_hcd_s
 	.probe		= ohci_hcd_sm501_drv_probe,
 	.remove		= ohci_hcd_sm501_drv_remove,
 	.shutdown	= usb_hcd_platform_shutdown,
-#ifdef	CONFIG_PM
 	.suspend	= ohci_sm501_suspend,
 	.resume		= ohci_sm501_resume,
-#endif
 	.driver		= {
 		.owner	= THIS_MODULE,
 		.name	= "sm501-usb",
