From SRS0=HjXU=BM=fh-wolfenbuettel.de=o.bock@srs.kundenserver.de Thu Jul 27 12:40:09 2006
From: Oliver Bock <o.bock@fh-wolfenbuettel.de>
To: Greg KH <greg@kroah.com>
Subject: USB: cypress driver comment updates
Date: Thu, 27 Jul 2006 21:34:58 +0200
Content-Disposition: inline
Message-Id: <200607272134.58954.o.bock@fh-wolfenbuettel.de>


From: Oliver Bock <o.bock@fh-wolfenbuettel.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/misc/cypress_cy7c63.c |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/misc/cypress_cy7c63.c
+++ gregkh-2.6/drivers/usb/misc/cypress_cy7c63.c
@@ -12,8 +12,13 @@
 *	the single I/O ports of the device.
 *
 *	Supported vendors:	AK Modul-Bus Computer GmbH
-*	Supported devices:	CY7C63001A-PC (to be continued...)
-*	Supported functions:	Read/Write Ports (to be continued...)
+*				(Firmware "Port-Chip")
+*
+*	Supported devices:	CY7C63001A-PC
+*				CY7C63001C-PXC
+*				CY7C63001C-SXC
+*
+*	Supported functions:	Read/Write Ports
 *
 *
 *	This program is free software; you can redistribute it and/or
