From harvey.harrison@gmail.com  Fri Jun  6 15:16:54 2008
From: Harvey Harrison <harvey.harrison@gmail.com>
Date: Fri, 30 May 2008 10:18:53 -0700
Subject: USB: digi_accelport.c trivial sparse lock annotation
To: Greg KH <gregkh@suse.de>
Cc: LKML <linux-kernel@vger.kernel.org>, Al Viro <viro@ZenIV.linux.org.uk>
Message-ID: <1212167934.28403.196.camel@brick>


Signed-off-by: Harvey Harrison <harvey.harrison@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/digi_acceleport.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/serial/digi_acceleport.c
+++ b/drivers/usb/serial/digi_acceleport.c
@@ -571,6 +571,7 @@ static struct usb_serial_driver digi_acc
 static long cond_wait_interruptible_timeout_irqrestore(
 	wait_queue_head_t *q, long timeout,
 	spinlock_t *lock, unsigned long flags)
+__releases(lock)
 {
 	DEFINE_WAIT(wait);
 
