From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: USB: disable tasklet if rtl8150 device is removed while active.

This fixes a oops seen in the wild.


Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


--- gregkh-2.6.orig/drivers/usb/net/rtl8150.c	2005-09-28 13:48:41.000000000 -0700
+++ gregkh-2.6/drivers/usb/net/rtl8150.c	2005-09-28 14:28:25.000000000 -0700
@@ -909,6 +909,7 @@
 	usb_set_intfdata(intf, NULL);
 	if (dev) {
 		set_bit(RTL8150_UNPLUG, &dev->flags);
+		tasklet_disable(&dev->tl);
 		unregister_netdev(dev->netdev);
 		unlink_all_urbs(dev);
 		free_all_urbs(dev);
