From wfp5p@virginia.edu  Tue May  4 12:59:23 2010
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Wed, 28 Apr 2010 16:59:35 -0400
Subject: USB: oti6858: declare send_data() as static
To: gregkh@suse.de
Cc: linux-usb@vger.kernel.org
Message-ID: <1272488378-29623-5-git-send-email-wfp5p@virginia.edu>


Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/oti6858.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/oti6858.c
+++ b/drivers/usb/serial/oti6858.c
@@ -295,7 +295,7 @@ static void setup_line(struct work_struc
 	}
 }
 
-void send_data(struct work_struct *work)
+static void send_data(struct work_struct *work)
 {
 	struct oti6858_private *priv = container_of(work,
 			struct oti6858_private, delayed_write_work.work);
