From Olivier.Bornet@puck.ch  Thu Aug 20 14:33:00 2009
From: Olivier Bornet <Olivier.Bornet@puck.ch>
Date: Tue, 18 Aug 2009 21:05:56 +0200
Subject: USB: iuu_phoenix: increment version number
To: Greg KH <greg@kroah.com>
Cc: Alan Cox <alan@lxorguk.ukuu.org.uk>, linux-usb@vger.kernel.org, Olivier Bornet <Olivier.Bornet@puck.ch>
Message-ID: <1250622357-29765-5-git-send-email-Olivier.Bornet@puck.ch>


Signed-off-by: Olivier Bornet <Olivier.Bornet@puck.ch>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/serial/iuu_phoenix.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/iuu_phoenix.c
+++ b/drivers/usb/serial/iuu_phoenix.c
@@ -40,7 +40,7 @@ static int debug;
 /*
  * Version Information
  */
-#define DRIVER_VERSION "v0.10"
+#define DRIVER_VERSION "v0.11"
 #define DRIVER_DESC "Infinity USB Unlimited Phoenix driver"
 
 static struct usb_device_id id_table[] = {
