From linux-usb-devel-admin@lists.sourceforge.net Tue Feb  7 00:07:07 2006
Message-ID: <cda58cb80602070006k29e9ea67v@mail.gmail.com>
From: Franck Bui-Huu <vagabon.xyz@gmail.com>
Cc: David Brownell <david-b@pacbell.net>, Alan Stern <stern@rowland.harvard.edu>
Content-Disposition: inline
Subject: USB: Zero driver: Removed duplicated code
Date: Tue, 7 Feb 2006 09:06:17 +0100

Signed-off-by: Franck Bui-Huu <vagabon.xyz@gmail.com>


---
 drivers/usb/gadget/zero.c |    6 ------
 1 file changed, 6 deletions(-)

--- gregkh-2.6.orig/drivers/usb/gadget/zero.c
+++ gregkh-2.6/drivers/usb/gadget/zero.c
@@ -1224,12 +1224,6 @@ autoconf_fail:
 		loopback_config.bmAttributes |= USB_CONFIG_ATT_WAKEUP;
 	}
 
-	if (gadget->is_otg) {
-		otg_descriptor.bmAttributes |= USB_OTG_HNP,
-		source_sink_config.bmAttributes |= USB_CONFIG_ATT_WAKEUP;
-		loopback_config.bmAttributes |= USB_CONFIG_ATT_WAKEUP;
-	}
-
 	usb_gadget_set_selfpowered (gadget);
 
 	init_timer (&dev->resume);
