From 359d031d60ddedb44459f08b7e95d1d630dd2933 Mon Sep 17 00:00:00 2001
From: Matt Colyer <matt@colyer.name>
Date: Tue, 28 Aug 2007 23:33:52 -0700
Subject: USB: Add iPhone device id to the quirk list.


---
 drivers/usb/core/quirks.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/usb/core/quirks.c
+++ b/drivers/usb/core/quirks.c
@@ -87,6 +87,9 @@ static const struct usb_device_id usb_qu
 	{ USB_DEVICE(0x0fca, 0x0004), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
 	{ USB_DEVICE(0x0fca, 0x0006), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
 
+	/* Apple iPhone */
+	{ USB_DEVICE(0x05ac, 0x1290), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
+
 	{ }  /* terminating entry must be last */
 };
 
