From kay.sievers@vrfy.org  Wed Nov 12 11:28:09 2008
From: Kay Sievers <kay.sievers@vrfy.org>
Date: Thu, 30 Oct 2008 01:09:37 +0100
Subject: USB: send "change" event when a device changes its configuration
To: Alan Stern <stern@rowland.harvard.edu>
Cc: Greg KH <greg@kroah.com>, USB list <linux-usb@vger.kernel.org>
Message-ID: <1225325377.5970.7.camel@nga.site>

From: Kay Sievers <kay.sievers@vrfy.org>

Signed-off-by: Kay Sievers <kay.sievers@vrfy.org>
Acked-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/message.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/usb/core/message.c
+++ b/drivers/usb/core/message.c
@@ -1571,6 +1571,10 @@ free_interfaces:
 	}
 
 	dev->actconfig = cp;
+
+	/* tell userspace that the device has changed its configuration */
+	kobject_uevent(&dev->dev.kobj, KOBJ_CHANGE);
+
 	if (!cp) {
 		usb_set_device_state(dev, USB_STATE_ADDRESS);
 		usb_autosuspend_device(dev);
