From linux-usb-devel-bounces@lists.sourceforge.net Sun Oct  8 22:56:53 2006
Message-ID: <4529E498.3060408@ipom.com>
From: Tobias Lorenz <tobias.lorenz@gmx.net>
Date: Sun, 08 Oct 2006 22:56:40 -0700
To: Tobias Lorenz <tobias.lorenz@gmx.net>
Cc: Alan Stern <stern@rowland.harvard.edu>, Pete Zaitcev <zaitcev@redhat.com>,
Subject: USB: Mitsumi USB FDD 061M: UNUSUAL_DEV multilun fix

From: Tobias Lorenz <tobias.lorenz@gmx.net>

From: Tobias Lorenz <tobias.lorenz@gmx.net>
Signed-off-by: Phil Dibowitz <phil@ipom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/unusual_devs.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/storage/unusual_devs.h
+++ gregkh-2.6/drivers/usb/storage/unusual_devs.h
@@ -55,7 +55,8 @@ UNUSUAL_DEV(  0x03eb, 0x2002, 0x0100, 0x
                 US_SC_DEVICE, US_PR_DEVICE, NULL,
                 US_FL_IGNORE_RESIDUE),
 
-UNUSUAL_DEV(  0x03ee, 0x6901, 0x0000, 0x0100,
+/* modified by Tobias Lorenz <tobias.lorenz@gmx.net> */
+UNUSUAL_DEV(  0x03ee, 0x6901, 0x0000, 0x0200,
 		"Mitsumi",
 		"USB FDD",
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
