From gadiyar@ti.com  Fri Nov 20 12:39:38 2009
From: Bryan Wu <cooloney@kernel.org>
Date: Mon, 16 Nov 2009 16:19:21 +0530
Subject: USB: musb: kill some useless comments in Blackfin driver
To: "linux-usb@vger.kernel.org; Greg KH" <greg@kroah.com>
Cc: Bryan Wu <cooloney@kernel.org>, Cliff Cai <cliff.cai@analog.com>, Mike Frysinger <vapier@gentoo.org>, Anand Gadiyar <gadiyar@ti.com>
Message-ID: <1258368569-4323-3-git-send-email-gadiyar@ti.com>


From: Bryan Wu <cooloney@kernel.org>

Signed-off-by: Bryan Wu <cooloney@kernel.org>
Signed-off-by: Cliff Cai <cliff.cai@analog.com>
Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Anand Gadiyar <gadiyar@ti.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/blackfin.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/usb/musb/blackfin.c
+++ b/drivers/usb/musb/blackfin.c
@@ -262,10 +262,6 @@ int __init musb_platform_init(struct mus
 		SSYNC();
 	}
 
-	/* TODO
-	 * Set SIC-IVG register
-	 */
-
 	/* Configure PLL oscillator register */
 	bfin_write_USB_PLLOSC_CTRL(0x30a8);
 	SSYNC();
