From bunk@stusta.de Mon Oct  9 00:45:41 2006
Date: Mon, 9 Oct 2006 01:16:30 +0200
From: Adrian Bunk <bunk@stusta.de>
To: Paul B Schroeder <pschroeder@uplogix.com>
Cc: linux-kernel@vger.kernel.org, gregkh@suse.de, linux-usb-devel@lists.sourceforge.net
Subject: USB: mos7840.c: remove dead code
Message-ID: <20061008231630.GP6755@stusta.de>
Content-Disposition: inline

The Coverity checker spotted this dead code.

Signed-off-by: Adrian Bunk <bunk@stusta.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


--- linux-2.6/drivers/usb/serial/mos7840.c.old	2006-10-09 00:31:41.000000000 +0200
+++ linux-2.6/drivers/usb/serial/mos7840.c	2006-10-09 00:32:12.000000000 +0200
@@ -1420,7 +1420,6 @@
 	int i;
 	int bytes_sent = 0;
 	int transfer_size;
-	int from_user = 0;
 
 	struct moschip_port *mos7840_port;
 	struct usb_serial *serial;
@@ -1511,15 +1510,7 @@
 	}
 	transfer_size = min(count, URB_TRANSFER_BUFFER_SIZE);
 
-	if (from_user) {
-		if (copy_from_user
-		    (urb->transfer_buffer, current_position, transfer_size)) {
-			bytes_sent = -EFAULT;
-			goto exit;
-		}
-	} else {
-		memcpy(urb->transfer_buffer, current_position, transfer_size);
-	}
+	memcpy(urb->transfer_buffer, current_position, transfer_size);
 
 	/* fill urb with data and submit  */
 	usb_fill_bulk_urb(urb,

