From akpm@linux-foundation.org  Wed Aug 15 16:32:03 2007
From: M4rkusXXL <M4rkusXXL@web.de>
Date: Fri, 10 Aug 2007 14:53:32 -0700
Subject: usb: typo in usb R8A66597 HCD config
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net, akpm@linux-foundation.org, M4rkusXXL@web.de
Message-ID: <200708102153.l7ALrW7N011533@imap1.linux-foundation.org>


From: M4rkusXXL <M4rkusXXL@web.de>

Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/host/Kconfig
+++ b/drivers/usb/host/Kconfig
@@ -237,7 +237,7 @@ config USB_SL811_CS
 	  module will be called "sl811_cs".
 
 config USB_R8A66597_HCD
-	tristate "R8A66597 HCD suppoort"
+	tristate "R8A66597 HCD support"
 	depends on USB
 	help
 	  The R8A66597 is a USB 2.0 host and peripheral controller.
