From lcapitulino@conectiva.com.br Sat Aug  5 16:28:38 2006
From: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net, Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>
Subject: USB: acecad: Use usb_endpoint_* functions.
Date: Sáb, 05 Ago 2006 20:28:15 -0300
Message-Id: <11548205011242-git-send-email-lcapitulino@mandriva.com.br>

Signed-off-by: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/input/acecad.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- gregkh-2.6.orig/drivers/usb/input/acecad.c
+++ gregkh-2.6/drivers/usb/input/acecad.c
@@ -141,10 +141,7 @@ static int usb_acecad_probe(struct usb_i
 
 	endpoint = &interface->endpoint[0].desc;
 
-	if (!(endpoint->bEndpointAddress & 0x80))
-		return -ENODEV;
-
-	if ((endpoint->bmAttributes & 3) != 3)
+	if (!usb_endpoint_is_int_in(endpoint))
 		return -ENODEV;
 
 	pipe = usb_rcvintpipe(dev, endpoint->bEndpointAddress);
