From linux-usb-owner@vger.kernel.org Sun Mar  9 10:55:37 2008
From: "Robert P. J. Day" <rpjday@crashcourse.ca>
Date: Sun, 9 Mar 2008 13:55:01 -0400 (EDT)
Subject: USB: Fix "cut and paste" booboo in usbmon Makefile.
To: Linux USB mailing list <linux-usb@vger.kernel.org>
Message-ID: <alpine.LFD.1.00.0803091353540.26271@localhost.localdomain>

Signed-off-by: Robert P. J. Day <rpjday@crashcourse.ca>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/mon/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/mon/Makefile
+++ b/drivers/usb/mon/Makefile
@@ -1,5 +1,5 @@
 #
-# Makefile for USB Core files and filesystem
+# Makefile for USB monitor
 #
 
 usbmon-objs	:= mon_main.o mon_stat.o mon_text.o mon_bin.o mon_dma.o
