From linux-usb-owner@vger.kernel.org  Mon Sep 15 10:13:22 2008
From: Paulius Zaleckas <paulius.zaleckas@teltonika.lt>
Date: Wed, 10 Sep 2008 17:18:46 +0300
Subject: USB: ftdi_sio: fix 'product' parameter description
To: linux-usb@vger.kernel.org
Cc: Linux-kernel@vger.kernel.org
Message-ID: <48C7D746.9070100@teltonika.lt>


Signed-off-by: Paulius Zaleckas <paulius.zaleckas@teltonika.lt>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/ftdi_sio.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -2455,5 +2455,5 @@ module_param(vendor, ushort, 0);
 MODULE_PARM_DESC(vendor, "User specified vendor ID (default="
 		__MODULE_STRING(FTDI_VID)")");
 module_param(product, ushort, 0);
-MODULE_PARM_DESC(vendor, "User specified product ID");
+MODULE_PARM_DESC(product, "User specified product ID");
 
