From koen@handhelds.org Wed Oct 26 23:06:41 2005
Message-ID: <435FC01D.4090807@handhelds.org>
Date: Wed, 26 Oct 2005 19:42:53 +0200
From: Koen Kooi <koen@handhelds.org>
To: Greg KH <greg@kroah.com>
Subject: USB: fix correct wording in drivers/usb/net/KConfig

Signed-off-by: Koen Kooi <koen@handhelds.org>

---
 drivers/usb/net/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/net/Kconfig
+++ gregkh-2.6/drivers/usb/net/Kconfig
@@ -294,7 +294,7 @@ config USB_NET_ZAURUS
 	  This also supports some related device firmware, as used in some
 	  PDAs from Olympus and some cell phones from Motorola.
 
-	  If you install an alternate ROM image, such as the Linux 2.6 based
+	  If you install an alternate image, such as the Linux 2.6 based
 	  versions of OpenZaurus, you should no longer need to support this
 	  protocol.  Only the "eth-fd" or "net_fd" drivers in these devices
 	  really need this non-conformant variant of CDC Ethernet (or in
