From rpjday@crashcourse.ca  Fri Jan 15 10:33:01 2010
From: "Robert P. J. Day" <rpjday@crashcourse.ca>
Date: Mon, 28 Dec 2009 06:31:08 -0500 (EST)
Subject: USB: Correct CONFIG variable typo.
Cc: David Brownell <dbrownell@users.sourceforge.net>
Message-ID: <alpine.LFD.2.00.0912280630070.16302@localhost>



Signed-off-by: Robert P. J. Day <rpjday@crashcourse.ca>
Cc: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/ether.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/gadget/ether.c
+++ b/drivers/usb/gadget/ether.c
@@ -259,7 +259,7 @@ static struct usb_configuration rndis_co
 
 /*-------------------------------------------------------------------------*/
 
-#ifdef USB_ETH_EEM
+#ifdef CONFIG_USB_ETH_EEM
 static int use_eem = 1;
 #else
 static int use_eem;
