From ben@decadent.org.uk  Fri Jan 15 14:56:06 2010
From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 13 Jan 2010 23:39:43 +0000
Subject: USB: isight-firmware: declare MODULE_FIRMWARE
To: Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <1263425983.3852.125.camel@localhost>


Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/isight_firmware.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/misc/isight_firmware.c
+++ b/drivers/usb/misc/isight_firmware.c
@@ -112,6 +112,8 @@ out:
 	return ret;
 }
 
+MODULE_FIRMWARE("isight.fw");
+
 static void isight_firmware_disconnect(struct usb_interface *intf)
 {
 }
