From downey@zymeta.com Mon Apr  3 07:58:14 2006
Message-ID: <443137FF.4050803@zymeta.com>
Date: Mon, 03 Apr 2006 08:58:07 -0600
From: Michael Downey <downey@zymeta.com>
To: Greg KH <greg@kroah.com>
CC: Adrian Bunk <bunk@stusta.de>,
Subject: USB: keyspan-remote bugfix


Signed-off-by: Michael Downey <downey@zymeta.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/input/keyspan_remote.c |    2 ++
 1 file changed, 2 insertions(+)

--- gregkh-2.6.orig/drivers/usb/input/keyspan_remote.c
+++ gregkh-2.6/drivers/usb/input/keyspan_remote.c
@@ -297,6 +297,8 @@ static void keyspan_check_data(struct us
 			remote->data.bits_left -= 6;
 		} else {
 			err("%s - Error in message, invalid toggle.\n", __FUNCTION__);
+			remote->stage = 0;
+			return;
 		}
 
 		keyspan_load_tester(remote, 5);
