From daniel@centurion.net.nz Fri Nov 10 18:48:04 2006
Subject: USB: airprime: New device ID
From: <daniel@centurion.net.nz>
To: Greg KH <gregkh@suse.de>
Content-Type: text/plain
Date: Sat, 11 Nov 2006 15:47:52 +1300
Message-Id: <1163213272.6807.23.camel@localhost.localdomain>
Mime-Version: 1.0


Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/airprime.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/usb/serial/airprime.c
+++ gregkh-2.6/drivers/usb/serial/airprime.c
@@ -19,6 +19,7 @@
 static struct usb_device_id id_table [] = {
 	{ USB_DEVICE(0x0c88, 0x17da) }, /* Kyocera Wireless KPC650/Passport */
 	{ USB_DEVICE(0x1410, 0x1110) }, /* Novatel Wireless Merlin CDMA */
+	{ USB_DEVICE(0x1410, 0x1430) },	/* Novatel Merlin XU870 HSDPA/3G */
 	{ USB_DEVICE(0x1410, 0x1100) }, /* ExpressCard34 Qualcomm 3G CDMA */
 	{ },
 };
