From akpm@osdl.org Tue Jan 24 17:42:44 2006
Message-Id: <200601250142.k0P1gfXO031519@shell0.pdx.osdl.net>
From: Andrew Morton <akpm@osdl.org>
Subject: USB: fix ehci early handoff issues warning
To: akpm@osdl.org, david-b@pacbell.net, greg@kroah.com
From: akpm@osdl.org
Date: Tue, 24 Jan 2006 17:42:24 -0800


From: Andrew Morton <akpm@osdl.org>

Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/host/ehci-pci.c |    1 -
 1 file changed, 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/host/ehci-pci.c
+++ gregkh-2.6/drivers/usb/host/ehci-pci.c
@@ -29,7 +29,6 @@ static int ehci_pci_reinit(struct ehci_h
 {
 	u32			temp;
 	int			retval;
-	unsigned		count = 256/4;
 
 	/* optional debug port, normally in the first BAR */
 	temp = pci_find_capability(pdev, 0x0a);
