From akpm@linux-foundation.org  Thu Jun 19 11:54:33 2008
From: Andrew Morton <akpm@linux-foundation.org>
Date: Mon, 09 Jun 2008 16:39:52 -0700
Subject: USB: drivers/usb/host/isp1760-hcd.c: processor flags have type `unsigned long'
To: greg@kroah.com
Cc: linux-usb@vger.kernel.org, akpm@linux-foundation.org
Message-ID: <200806092339.m59NdqqZ015004@imap1.linux-foundation.org>


From: Andrew Morton <akpm@linux-foundation.org>

Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/isp1760-hcd.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/host/isp1760-hcd.c
+++ b/drivers/usb/host/isp1760-hcd.c
@@ -1592,7 +1592,7 @@ static int isp1760_urb_dequeue(struct us
 	struct inter_packet_info *ints;
 	u32 i;
 	u32 reg_base, or_reg, skip_reg;
-	int flags;
+	unsigned long flags;
 	struct ptd ptd;
 
 	switch (usb_pipetype(urb->pipe)) {
@@ -2061,7 +2061,7 @@ static void isp1760_endpoint_disable(str
 	struct isp1760_hcd *priv = hcd_to_priv(usb_hcd);
 	struct isp1760_qh *qh;
 	struct isp1760_qtd *qtd;
-	u32 flags;
+	unsigned long flags;
 
 	spin_lock_irqsave(&priv->lock, flags);
 	qh = ep->hcpriv;
