From linux-usb-owner@vger.kernel.org  Mon Sep 15 10:11:51 2008
From: Peter Korsgaard <jacmet@sunsite.dk>
Date: Mon,  8 Sep 2008 10:14:47 +0200
Subject: usb core: fix USB_OTG_BLACKLIST_HUB typo
To: gregkh@suse.de, rgetz@blackfin.uclinux.org, linux-usb@vger.kernel.org
Cc: Peter Korsgaard <jacmet@sunsite.dk>
Message-ID: <1220861687-15189-1-git-send-email-jacmet@sunsite.dk>


Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/core/Kconfig
+++ b/drivers/usb/core/Kconfig
@@ -134,5 +134,5 @@ config USB_OTG_BLACKLIST_HUB
 	  If you say Y here, then Linux will refuse to enumerate
 	  external hubs.  OTG hosts are allowed to reduce hardware
 	  and software costs by not supporting external hubs.  So
-	  are "Emedded Hosts" that don't offer OTG support.
+	  are "Embedded Hosts" that don't offer OTG support.
 
