From alon+git@nolaviz.org  Tue Oct 12 11:03:30 2010
From: Alon Ziv <alon+git@nolaviz.org>
Date: Sun, 10 Oct 2010 08:32:20 +0200
Subject: USB: opticon: Whitespace fixes in opticon driver
To: linux-usb@vger.kernel.org
Cc: linux-kernel@vger.kernel.org, Alon Ziv <alon+git@nolaviz.org>
Message-ID: <1286692340-18639-4-git-send-email-alon-git@nolaviz.org>


Signed-off-by: Alon Ziv <alon-git@nolaviz.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/opticon.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/usb/serial/opticon.c
+++ b/drivers/usb/serial/opticon.c
@@ -108,10 +108,10 @@ static void opticon_bulk_callback(struct
 				else
 					priv->rts = true;
 			} else {
-			dev_dbg(&priv->udev->dev,
-				"Unknown data packet received from the device:"
-				" %2x %2x\n",
-				data[0], data[1]);
+				dev_dbg(&priv->udev->dev,
+					"Unknown data packet received from the device:"
+					" %2x %2x\n",
+					data[0], data[1]);
 			}
 		}
 	} else {
