From oliver@neukum.org  Mon Oct 26 14:29:22 2009
From: Oliver Neukum <oliver@neukum.org>
Date: Thu, 24 Sep 2009 00:33:45 +0200
Subject: USB: usbtmc: minor formatting cleanups
To: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Cc: "Greg Kroah-Hartman" <gregkh@suse.de>, linux-usb@vger.kernel.org
Message-ID: <200909240033.46733.oliver@neukum.org>
Content-Disposition: inline


From: Oliver Neukum <oliver@neukum.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/usbtmc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/class/usbtmc.c
+++ b/drivers/usb/class/usbtmc.c
@@ -1109,13 +1109,13 @@ static void usbtmc_disconnect(struct usb
 	kref_put(&data->kref, usbtmc_delete);
 }
 
-static int usbtmc_suspend (struct usb_interface *intf, pm_message_t message)
+static int usbtmc_suspend(struct usb_interface *intf, pm_message_t message)
 {
 	/* this driver does not have pending URBs */
 	return 0;
 }
 
-static int usbtmc_resume (struct usb_interface *intf)
+static int usbtmc_resume(struct usb_interface *intf)
 {
 	return 0;
 }
