From mmporter@cox.net Tue Sep 20 00:44:21 2005
Date: Mon, 19 Sep 2005 09:40:24 -0700
From: Matt Porter <mporter@kernel.crashing.org>
To: greg@kroah.com, David Brownell <david-b@pacbell.net>
Subject: USB: Fix usb hub build
Message-ID: <20050919094024.C10118@cox.net>


Signed-off-by: Matt Porter <mporter@kernel.crashing.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

 drivers/usb/core/hub.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: gregkh-2.6/drivers/usb/core/hub.c
===================================================================
--- gregkh-2.6.orig/drivers/usb/core/hub.c	2005-09-20 08:25:57.000000000 -0700
+++ gregkh-2.6/drivers/usb/core/hub.c	2005-09-21 05:39:29.000000000 -0700
@@ -1977,7 +1977,7 @@
 
 int usb_resume_device(struct usb_device *udev)
 {
-	udev->dev.power_state.event = PM_EVENT_ON;
+	udev->dev.power.power_state.event = PM_EVENT_ON;
 	return 0;
 }
 
