From jeff@garzik.org  Mon Mar  5 10:31:31 2007
From: YOSHIFUJI Hideaki <yoshfuji@linux-ipv6.org>
Date: Fri, 02 Mar 2007 20:33:55 -0500
Subject: USBNET: DM9501: Add Corega FEther USB-TXC support.
To: YOSHIFUJI Hideaki / 吉藤英明 <yoshfuji@linux-ipv6.org>
Cc: Jeff Garzik <jeff@garzik.org>, <davem@davemloft.net>,  netdev@vger.kernel.org,  Greg KH <greg@kroah.com>
Message-ID: <45E8D083.3010907@garzik.org>

Signed-off-by: YOSHIFUJI Hideaki <yoshfuji@linux-ipv6.org>
Cc: Jeff Garzik <jeff@garzik.org>
Cc: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/net/dm9601.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/usb/net/dm9601.c
+++ b/drivers/usb/net/dm9601.c
@@ -571,6 +571,10 @@ static const struct driver_info dm9601_i
 
 static const struct usb_device_id products[] = {
 	{
+	 USB_DEVICE(0x07aa, 0x9601),	/* Corega FEther USB-TXC */
+	 .driver_info = (unsigned long)&dm9601_info,
+	 },
+	{
 	 USB_DEVICE(0x0a46, 0x9601),	/* Davicom USB-100 */
 	 .driver_info = (unsigned long)&dm9601_info,
 	 },
