From ben@decadent.org.uk Thu Jun 24 17:35:06 2010
From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 25 Jun 2010 01:35:01 +0100
Subject: Staging: rtl8192s_usb: Remove duplicate device ID
To: Greg KH <greg@kroah.com>
Cc: devel@driverdev.osuosl.org, Florian Schilhabel <florian.c.schilhabel@googlemail.com>, Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>, Guy Sheffer <guysoft@gmail.com>
Message-ID: <1277426101.26161.168.camel@localhost>


Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rtl8192su/r8192U_core.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/staging/rtl8192su/r8192U_core.c
+++ b/drivers/staging/rtl8192su/r8192U_core.c
@@ -134,7 +134,6 @@ static const struct usb_device_id rtl819
 	{USB_DEVICE(0x0E66, 0x0016)},
 	{USB_DEVICE(0x0b05, 0x1786)},
 	/* these are not in the official list */
-	{USB_DEVICE(0x050d, 0x815F)}, /* Belkin F5D8053 v6 */
 	{USB_DEVICE(0x0df6, 0x004b)}, /* WL-349 */
 	{}
 };
