From andrew@lunn.ch  Fri Dec 18 10:29:44 2009
From: Andrew Lunn <andrew@lunn.ch>
Date: Sat, 12 Dec 2009 23:39:47 +0100
Subject: Staging: batman-adv: Add Kconfig dependancies on PROC_FS and PACKET.
To: gregkh@suse.de
Cc: devel@linuxdriverproject.org, Andrew Lunn <andrew@lunn.ch>
Message-ID: <1260657587-1135-7-git-send-email-andrew@lunn.ch>


Signed-off-by: Andrew Lunn <andrew@lunn.ch>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/batman-adv/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/staging/batman-adv/Kconfig
+++ b/drivers/staging/batman-adv/Kconfig
@@ -4,6 +4,7 @@
 
 config BATMAN_ADV
 	tristate "B.A.T.M.A.N. Advanced Meshing Protocol"
+	depends on PROC_FS && PACKET
         default n
 	---help---
 
