From proski@gnu.org  Thu Apr 22 16:46:24 2010
From: Pavel Roskin <proski@gnu.org>
Date: Tue, 09 Mar 2010 23:11:07 -0500
Subject: Staging: rtl8192su: add USB ID for 0bda:8171
To: Greg KH <greg@kroah.com>
Cc: devel@driverdev.osuosl.org
Message-ID: <20100310041106.791.60267.stgit@mj.roinet.com>


Signed-off-by: Pavel Roskin <proski@gnu.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rtl8192su/r8192U_core.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/staging/rtl8192su/r8192U_core.c
+++ b/drivers/staging/rtl8192su/r8192U_core.c
@@ -113,6 +113,7 @@ u32 rt_global_debug_component = \
 
 static const struct usb_device_id rtl8192_usb_id_tbl[] = {
 	/* Realtek */
+	{USB_DEVICE(0x0bda, 0x8171)},
 	{USB_DEVICE(0x0bda, 0x8192)},
 	{USB_DEVICE(0x0bda, 0x8709)},
 	/* Corega */
