From vapier@gentoo.org  Tue Jun  1 13:36:07 2010
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 24 May 2010 00:55:42 -0400
Subject: Staging: adis16255: add proper section markings to hotplug funcs
To: devel@driverdev.osuosl.org, Greg Kroah-Hartman <gregkh@suse.de>
Cc: Matthias Brugger <mensch0815@gmail.com>
Message-ID: <1274676942-18557-2-git-send-email-vapier@gentoo.org>


Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/adis16255/adis16255.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/adis16255/adis16255.c
+++ b/drivers/staging/adis16255/adis16255.c
@@ -361,7 +361,7 @@ err:
 
 /*-------------------------------------------------------------------------*/
 
-static int spi_adis16255_probe(struct spi_device *spi)
+static int __devinit spi_adis16255_probe(struct spi_device *spi)
 {
 
 	struct adis16255_init_data *init_data = spi->dev.platform_data;
@@ -421,7 +421,7 @@ err:
 	return status;
 }
 
-static int spi_adis16255_remove(struct spi_device *spi)
+static int __devexit spi_adis16255_remove(struct spi_device *spi)
 {
 	struct spi_adis16255_data  *spiadis    = dev_get_drvdata(&spi->dev);
 
