From mbizon@freebox.fr  Mon Feb  1 14:30:29 2010
From: Maxime Bizon <mbizon@freebox.fr>
Date: Sat, 30 Jan 2010 18:42:56 +0100
Subject: serial: bcm63xx_uart: don't use kfree() on non kmalloced area.
To: Greg Kroah-Hartman <gregkh@suse.de>, linux-serial@vger.kernel.org
Cc: Ralf Baechle <ralf@linux-mips.org>, linux-mips@linux-mips.org, Maxime Bizon <mbizon@freebox.fr>
Message-ID: <1264873377-28479-2-git-send-email-mbizon@freebox.fr>


Signed-off-by: Maxime Bizon <mbizon@freebox.fr>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/bcm63xx_uart.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/serial/bcm63xx_uart.c
+++ b/drivers/serial/bcm63xx_uart.c
@@ -830,7 +830,7 @@ static int __devinit bcm_uart_probe(stru
 
 	ret = uart_add_one_port(&bcm_uart_driver, port);
 	if (ret) {
-		kfree(port);
+		ports[pdev->id].membase = 0;
 		return ret;
 	}
 	platform_set_drvdata(pdev, port);
