From vapier@gentoo.org  Wed Jan 27 14:44:37 2010
From: Mike Frysinger <vapier@gentoo.org>
Date: Tue, 19 Jan 2010 06:13:13 -0500
Subject: serial: bfin_5xx: pull in linux/io.h for ioremap prototypes
To: linux-serial@vger.kernel.org, Alan Cox <alan@lxorguk.ukuu.org.uk>
Cc: linux-kernel@vger.kernel.org, Andrew Morton <akpm@linux-foundation.org>
Message-ID: <1263899593-12178-4-git-send-email-vapier@gentoo.org>


Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/bfin_5xx.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/serial/bfin_5xx.c
+++ b/drivers/serial/bfin_5xx.c
@@ -14,6 +14,7 @@
 
 #include <linux/module.h>
 #include <linux/ioport.h>
+#include <linux/io.h>
 #include <linux/init.h>
 #include <linux/console.h>
 #include <linux/sysrq.h>
