From alan@linux.intel.com  Fri Sep 24 09:42:12 2010
From: Alan Cox <alan@linux.intel.com>
Subject: tty: Fix warning left over from TIOCGICOUNT changes
To: linux-next@vger.kernel.org, greg@kroah.com, linux-serial@vger.kernel.org
Date: Fri, 24 Sep 2010 15:49:18 +0100
Message-ID: <20100924144750.11385.24596.stgit@localhost.localdomain>

Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/ssu100.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/usb/serial/ssu100.c
+++ b/drivers/usb/serial/ssu100.c
@@ -444,8 +444,6 @@ static int ssu100_ioctl(struct tty_struc
 		    unsigned int cmd, unsigned long arg)
 {
 	struct usb_serial_port *port = tty->driver_data;
-	struct ssu100_port_private *priv = usb_get_serial_port_data(port);
-	void __user *user_arg = (void __user *)arg;
 
 	dbg("%s cmd 0x%04x", __func__, cmd);
 
