From randy.dunlap@oracle.com  Tue Apr 21 22:37:36 2009
From: Henrik Austad <henrik@austad.us>
Date: Mon, 20 Apr 2009 18:42:38 -0700
Subject: Doc/sysfs-rules: Swap the order of the words so the sentence makes more sense
Cc: gregkh@suse.de, Henrik Austad <henrik@austad.us>
Message-ID: <20090420184238.51cd7092.randy.dunlap@oracle.com>


From: Henrik Austad <henrik@austad.us>

Signed-off-by: Henrik Austad <henrik@austad.us>
Signed-off-by: Randy Dunlap <randy.dunlap@oracle.com>
Cc: Greg Kroah-Hartman <gregkh@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/sysfs-rules.txt |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Documentation/sysfs-rules.txt
+++ b/Documentation/sysfs-rules.txt
@@ -113,7 +113,7 @@ versions of the sysfs interface.
   "devices" directory at /sys/subsystem/<name>/devices.
 
   If /sys/subsystem exists, /sys/bus, /sys/class and /sys/block can be
-  ignored. If it does not exist, you have always to scan all three
+  ignored. If it does not exist, you always have to scan all three
   places, as the kernel is free to move a subsystem from one place to
   the other, as long as the devices are still reachable by the same
   subsystem name.
