From akpm@linux-foundation.org  Thu Jul 31 10:54:42 2008
From: Joe Perches <joe@perches.com>
Date: Wed, 30 Jul 2008 12:29:20 -0700
Subject: drivers/base/driver.c: remove unused to_dev() macro
To: greg@kroah.com
Cc: akpm@linux-foundation.org, joe@perches.com
Message-ID: <200807301929.m6UJTKip011923@imap1.linux-foundation.org>


From: Joe Perches <joe@perches.com>

Signed-off-by: Joe Perches <joe@perches.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/driver.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/base/driver.c
+++ b/drivers/base/driver.c
@@ -16,9 +16,6 @@
 #include <linux/string.h>
 #include "base.h"
 
-#define to_dev(node) container_of(node, struct device, driver_list)
-
-
 static struct device *next_device(struct klist_iter *i)
 {
 	struct klist_node *n = klist_next(i);
