From rpjday@crashcourse.ca  Tue Mar 16 13:35:44 2010
From: "Robert P. J. Day" <rpjday@crashcourse.ca>
Date: Fri, 12 Mar 2010 07:30:35 -0500 (EST)
Subject: kobject: documentation: Update to refer to kset-example.c.
Cc: Greg KH <greg@kroah.com>
Message-ID: <alpine.LFD.2.00.1003120728450.7261@localhost>



Signed-off-by: Robert P. J. Day <rpjday@crashcourse.ca>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/kobject.txt |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/Documentation/kobject.txt
+++ b/Documentation/kobject.txt
@@ -406,4 +406,5 @@ called, and the objects in the former ci
 Example code to copy from
 
 For a more complete example of using ksets and kobjects properly, see the
-sample/kobject/kset-example.c code.
+example programs samples/kobject/{kobject-example.c,kset-example.c},
+which will be built as loadable modules if you select CONFIG_SAMPLE_KOBJECT.
