From paolo.ciarrocchi@gmail.com  Tue Apr 25 14:58:08 2006
Message-ID: <4d8e3fd30604251347t4a3fb7b2id61496583553ff84@mail.gmail.com>
Date: Tue, 25 Apr 2006 22:47:51 +0200
From: "Paolo Ciarrocchi" <paolo.ciarrocchi@gmail.com>
To: "Greg KH" <greg@kroah.com>
Subject: Added URI of "linux kernel development process"
Content-Disposition: inline


Signed-off-by: Paolo Ciarrocchi <paolo.ciarrocchi@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/HOWTO |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/Documentation/HOWTO
+++ gregkh-2.6/Documentation/HOWTO
@@ -603,7 +603,8 @@ start exactly where you are now.
 
 
 ----------
-Thanks to Paolo Ciarrocchi who allowed the "Development Process" section
+Thanks to Paolo Ciarrocchi who allowed the "Development Process"
+(http://linux.tar.bz/articles/2.6-development_process) section
 to be based on text he had written, and to Randy Dunlap and Gerrit
 Huizenga for some of the list of things you should and should not say.
 Also thanks to Pat Mochel, Hanna Linder, Randy Dunlap, Kay Sievers,
