From foo@baz Tue Apr  9 12:12:43 2002
Date: Thu, 27 Apr 2006 15:46:39 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: CCISS: add device symlink to the block cciss block devices in sysfs


Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/block/cciss.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/block/cciss.c
+++ gregkh-2.6/drivers/block/cciss.c
@@ -3236,6 +3236,7 @@ static int __devinit cciss_init_one(stru
 		disk->fops = &cciss_fops;
 		disk->queue = q;
 		disk->private_data = drv;
+		disk->driverfs_dev = &pdev->dev;
 		/* we must register the controller even if no disks exist */
 		/* this is for the online array utilities */
 		if(!drv->heads && j)
