From bphilips@suse.de  Tue Jul 17 22:16:12 2007
From: Brandon Philips <bphilips@suse.de>
Date: Tue, 17 Jul 2007 22:09:34 -0700
Subject: Documentation fix devres.txt: lib/iomap.c -> lib/devres.c
To: Greg KH <gregkh@suse.de>, Tejun Heo <teheo@suse.de>, linux-kernel@vger.kernel.org
Message-ID: <20070718050934.GA4397@ifup.org>
Content-Disposition: inline


Signed-off-by: Brandon Philips <bphilips@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/driver-model/devres.txt |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Documentation/driver-model/devres.txt
+++ b/Documentation/driver-model/devres.txt
@@ -207,7 +207,7 @@ responsibility.  This is usually non-iss
 resource allocations already do the job.
 
 For an example of single-instance devres type, read pcim_iomap_table()
-in lib/iomap.c.
+in lib/devres.c.
 
 All devres interface functions can be called without context if the
 right gfp mask is given.
