From stefanr@s5r6.in-berlin.de Sun Jun 17 02:01:50 2007
From: Stefan Richter <stefanr@s5r6.in-berlin.de>
Date: Sun, 17 Jun 2007 11:01:18 +0200 (CEST)
Subject: driver core: properly get driver in device_release_driver
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: linux-kernel@vger.kernel.org
Message-ID: <tkrat.d01dbfd3bbd7140c@s5r6.in-berlin.de>
Content-Disposition: INLINE


Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/dd.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/base/dd.c
+++ b/drivers/base/dd.c
@@ -296,9 +296,8 @@ static void __device_release_driver(stru
 {
 	struct device_driver * drv;
 
-	drv = dev->driver;
+	drv = get_driver(dev->driver);
 	if (drv) {
-		get_driver(drv);
 		driver_sysfs_remove(dev);
 		sysfs_remove_link(&dev->kobj, "driver");
 		klist_remove(&dev->knode_driver);
