From akpm@osdl.org Fri Sep 23 00:21:12 2005
Message-Id: <200509230348.j8N3mYCf003244@shell0.pdx.osdl.net>
Subject: drivers/base/*: use kzalloc instead of kmalloc+memset (gregkh bits)
From: Jiri Slaby <jirislaby@gmail.com>
To: greg@kroah.com
Cc: akpm@osdl.org, jirislaby@gmail.com, webmaster@schiggl.de, xslaby@fi.muni.cz
Date: Thu, 22 Sep 2005 20:47:55 -0700


From: Jiri Slaby <jirislaby@gmail.com>

Signed-off-by: Lion Vollnhals <webmaster@schiggl.de>
Signed-off-by: Jiri Slaby <xslaby@fi.muni.cz>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


 drivers/base/class.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/base/class.c	2005-09-28 14:30:18.000000000 -0700
+++ gregkh-2.6/drivers/base/class.c	2005-09-28 15:36:37.000000000 -0700
@@ -527,13 +527,13 @@
 		class_device_create_file(class_dev, attr);
 		class_dev->devt_attr = attr;
 
-		attr = kmalloc(sizeof(*attr), GFP_KERNEL);
+		attr = kzalloc(sizeof(*attr), GFP_KERNEL);
 		if (!attr) {
 			error = -ENOMEM;
 			kobject_del(&class_dev->kobj);
 			goto register_done;
 		}
-		memset(attr, sizeof(*attr), 0x00);
+
 		attr->attr.name = "sample.sh";
 		attr->attr.mode = S_IRUSR | S_IXUSR | S_IRUGO;
 		attr->attr.owner = parent_class->owner;
