From k-keiichi@bx.jp.nec.com  Fri Oct 26 22:21:39 2007
From: Keiichi KII <k-keiichi@bx.jp.nec.com>
Date: Fri, 26 Oct 2007 15:51:44 +0900
Subject: fix typo in SubmittingPatches
Cc: gregkh@suse.de, Andy Whitcroft <apw@shadowen.org>
Message-ID: <47218E80.50700@bx.jp.nec.com>


From: Keiichi Kii <k-keiichi@bx.jp.nec.com>

Fix typo.

Signed-off-by: Keiichi Kii <k-keiichi@bx.jp.nec.com>
Cc: Andy Whitcroft <apw@shadowen.org>
Cc: Randy Dunlap <randy.dunlap@oracle.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/SubmittingPatches |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Documentation/SubmittingPatches
+++ b/Documentation/SubmittingPatches
@@ -464,7 +464,7 @@ section Linus Computer Science 101.
 Nuff said.  If your code deviates too much from this, it is likely
 to be rejected without further review, and without comment.
 
-Once significant exception is when moving code from one file to
+One significant exception is when moving code from one file to
 another in this case you should not modify the moved code at all in
 the same patch which moves it.  This clearly delineates the act of
 moving the code and your changes.  This greatly aids review of the
