 drivers/message/i2o/device.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/message/i2o/device.c	2005-09-26 16:53:51.000000000 -0700
+++ gregkh-2.6/drivers/message/i2o/device.c	2005-09-26 16:54:49.000000000 -0700
@@ -385,7 +385,7 @@
  *
  *	Returns 0 on success or negative error code on failure.
  */
-static int i2o_device_class_add(struct class_device *cd)
+static int i2o_device_class_add(struct class_device *cd, struct class_interface *intf)
 {
 	struct i2o_device *i2o_dev, *tmp;
 	struct i2o_controller *c;
