From jwboyer@gmail.com Mon Nov 21 05:52:34 2005
Message-ID: <625fc13d0511210612m7c659b9agb7f9a5e1a1de117f@mail.gmail.com>
Date: Mon, 21 Nov 2005 08:12:56 -0600
From: Josh Boyer <jwboyer@gmail.com>
To: Greg KH <greg@kroah.com>
Subject: MTD git tree location added to MAINTAINERS
Cc: Greg Kroah-Hartman <gregkh@suse.de>, Linus Torvalds <torvalds@osdl.org>, Andrew Morton <akpm@osdl.org>, scjody@steamballoon.com
Content-Disposition: inline

Here's the MTD one.  More later as I find them.

Signed-off-by: Josh Boyer <jwboyer@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

diff --git a/MAINTAINERS b/MAINTAINERS
index c5cf7d7..1e84be3 100644
---
 MAINTAINERS |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/MAINTAINERS
+++ gregkh-2.6/MAINTAINERS
@@ -1695,6 +1695,7 @@ P:	David Woodhouse
 M:	dwmw2@infradead.org
 W:	http://www.linux-mtd.infradead.org/
 L:	linux-mtd@lists.infradead.org
+T:	git kernel.org:/pub/scm/linux/kernel/git/tglx/mtd-2.6.git
 S:	Maintained
 
 MICROTEK X6 SCANNER
