From akpm@linux-foundation.org  Tue Aug 14 22:22:56 2007
From: Rolf Eike Beer <eike-kernel@sf-tec.de>
Date: Fri, 10 Aug 2007 13:51:07 -0700
Subject: sysfs: Fix typos in fs/sysfs/file.c
To: greg@kroah.com
Cc: akpm@linux-foundation.org, eike-kernel@sf-tec.de
Message-ID: <200708102051.l7AKp7ke008215@imap1.linux-foundation.org>


From: Rolf Eike Beer <eike-kernel@sf-tec.de>

Signed-off-by: Rolf Eike Beer <eike-kernel@sf-tec.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/sysfs/file.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/sysfs/file.c
+++ b/fs/sysfs/file.c
@@ -335,7 +335,7 @@ static int sysfs_release(struct inode * 
  * again will not get new data, or reset the state of 'poll'.
  * Reminder: this only works for attributes which actively support
  * it, and it is not possible to test an attribute from userspace
- * to see if it supports poll (Nether 'poll' or 'select' return
+ * to see if it supports poll (Neither 'poll' nor 'select' return
  * an appropriate error code).  When in doubt, set a suitable timeout value.
  */
 static unsigned int sysfs_poll(struct file *filp, poll_table *wait)
