From foo@baz Tue Apr  9 12:12:43 2002
Date: Tue, 09 Apr 2002 12:14:34 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Stupid patch for my laptop which can't get sysrq-u

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/char/sysrq.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/char/sysrq.c
+++ gregkh-2.6/drivers/char/sysrq.c
@@ -316,7 +316,7 @@ static struct sysrq_key_op *sysrq_key_ta
 	NULL,				/* v */
 	NULL,				/* w */
 	NULL,				/* x */
-	NULL,				/* y */
+	&sysrq_mountro_op,		/* y stupid fujitsu laptop, can't hit the U key */
 	NULL				/* z */
 };
 
