 kernel/sysctl.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

---
--- gregkh-2.6.orig/kernel/sysctl.c
+++ gregkh-2.6/kernel/sysctl.c
@@ -125,7 +125,7 @@ extern int sysctl_hz_timer;
 extern int acct_parm[];
 #endif
 
-int randomize_va_space = 1;
+int randomize_va_space = 0;
 
 static int parse_table(int __user *, int, void __user *, size_t __user *, void __user *, size_t,
 		       ctl_table *, void **);
