From vapier.adi@gmail.com  Tue Jun  2 15:21:42 2009
From: Bryan Wu <cooloney@kernel.org>
Date: Tue, 2 Jun 2009 03:25:58 -0400
Subject: usb: misc: SiS usbvga dangle: accept MUSB_HDRC as a fast enough host controller
To: felipe.balbi@nokia.com, greg@kroah.com
Cc: Bryan Wu <cooloney@kernel.org>, linux-usb@vger.kernel.org
Message-ID: <8bd0f97a0906020025t4662612cgaed7d6f616ff0125@mail.gmail.com>

From: Bryan Wu <cooloney@kernel.org>
Acked-by: Mike Frysinger <vapier.adi@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/sisusbvga/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/misc/sisusbvga/Kconfig
+++ b/drivers/usb/misc/sisusbvga/Kconfig
@@ -1,7 +1,7 @@
 
 config USB_SISUSBVGA
 	tristate "USB 2.0 SVGA dongle support (Net2280/SiS315)"
-	depends on USB && USB_EHCI_HCD
+	depends on USB && (USB_MUSB_HDRC || USB_EHCI_HCD)
         ---help---
 	  Say Y here if you intend to attach a USB2VGA dongle based on a
 	  Net2280 and a SiS315 chip.
