From dbaryshkov@gmail.com  Fri Aug  8 16:18:49 2008
From: Dmitry Baryshkov <dbaryshkov@gmail.com>
Date: Thu,  7 Aug 2008 16:29:25 +0400
Subject: USB: omap_udc: fix compilation with debug enabled
To: linux-usb@vger.kernel.org
Cc: David Brownell <david-b@pacbell.net>, Dmitry Baryshkov <dbaryshkov@gmail.com>
Message-ID: <619853cc417c86624ba22c87c2f0e7f1ad022895.1218111855.git.dbaryshkov@gmail.com>


Signed-off-by: Dmitry Baryshkov <dbaryshkov@gmail.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/omap_udc.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/usb/gadget/omap_udc.c
+++ b/drivers/usb/gadget/omap_udc.c
@@ -54,6 +54,7 @@
 
 #include <mach/dma.h>
 #include <mach/usb.h>
+#include <mach/control.h>
 
 #include "omap_udc.h"
 
@@ -2310,10 +2311,10 @@ static int proc_otg_show(struct seq_file
 	u32		trans;
 	char		*ctrl_name;
 
-	tmp = OTG_REV_REG;
+	tmp = omap_readl(OTG_REV);
 	if (cpu_is_omap24xx()) {
 		ctrl_name = "control_devconf";
-		trans = CONTROL_DEVCONF_REG;
+		trans = omap_ctrl_readl(OMAP2_CONTROL_DEVCONF0);
 	} else {
 		ctrl_name = "tranceiver_ctrl";
 		trans = omap_readw(USB_TRANSCEIVER_CTRL);
