From dbaryshkov@gmail.com  Thu Aug  7 15:07:04 2008
From: Dmitry Baryshkov <dbaryshkov@gmail.com>
Date: Thu,  7 Aug 2008 16:29:24 +0400
Subject: USB: isp1301: fix compilation
To: linux-usb@vger.kernel.org
Cc: David Brownell <david-b@pacbell.net>, Dmitry Baryshkov <dbaryshkov@gmail.com>
Message-ID: <c4d733ec759f91078f31fdf48c948962983f7cc8.1218111855.git.dbaryshkov@gmail.com>


Signed-off-by: Dmitry Baryshkov <dbaryshkov@gmail.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/i2c/chips/isp1301_omap.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/i2c/chips/isp1301_omap.c
+++ b/drivers/i2c/chips/isp1301_omap.c
@@ -1593,7 +1593,7 @@ fail1:
 	if (machine_is_omap_h2()) {
 		/* full speed signaling by default */
 		isp1301_set_bits(isp, ISP1301_MODE_CONTROL_1,
-			MC1_SPEED_REG);
+			MC1_SPEED);
 		isp1301_set_bits(isp, ISP1301_MODE_CONTROL_2,
 			MC2_SPD_SUSP_CTRL);
 
