From akpm@linux-foundation.org  Wed Nov 12 11:42:45 2008
From: Albert Comerma <albert@ecm.ub.es>
Date: Tue, 04 Nov 2008 10:44:01 -0800
Subject: USB: SISUSB2VGA driver: add 0x0711, 0x0903
To: mm-commits@vger.kernel.org
Cc: albert@ecm.ub.es, alan@lxorguk.ukuu.org.uk, albert.comerma@gmail.com, david-b@pacbell.net, greg@kroah.com, mchehab@infradead.org
Message-ID: <200811041844.mA4Ii14X028631@imap1.linux-foundation.org>


From: Albert Comerma <albert@ecm.ub.es>

Signed-off-by: Albert Comerma <albert.comerma@gmail.com>
Cc: Alan Cox <alan@lxorguk.ukuu.org.uk>
Cc: David Brownell <david-b@pacbell.net>
Cc: Mauro Carvalho Chehab <mchehab@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/sisusbvga/sisusb.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/misc/sisusbvga/sisusb.c
+++ b/drivers/usb/misc/sisusbvga/sisusb.c
@@ -3270,6 +3270,7 @@ static struct usb_device_id sisusb_table
 	{ USB_DEVICE(0x0711, 0x0900) },
 	{ USB_DEVICE(0x0711, 0x0901) },
 	{ USB_DEVICE(0x0711, 0x0902) },
+	{ USB_DEVICE(0x0711, 0x0903) },
 	{ USB_DEVICE(0x0711, 0x0918) },
 	{ USB_DEVICE(0x182d, 0x021c) },
 	{ USB_DEVICE(0x182d, 0x0269) },
