From cooloney@kernel.org  Fri Feb  6 16:01:03 2009
From: Ivan Kuten <ivan.kuten@promwad.com>
Date: Fri,  6 Feb 2009 17:42:34 +0800
Subject: USB: Correct Makefile to make isp1760 buildable
To: greg@kroah.com
Cc: Ivan Kuten <ivan.kuten@promwad.com>, Michael Hennerich <michael.hennerich@analog.com>, Bryan Wu <cooloney@kernel.org>
Message-ID: <1233913354-13755-1-git-send-email-cooloney@kernel.org>


From: Ivan Kuten <ivan.kuten@promwad.com>

Signed-off-by: Ivan Kuten <ivan.kuten@promwad.com>
Signed-off-by: Michael Hennerich <michael.hennerich@analog.com>
Signed-off-by: Bryan Wu <cooloney@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/Makefile |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/Makefile
+++ b/drivers/usb/Makefile
@@ -11,6 +11,7 @@ obj-$(CONFIG_USB_MON)		+= mon/
 obj-$(CONFIG_PCI)		+= host/
 obj-$(CONFIG_USB_EHCI_HCD)	+= host/
 obj-$(CONFIG_USB_ISP116X_HCD)	+= host/
+obj-$(CONFIG_USB_ISP1760_HCD)	+= host/
 obj-$(CONFIG_USB_OHCI_HCD)	+= host/
 obj-$(CONFIG_USB_UHCI_HCD)	+= host/
 obj-$(CONFIG_USB_FHCI_HCD)	+= host/
