From oliver@neukum.org  Thu Jul  2 15:52:21 2009
From: Oliver Neukum <oliver@neukum.org>
Date: Tue, 30 Jun 2009 09:44:24 +0200
Subject: USB: fix uninitialised variable in ti_do_download
To: Andrew Morton <akpm@linux-foundation.org>, greg@kroah.com, USB list <linux-usb@vger.kernel.org>, stable@kernel.org
Message-ID: <200906300944.24551.oliver@neukum.org>
Content-Disposition: inline



Signed-off-by: Oliver Neukum <oliver@neukum.org>
Cc: stable <stable@kernel.org>
Cc: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/serial/ti_usb_3410_5052.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/ti_usb_3410_5052.c
+++ b/drivers/usb/serial/ti_usb_3410_5052.c
@@ -1657,7 +1657,7 @@ static int ti_do_download(struct usb_dev
 	u8 cs = 0;
 	int done;
 	struct ti_firmware_header *header;
-	int status;
+	int status = 0;
 	int len;
 
 	for (pos = sizeof(struct ti_firmware_header); pos < size; pos++)
