From dbaryshkov@gmail.com  Wed Aug  6 16:41:16 2008
From: Dmitry Baryshkov <dbaryshkov@gmail.com>
Date: Sun, 6 Jul 2008 14:25:53 +0400
Subject: USB: omap_udc: fix compilation with debug enabled
To: linux-usb@vger.kernel.org
Message-ID: <20080706102553.GA29914@doriath.ww600.siemens.net>
Content-Disposition: inline


Signed-off-by: Dmitry Baryshkov <dbaryshkov@gmail.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Cc: stable <stable@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/omap_udc.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/usb/gadget/omap_udc.c
+++ b/drivers/usb/gadget/omap_udc.c
@@ -54,6 +54,7 @@
 
 #include <asm/arch/dma.h>
 #include <asm/arch/usb.h>
+#include <asm/arch/control.h>
 
 #include "omap_udc.h"
 
@@ -2313,7 +2314,7 @@ static int proc_otg_show(struct seq_file
 	tmp = OTG_REV_REG;
 	if (cpu_is_omap24xx()) {
 		ctrl_name = "control_devconf";
-		trans = CONTROL_DEVCONF_REG;
+		trans = omap_ctrl_readl(OMAP2_CONTROL_DEVCONF0);
 	} else {
 		ctrl_name = "tranceiver_ctrl";
 		trans = omap_readw(USB_TRANSCEIVER_CTRL);
