From dfo@antex.ru  Fri Oct  9 11:00:36 2009
From: Sergey Pinaev <dfo@antex.ru>
Date: Thu, 17 Sep 2009 17:26:50 +0400
Subject: USB: storage: iRiver P7 UNUSUAL_DEV patch
To: Phil Dibowitz <phil@ipom.com>
Cc: linux-usb@vger.kernel.org, usb-storage@lists.one-eyed-alien.net
Message-ID: <20090917172650.125b356f.dfo@antex.ru>


From: Sergey Pinaev <dfo@antex.ru>
Signed-off-by: Phil Dibowitz <phil@ipom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/unusual_devs.h |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/drivers/usb/storage/unusual_devs.h
+++ b/drivers/usb/storage/unusual_devs.h
@@ -1823,6 +1823,13 @@ UNUSUAL_DEV(  0x4102, 0x1020, 0x0100,  0
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_IGNORE_RESIDUE ),
 
+/* Reported by Sergey Pinaev <dfo@antex.ru> */
+UNUSUAL_DEV(  0x4102, 0x1059, 0x0000,  0x0000,
+               "iRiver",
+               "P7K",
+               US_SC_DEVICE, US_PR_DEVICE, NULL,
+               US_FL_MAX_SECTORS_64 ),
+
 /*
  * David Härdeman <david@2gen.com>
  * The key makes the SCSI stack print confusing (but harmless) messages
