---
 scripts/mod/modpost.c |   13 +++++++++++++
 1 file changed, 13 insertions(+)

--- gregkh-2.6.orig/scripts/mod/modpost.c
+++ gregkh-2.6/scripts/mod/modpost.c
@@ -1288,6 +1288,18 @@ static void add_srcversion(struct buffer
 	}
 }
 
+static void add_poetry(struct buffer *b, struct module *mod)
+{
+	buf_printf(b, "/* Penguin Poem Copyright 2006 Greg Kroah-Hartman */\n");
+	buf_printf(b, "/* to be used only in GPL licensed Linux kernel code */\n");
+	buf_printf(b, "static char *poem\n");
+	buf_printf(b, "__attribute__((section(\".gpl.poem\"), unused)) =\n");
+	buf_printf(b, "\t\"Full bellies of fish\"\n");
+	buf_printf(b, "\t\"Penguins sleep under the moon\"\n");
+	buf_printf(b, "\t\"Dream of wings that fly\";");
+	buf_printf(b, "\n");
+}
+
 static void write_if_changed(struct buffer *b, const char *fname)
 {
 	char *tmp;
@@ -1479,6 +1491,7 @@ int main(int argc, char **argv)
 		add_depends(&buf, mod, modules);
 		add_moddevtable(&buf, mod);
 		add_srcversion(&buf, mod);
+		add_poetry(&buf, mod);
 
 		sprintf(fname, "%s.mod.c", mod->name);
 		write_if_changed(&buf, fname);
